--- /home/benjaminm/code/oops-bundle/ecsaber/oops_patch/src/oops/interface/LinearVariableChange.h.tmp.bak	2025-02-21 08:55:45.107465617 +0100
+++ /home/benjaminm/code/oops-bundle/ecsaber/oops_patch/src/oops/interface/LinearVariableChange.h	2025-02-03 13:49:14.066078079 +0100
@@ -13,11 +13,19 @@
 
 #include <boost/noncopyable.hpp>
 
-#include "oops/base/Geometry.h"
-#include "oops/base/Increment.h"
-#include "oops/base/State.h"
+#include "oops/interface/Geometry.h"
+#include "oops/interface/Increment.h"
+#include "oops/base/LinearVariableChangeParametersBase.h"
+#include "oops/interface/State.h"
+#include "oops/interface/Variables.h"
 #include "oops/base/Variables.h"
 #include "oops/util/Logger.h"
+#include "oops/util/ObjectCounter.h"
+#include "oops/util/parameters/ConfigurationParameter.h"
+#include "oops/util/parameters/GenericParameters.h"
+#include "oops/util/parameters/HasParameters_.h"
+#include "oops/util/parameters/Parameters.h"
+#include "oops/util/parameters/ParametersOrConfiguration.h"
 #include "oops/util/Printable.h"
 #include "oops/util/Timer.h"
 
@@ -30,25 +38,38 @@
 // -----------------------------------------------------------------------------
 /// \brief MODEL-agnostic part of the linear variable change
 ///
+/// Note: each implementation should typedef `LinVariableChangeParams_` to the name of a
+/// subclass of oops::LinearVariableChangeParametersBase holding its configuration settings and
+/// provide a constructor with the following signature:
+///
+///     LinearVariableChange(const Geometry_ &, const LinVariableChangeParams_ &);
 template <typename MODEL>
 class LinearVariableChange {
   typedef typename MODEL::LinearVariableChange LinearVariableChange_;
   typedef Geometry<MODEL>            Geometry_;
   typedef Increment<MODEL>           Increment_;
   typedef State<MODEL>               State_;
+  typedef Variables<MODEL>           Variables_;
 
  public:
+  /// A subclass of oops::LinearVariableChangeParametersBase holding the configuration settings of
+  /// the variable change.
+  /// Defined as LinearVariableChange_::Parameters_ if LinearVariableChange_ defines a Parameters_
+  /// type; otherwise as GenericLinearVariableChangeParameters.
+  typedef TParameters_IfAvailableElseFallbackType_t<LinearVariableChange_,
+                                    GenericLinearVariableChangeParameters> Parameters_;
   static const std::string classname() {return "oops::LinearVariableChange";}
 
+  LinearVariableChange(const Geometry_ &, const Parameters_ &);
   LinearVariableChange(const Geometry_ &, const eckit::Configuration &);
 
   virtual ~LinearVariableChange();
 
-  void changeVarTraj(const State_ &, const JediVariables &);
-  void changeVarTL(Increment_ &, const JediVariables &) const;
-  void changeVarInverseTL(Increment_ &, const JediVariables &) const;
-  void changeVarAD(Increment_ &, const JediVariables &) const;
-  void changeVarInverseAD(Increment_ &, const JediVariables &) const;
+  void changeVarTraj(const State_ &, const Variables_ &);
+  void changeVarTL(Increment_ &, const Variables_ &) const;
+  void changeVarInverseTL(Increment_ &, const Variables_ &) const;
+  void changeVarAD(Increment_ &, const Variables_ &) const;
+  void changeVarInverseAD(Increment_ &, const Variables_ &) const;
 
  private:
   void print(std::ostream &) const;
@@ -70,6 +91,14 @@
 // -----------------------------------------------------------------------------
 
 template<typename MODEL>
+LinearVariableChange<MODEL>::LinearVariableChange(const Geometry_ & resol,
+                                                  const Parameters_ & parameters)
+  :  LinearVariableChange<MODEL>::LinearVariableChange(resol, parameters.toConfiguration())
+{}
+
+// -----------------------------------------------------------------------------
+
+template<typename MODEL>
 LinearVariableChange<MODEL>::~LinearVariableChange() {
   Log::trace() << "LinearVariableChange<MODEL>::~LinearVariableChange starting" << std::endl;
   util::Timer timer(classname(), "~LinearVariableChange");
@@ -80,30 +109,30 @@
 // -----------------------------------------------------------------------------
 
 template<typename MODEL>
-void LinearVariableChange<MODEL>::changeVarTL(Increment_ & dx, const JediVariables & vars) const {
+void LinearVariableChange<MODEL>::changeVarTL(Increment_ & dx, const Variables_ & vars) const {
   Log::trace() << "LinearVariableChange<MODEL>::changeVarTL starting" << std::endl;
   util::Timer timer(classname(), "changeVarTL");
-  chvar_->changeVarTL(dx.increment(), vars);
+  chvar_->changeVarTL(dx.increment(), vars.variables());
   Log::trace() << "LinearVariableChange<MODEL>::changeVarTL done" << std::endl;
 }
 // -----------------------------------------------------------------------------
 
 template<typename MODEL>
 void LinearVariableChange<MODEL>::changeVarInverseTL(Increment_ & dx,
-                                                     const JediVariables & vars) const {
+                                                     const Variables_ & vars) const {
   Log::trace() << "LinearVariableChange<MODEL>::changeVarInverseTL starting" << std::endl;
   util::Timer timer(classname(), "changeVarInverseTL");
-  chvar_->changeVarInverseTL(dx.increment(), vars);
+  chvar_->changeVarInverseTL(dx.increment(), vars.variables());
   Log::trace() << "LinearVariableChange<MODEL>::changeVarInverseTL done" << std::endl;
 }
 
 // -----------------------------------------------------------------------------
 
 template<typename MODEL>
-void LinearVariableChange<MODEL>::changeVarAD(Increment_ & dx, const JediVariables & vars) const {
+void LinearVariableChange<MODEL>::changeVarAD(Increment_ & dx, const Variables_ & vars) const {
   Log::trace() << "LinearVariableChange<MODEL>::changeVarAD starting" << std::endl;
   util::Timer timer(classname(), "changeVarAD");
-  chvar_->changeVarAD(dx.increment(), vars);
+  chvar_->changeVarAD(dx.increment(), vars.variables());
   Log::trace() << "LinearVariableChange<MODEL>::changeVarAD done" << std::endl;
 }
 
@@ -111,10 +140,10 @@
 
 template<typename MODEL>
 void LinearVariableChange<MODEL>::changeVarInverseAD(Increment_ & dx,
-                                                     const JediVariables & vars) const {
+                                                     const Variables_ & vars) const {
   Log::trace() << "LinearVariableChange<MODEL>::changeVarInverseAD starting" << std::endl;
   util::Timer timer(classname(), "changeVarInverseAD");
-  chvar_->changeVarInverseAD(dx.increment(), vars);
+  chvar_->changeVarInverseAD(dx.increment(), vars.variables());
   Log::trace() << "LinearVariableChange<MODEL>::changeVarInverseAD done" << std::endl;
 }
 
@@ -122,10 +151,10 @@
 
 template<typename MODEL>
 void LinearVariableChange<MODEL>::changeVarTraj(const State_ & xFirstGuess,
-                                                const JediVariables & vars) {
+                                                const Variables_ & vars) {
   Log::trace() << "LinearVariableChange<MODEL>::changeVarTraj starting" << std::endl;
   util::Timer timer(classname(), "changeVarTraj");
-  chvar_->changeVarTraj(xFirstGuess.state(), vars);
+  chvar_->changeVarTraj(xFirstGuess.state(), vars.variables());
   Log::trace() << "LinearVariableChange<MODEL>::changeVarTraj done" << std::endl;
 }
 
