--- /home/benjaminm/code/oops-bundle/ecsaber/oops_patch/src/oops/interface/VariableChange.h.tmp	2025-11-05 12:49:56.397719994 +0100
+++ /home/benjaminm/code/oops-bundle/ecsaber/oops_patch/src/oops/interface/VariableChange.h	2025-02-03 13:49:14.066078079 +0100
@@ -12,7 +12,9 @@
 #include <string>
 
 #include "eckit/config/Configuration.h"
-#include "oops/base/State.h"
+#include "oops/interface/Variables.h"
+#include "oops/interface/State.h"
+#include "oops/util/ECUtilities.h"
 #include "oops/util/Logger.h"
 #include "oops/util/ObjectCounter.h"
 #include "oops/util/Printable.h"
@@ -79,7 +81,8 @@
 void VariableChange<MODEL>::changeVar(State_ & xx, const JediVariables & vars) const {
   Log::trace() << "VariableChange<MODEL>::changeVar starting" << std::endl;
   util::Timer timer(classname(), "changeVar");
-  chvar_->changeVar(xx.state(), vars);
+  Variables<MODEL> varsT(util::templatedVarsConf(vars));
+  chvar_->changeVar(xx.state(), varsT.variables());
   Log::trace() << "VariableChange<MODEL>::changeVar done" << std::endl;
 }
 
@@ -89,7 +92,8 @@
 void VariableChange<MODEL>::changeVarInverse(State_ & xx, const JediVariables & vars) const {
   Log::trace() << "VariableChange<MODEL>::changeVarInverse starting" << std::endl;
   util::Timer timer(classname(), "changeVarInverse");
-  chvar_->changeVarInverse(xx.state(), vars);
+  Variables<MODEL> varsT(util::templatedVarsConf(vars));
+  chvar_->changeVarInverse(xx.state(), varsT.variables());
   Log::trace() << "VariableChange<MODEL>::changeVarInverse done" << std::endl;
 }
 
