--- /home/benjaminm/code/oops-bundle/ecsaber/oops_patch/src/oops/base/DeparturesEnsemble.h.tmp.bak	2025-02-21 14:04:24.878874747 +0100
+++ /home/benjaminm/code/oops-bundle/ecsaber/oops_patch/src/oops/base/DeparturesEnsemble.h	2025-02-21 14:25:56.563419730 +0100
@@ -13,8 +13,8 @@
 #include <vector>
 
 #include "oops/base/Departures.h"
-#include "oops/base/ObsSpaces.h"
-#include "oops/interface/ObsDataVector.h"
+#include "oops/base/ObservationSpaces.h"
+#include "oops/util/ECUtilities.h"
 #include "oops/util/Logger.h"
 
 namespace oops {
@@ -22,9 +22,9 @@
 // -----------------------------------------------------------------------------
 
 /// \brief Ensemble of Departures (can hold ensemble perturbations in the observation space)
-template<typename OBS> class DeparturesEnsemble {
-  typedef Departures<OBS>          Departures_;
-  typedef ObsSpaces<OBS>           ObsSpaces_;
+template<typename MODEL> class DeparturesEnsemble {
+  typedef Departures<MODEL>          Departures_;
+  typedef ObservationSpaces<MODEL>   ObsSpaces_;
   template <typename DATA> using ObsData_ = ObsDataVector<OBS, DATA>;
   template <typename DATA> using ObsDataVec_ = std::vector<std::shared_ptr<ObsData_<DATA>>>;
 
@@ -46,8 +46,8 @@
 
 // ====================================================================================
 
-template<typename OBS>
-DeparturesEnsemble<OBS>::DeparturesEnsemble(const ObsSpaces_ & obsdb, const size_t nens)
+template<typename MODEL>
+DeparturesEnsemble<MODEL>::DeparturesEnsemble(const ObsSpaces_ & obsdb, const size_t nens)
       : ensemblePerturbs_() {
   ensemblePerturbs_.reserve(nens);
   for (size_t iens = 0; iens < nens; ++iens) {
@@ -58,15 +58,15 @@
 
 // -----------------------------------------------------------------------------
 
-template<typename OBS>
-Eigen::MatrixXd DeparturesEnsemble<OBS>::packEigen(const Departures_ & mask) const {
+template<typename MODEL>
+Eigen::MatrixXd DeparturesEnsemble<MODEL>::packEigen(const Departures_ & mask) const {
   const size_t myNens = ensemblePerturbs_.size();
-  const Eigen::VectorXd dep0 = ensemblePerturbs_.front().packEigen(mask);
+  const Eigen::VectorXd dep0 = util::packEigen(ensemblePerturbs_.front(), mask);
   const size_t myNobs = dep0.size();
   Eigen::MatrixXd depEns(myNens, myNobs);
   depEns.row(0) = dep0;
   for (std::size_t iens = 1; iens < myNens; ++iens) {
-    depEns.row(iens) = ensemblePerturbs_[iens].packEigen(mask);
+    depEns.row(iens) = util::packEigen(ensemblePerturbs_[iens], mask);
   }
   Log::trace() << "DeparturesEnsemble::packEigen() completed" << std::endl;
   return depEns;
