--- /home/benjaminm/code/oops-bundle/ecsaber/oops_patch/src/oops/base/FieldSet3D.cc.tmp.bak	2026-01-14 15:26:50.011551083 +0100
+++ /home/benjaminm/code/oops-bundle/ecsaber/oops_patch/src/oops/base/FieldSet3D.cc	2026-01-14 10:07:10.364585105 +0100
@@ -20,6 +20,7 @@
 #include "eckit/exception/Exceptions.h"
 
 #include "oops/base/Variable.h"
+#include "oops/util/ECUtilities.h"
 #include "oops/util/FieldSetHelpers.h"
 #include "oops/util/FieldSetOperations.h"
 #include "oops/util/Logger.h"
