--- /home/benjaminm/code/oops-bundle/ecsaber/oops_patch/src/oops/base/IncrementEnsemble.h.tmp	2025-01-31 15:41:04.735508720 +0100
+++ /home/benjaminm/code/oops-bundle/ecsaber/oops_patch/src/oops/base/IncrementEnsemble.h	2025-01-30 14:08:26.750809104 +0100
@@ -19,14 +19,16 @@
 #include <vector>
 
 #include "eckit/config/LocalConfiguration.h"
-#include "oops/base/Geometry.h"
-#include "oops/base/Increment.h"
+#include "oops/interface/Geometry.h"
+#include "oops/interface/Increment.h"
 #include "oops/base/LocalIncrement.h"
-#include "oops/base/State.h"
+#include "oops/interface/State.h"
 #include "oops/base/StateEnsemble.h"
 #include "oops/base/Variables.h"
+#include "oops/interface/Variables.h"
 #include "oops/util/ConfigFunctions.h"
 #include "oops/util/DateTime.h"
+#include "oops/util/ECUtilities.h"
 #include "oops/util/Logger.h"
 #include "oops/util/parameters/OptionalParameter.h"
 #include "oops/util/parameters/Parameter.h"
@@ -215,8 +217,9 @@
   : ensemblePerturbs_()
 {
   ensemblePerturbs_.reserve(rank);
+  const oops::Variables<MODEL> varsT(templatedVarsConf(vars));
   for (int m = 0; m < rank; ++m) {
-    ensemblePerturbs_.emplace_back(resol, vars, time);
+    ensemblePerturbs_.emplace_back(resol, varsT, time);
   }
   Log::trace() << "IncrementEnsemble:contructor done" << std::endl;
 }
