--- /home/benjaminm/code/oops-bundle/ecsaber/oops_patch/src/oops/base/IncrementSet.h.tmp	2025-02-27 14:15:30.020114892 +0100
+++ /home/benjaminm/code/oops-bundle/ecsaber/oops_patch/src/oops/base/IncrementSet.h	2025-02-27 11:07:10.203179409 +0100
@@ -14,8 +14,8 @@
 #include <vector>
 
 #include "oops/base/DataSetBase.h"
-#include "oops/base/Geometry.h"
-#include "oops/base/Increment.h"
+#include "oops/interface/Geometry.h"
+#include "oops/interface/Increment.h"
 #include "oops/base/StateSet.h"
 #include "oops/mpi/mpi.h"
 #include "oops/util/DateTime.h"
