--- /home/benjaminm/code/oops-bundle/ecsaber/oops_patch/src/oops/base/LocalIncrement.h.tmp	2024-11-21 11:22:57.056425958 +0100
+++ /home/benjaminm/code/oops-bundle/ecsaber/oops_patch/src/oops/base/LocalIncrement.h	2024-11-20 14:39:39.531564203 +0100
@@ -34,7 +34,7 @@
   void print(std::ostream & os) const {
     os << "LocalIncrement, size: " << vals_.size() << ", first element: "
        << vals_[0] << std::endl; }
-  const oops::JediVariables vars_;      // variables in the object
+  const oops::JediVariables vars_;  // variables in the object
   std::vector<double> vals_;        // data in flat array
   const std::vector<int> varlens_;  // vector containing nlevs for each variable
 };
