--- /home/benjaminm/code/oops-bundle/ecsaber/oops_patch/src/oops/base/ObsEnsemble.h.tmp	2024-11-21 11:22:57.069426007 +0100
+++ /home/benjaminm/code/oops-bundle/ecsaber/oops_patch/src/oops/base/ObsEnsemble.h	2024-11-20 14:41:00.795851266 +0100
@@ -11,16 +11,16 @@
 #include <vector>
 
 #include "oops/base/Observations.h"
-#include "oops/base/ObsSpaces.h"
+#include "oops/base/ObservationSpaces.h"
 #include "oops/util/Logger.h"
 
 namespace oops {
 
 // -----------------------------------------------------------------------------
 /// \brief Ensemble of observations (can hold ensemble of H(x))
-template<typename OBS> class ObsEnsemble {
-  typedef Observations<OBS>        Observations_;
-  typedef ObsSpaces<OBS>           ObsSpaces_;
+template<typename MODEL> class ObsEnsemble {
+  typedef Observations<MODEL>      Observations_;
+  typedef ObservationSpaces<MODEL> ObsSpaces_;
 
  public:
   /// Create ensemble of empty Observations size \p nens
@@ -41,8 +41,8 @@
 
 // ====================================================================================
 
-template<typename OBS>
-ObsEnsemble<OBS>::ObsEnsemble(const ObsSpaces_ & obsdb, const size_t & nens)
+template<typename MODEL>
+ObsEnsemble<MODEL>::ObsEnsemble(const ObsSpaces_ & obsdb, const size_t & nens)
   : obsdb_(obsdb), ensemble_()
 {
   ensemble_.reserve(nens);
@@ -54,14 +54,23 @@
 
 // -----------------------------------------------------------------------------
 
-template<typename OBS>
-Observations<OBS> ObsEnsemble<OBS>::mean() const {
+template<typename MODEL>
+Observations<MODEL> ObsEnsemble<MODEL>::mean() const {
   Observations_ mean_obs(obsdb_);
-  mean_obs.zero();
+  for (size_t jj = 0; jj < mean_obs.size(); ++jj) {
+    mean_obs[jj].obsvector().zero();
+    if (mean_obs[jj].hasBias()) mean_obs[jj].biasvector().zero();
+  }
   for (const auto & yy : ensemble_) {
-    mean_obs.accumul(yy);
+    for (size_t jj = 0; jj < mean_obs.size(); ++jj) {
+      mean_obs[jj].obsvector() += yy[jj].obsvector();
+      if (mean_obs[jj].hasBias()) mean_obs[jj].biasvector() += yy[jj].biasvector();
+    }
+  }
+  for (size_t jj = 0; jj < mean_obs.size(); ++jj) {
+    mean_obs[jj].obsvector() *= 1.0/static_cast<float>(ensemble_.size());
+    if (mean_obs[jj].hasBias()) mean_obs[jj].biasvector() *= 1.0/static_cast<float>(ensemble_.size());
   }
-  mean_obs *= 1.0/static_cast<float>(ensemble_.size());
   Log::trace() << "ObsEnsemble::mean done" << std::endl;
   return mean_obs;
 }
