--- /home/benjaminm/code/oops-bundle/ecsaber/oops_patch/src/oops/base/ObsLocalizations.h.tmp.bak	2024-11-27 12:20:20.835073184 +0100
+++ /home/benjaminm/code/oops-bundle/ecsaber/oops_patch/src/oops/base/ObsLocalizations.h	2024-11-27 12:24:22.605354378 +0100
@@ -17,7 +17,8 @@
 
 #include "oops/base/Departures.h"
 #include "oops/base/ObsLocalizationBase.h"
-#include "oops/base/ObsSpaces.h"
+#include "oops/base/ObservationSpaces.h"
+#include "oops/util/ECUtilities.h"
 #include "oops/util/Printable.h"
 
 namespace oops {
@@ -26,20 +27,20 @@
 /// \brief Container for ObsLocalizations for all observation types that are used in DA
 ///        obsLocalizations will also loop through localizations specified for each obs space
 ///        and update localization factors in this loop.
-template <typename MODEL, typename OBS>
+template <typename MODEL>
 class ObsLocalizations : public util::Printable,
                          private boost::noncopyable {
-  typedef GeometryIterator<MODEL>  GeometryIterator_;
-  typedef Departures<OBS>          Observations_;
-  typedef ObsLocalizationBase<MODEL, OBS> ObsLocalization_;
-  typedef ObsSpaces<OBS>           ObsSpaces_;
+  typedef typename MODEL::GeometryIterator  GeometryIterator__;
+  typedef Departures<MODEL>          Observations_;
+  typedef ObsLocalizationBase<MODEL> ObsLocalization_;
+  typedef ObservationSpaces<MODEL>   ObsSpaces_;
 
  public:
   static const std::string classname() {return "oops::ObsLocalizations";}
 
   ObsLocalizations(const eckit::Configuration &, const ObsSpaces_ &);
 
-  void computeLocalization(const GeometryIterator_ & point,
+  void computeLocalization(const GeometryIterator__ & point,
                            Observations_ & obsvectors) const;
 
  private:
@@ -49,10 +50,10 @@
 
 // -----------------------------------------------------------------------------
 
-template <typename MODEL, typename OBS>
-ObsLocalizations<MODEL, OBS>::ObsLocalizations(const eckit::Configuration & config,
-                                               const ObsSpaces_ & obspaces) {
-  Log::trace() << "ObsLocalizations<MODEL, OBS>::create starting" << std::endl;
+template <typename MODEL>
+ObsLocalizations<MODEL>::ObsLocalizations(const eckit::Configuration & config,
+                                          const ObsSpaces_ & obspaces) {
+  Log::trace() << "ObsLocalizations<MODEL>::create starting" << std::endl;
   std::vector<eckit::LocalConfiguration> obsconf = config.getSubConfigurations();
   //  loop over ob spaces
   for (size_t jj = 0; jj < obsconf.size(); ++jj) {
@@ -61,33 +62,33 @@
                         obsconf[jj].getSubConfigurations("obs localizations");
     std::vector<std::unique_ptr<ObsLocalization_> > tmpVector;
     for (size_t oli = 0; oli < obsLocConfigs.size(); ++oli) {
-      tmpVector.emplace_back(ObsLocalizationFactory<MODEL, OBS>::
+      tmpVector.emplace_back(ObsLocalizationFactory<MODEL>::
                              create(obsLocConfigs[oli], obspaces[jj]));
     }
     //  move a vector of unique pointers form temp array to vector of vectors
     local_.emplace_back(std::move(tmpVector));
   }
-  Log::trace() << "ObsLocalizations<MODEL, OBS>::create done" << std::endl;
+  Log::trace() << "ObsLocalizations<MODEL>::create done" << std::endl;
 }
 
 // -----------------------------------------------------------------------------
 
-template <typename MODEL, typename OBS>
-void ObsLocalizations<MODEL, OBS>::computeLocalization(const GeometryIterator_ & point,
-                                                       Observations_ & locfactor) const {
+template <typename MODEL>
+void ObsLocalizations<MODEL>::computeLocalization(const GeometryIterator__ & point,
+                                                  Observations_ & locfactor) const {
   //  initialize locafactors to ones and then update them in the loop bellow
-  locfactor.ones();
+  util::ones(locfactor);
   for (size_t jj = 0; jj < local_.size(); ++jj) {
     for (size_t oli = 0; oli < local_[jj].size(); ++oli) {
-      if (local_[jj][oli]) local_[jj][oli]->computeLocalization(point, locfactor[jj]);
+      if (local_[jj][oli]) local_[jj][oli]->computeLocalization(point, locfactor[jj].obsvector());
     }
   }
 }
 
 // -----------------------------------------------------------------------------
 
-template<typename MODEL, typename OBS>
-void ObsLocalizations<MODEL, OBS>::print(std::ostream & os) const {
+template<typename MODEL>
+void ObsLocalizations<MODEL>::print(std::ostream & os) const {
   for (size_t jj = 0; jj < local_.size(); ++jj) {
     for (size_t oli = 0; oli < local_[jj].size(); ++oli) {
       if (local_[jj][oli]) {
