--- /home/benjaminm/code/oops-bundle/ecsaber/oops_patch/src/oops/base/StateEnsemble4D.h.tmp	2025-02-21 14:42:08.724963436 +0100
+++ /home/benjaminm/code/oops-bundle/ecsaber/oops_patch/src/oops/base/StateEnsemble4D.h	2025-02-21 14:56:12.148807771 +0100
@@ -14,8 +14,9 @@
 
 #include "eckit/config/LocalConfiguration.h"
 #include "oops/base/Accumulator.h"
-#include "oops/base/Geometry.h"
-#include "oops/base/State.h"
+#include "oops/interface/Geometry.h"
+#include "oops/interface/Model.h"
+#include "oops/interface/State.h"
 #include "oops/base/StateSet.h"
 #include "oops/util/abor1_cpp.h"
 #include "oops/util/ConfigFunctions.h"
@@ -30,14 +31,16 @@
 /// \brief Ensemble of 4D states
 template<typename MODEL> class StateEnsemble4D {
   typedef Geometry<MODEL>      Geometry_;
+  typedef Model<MODEL>         Model_;
   typedef StateSet<MODEL>      StateSet_;
   typedef State<MODEL>         State_;
 
  public:
   /// Create ensemble of 4D states
-  StateEnsemble4D(const Geometry_ &, const eckit::Configuration &);
+  StateEnsemble4D(const Geometry_ &, const Model_ &, const eckit::Configuration &);
 
   StateEnsemble4D(const Geometry_ & resol,
+                                     const Model_ & model,
                                      const eckit::Configuration & config,
                                      const JediVariables & vars,
                                      const std::vector<util::DateTime> & times,
@@ -61,7 +64,7 @@
   const StateSet_ & operator[](const size_t ii) const { return states_[ii]; }
 
   /// Information
-  const JediVariables & variables() const {return states_[0].variables();}
+  const JediVariables & variables() const {return variables_;}
 
  private:
   void getMembers(const eckit::Configuration &);
@@ -69,6 +72,7 @@
   // TODO(mpotts) remove std::vector<StateSet_> and replace it with a singular StateSet_
   // variable states_
   std::vector<StateSet_> states_;
+  JediVariables variables_;
 };
 
 // ====================================================================================
@@ -94,6 +98,7 @@
 
 template<typename MODEL>
 StateEnsemble4D<MODEL>::StateEnsemble4D(const Geometry_ & resol,
+                                        const Model_ & model,
                                         const eckit::Configuration & config,
                                         const JediVariables & vars,
                                         const std::vector<util::DateTime> & times,
@@ -109,7 +114,7 @@
   states_.reserve(times.size());
 
   // read in ensemble members on appropriate communicator
-  states_.emplace_back(StateSet_(resol, membersConfig[mymember-1]));
+  states_.emplace_back(StateSet_(resol, model, membersConfig[mymember-1]));
 
   Log::trace() << "StateEnsemble4D:contructor done" << std::endl;
 }
@@ -117,6 +122,7 @@
 // -----------------------------------------------------------------------------
 template<typename MODEL>
 StateEnsemble4D<MODEL>::StateEnsemble4D(const Geometry_ & resol,
+                                        const Model_ & model,
                                         const eckit::Configuration & config)
   : states_() {
   // Abort if both "members" and "members from template" are specified
@@ -127,7 +133,7 @@
 
   // Loop over all ensemble members
   for (size_t jj = 0; jj < membersConfig.size(); ++jj) {
-    states_.emplace_back(StateSet_(resol, membersConfig[jj]));
+    states_.emplace_back(StateSet_(resol, model, membersConfig[jj]));
   }
   Log::trace() << "StateEnsemble4D:contructor done" << std::endl;
 }
@@ -145,6 +151,20 @@
     ensmean.accumul(rr, states_[iens]);
   }
 
+/*
+  // Compute ensemble mean
+  State4D_ ensmean4D;
+  for (size_t jt = 0; jt < states_[0].statesNumber(); ++jt) {
+    Accumulator<MODEL, State_, State_> ensmean(states_[0][jt]);
+
+    const double rr = 1.0/static_cast<double>(states_.size());
+    for (size_t iens = 0; iens < states_.size(); ++iens) {
+      ensmean.accumul(rr, states_[iens][jt]);
+    }
+    ensmean4D.push_back(ensmean);
+  }
+*/
+
   Log::trace() << "StateEnsemble4D::mean done" << std::endl;
   return std::move(ensmean);
 }
@@ -194,6 +214,9 @@
   } else {
     ABORT("StateEnsemble4D: ensemble not specified");
   }
+
+  // Define variables
+  variables_ = JediVariables(states_[0][0].state().variables().variablesList());
 }
 
 }  // namespace oops
