--- /home/benjaminm/code/oops-bundle/ecsaber/oops_patch/src/oops/base/StateEnsemble4D.h.tmp.bak	2025-02-27 14:15:30.203115811 +0100
+++ /home/benjaminm/code/oops-bundle/ecsaber/oops_patch/src/oops/base/StateEnsemble4D.h	2025-02-27 13:43:33.492420868 +0100
@@ -14,9 +14,10 @@
 
 #include "eckit/config/LocalConfiguration.h"
 #include "oops/base/Accumulator.h"
-#include "oops/base/Geometry.h"
-#include "oops/base/State.h"
-#include "oops/base/StateSet.h"
+#include "oops/interface/Geometry.h"
+#include "oops/interface/Model.h"
+#include "oops/interface/State.h"
+#include "oops/base/State4D.h"
 #include "oops/util/abor1_cpp.h"
 #include "oops/util/ConfigFunctions.h"
 #include "oops/util/Logger.h"
@@ -30,14 +31,16 @@
 /// \brief Ensemble of 4D states
 template<typename MODEL> class StateEnsemble4D {
   typedef Geometry<MODEL>      Geometry_;
-  typedef StateSet<MODEL>      StateSet_;
+  typedef Model<MODEL>         Model_;
+  typedef State4D<MODEL>       State4D_;
   typedef State<MODEL>         State_;
 
  public:
   /// Create ensemble of 4D states
-  StateEnsemble4D(const Geometry_ &, const eckit::Configuration &);
+  StateEnsemble4D(const Geometry_ &, const Model_ &, const eckit::Configuration &);
 
   StateEnsemble4D(const Geometry_ & resol,
+                                     const Model_ & model,
                                      const eckit::Configuration & config,
                                      const JediVariables & vars,
                                      const std::vector<util::DateTime> & times,
@@ -48,33 +51,34 @@
 
   /// Create ensemble of 4D states
   StateEnsemble4D(const Geometry_ &, const eckit::Configuration &,
-                  StateSet_ & stateSet);
+                  State4D_ & stateSet);
 
-  explicit StateEnsemble4D(std::vector<StateSet_> & stateSetVec, const int ensNum);
+  explicit StateEnsemble4D(std::vector<State4D_> & stateSetVec, const int ensNum);
 
   /// calculate ensemble mean
-  StateSet_ mean() const;
+  State4D_ mean() const;
 
   /// Accessors
   unsigned int size() const { return states_.size(); }
-  StateSet_ & operator[](const size_t ii) { return states_[ii]; }
-  const StateSet_ & operator[](const size_t ii) const { return states_[ii]; }
+  State4D_ & operator[](const size_t ii) { return states_[ii]; }
+  const State4D_ & operator[](const size_t ii) const { return states_[ii]; }
 
   /// Information
-  const JediVariables & variables() const {return states_[0].variables();}
+  const JediVariables & variables() const {return variables_;}
 
  private:
   void getMembers(const eckit::Configuration &);
   std::vector<eckit::LocalConfiguration> membersConfig;
   // TODO(mpotts) remove std::vector<StateSet_> and replace it with a singular StateSet_
   // variable states_
-  std::vector<StateSet_> states_;
+  std::vector<State4D_> states_;
+  JediVariables variables_;
 };
 
 // ====================================================================================
 
 template<typename MODEL>
-StateEnsemble4D<MODEL>::StateEnsemble4D(std::vector<StateSet_> & stateSetVec, const int ensNum) :
+StateEnsemble4D<MODEL>::StateEnsemble4D(std::vector<State4D_> & stateSetVec, const int ensNum) :
   states_(stateSetVec) {
   Log::trace() << "StateEnsemble4D:contructor done" << std::endl;
 }
@@ -83,7 +87,7 @@
 
 template<typename MODEL>
 StateEnsemble4D<MODEL>::StateEnsemble4D(const Geometry_ & resol,
-      const eckit::Configuration & config, StateSet_ & stateSet) : states_() {
+      const eckit::Configuration & config, State4D_ & stateSet) : states_() {
   Log::trace() << "StateEnsemble4D:contructor starting" << std::endl;
   states_.emplace_back(stateSet);
 
@@ -94,6 +98,7 @@
 
 template<typename MODEL>
 StateEnsemble4D<MODEL>::StateEnsemble4D(const Geometry_ & resol,
+                                        const Model_ & model,
                                         const eckit::Configuration & config,
                                         const JediVariables & vars,
                                         const std::vector<util::DateTime> & times,
@@ -109,7 +114,10 @@
   states_.reserve(times.size());
 
   // read in ensemble members on appropriate communicator
-  states_.emplace_back(StateSet_(resol, membersConfig[mymember-1]));
+  states_.emplace_back(State4D_(membersConfig[mymember-1], resol, model));
+
+  // Define variables
+  variables_ = JediVariables(states_[0][0].state().variables().variablesList());
 
   Log::trace() << "StateEnsemble4D:contructor done" << std::endl;
 }
@@ -117,6 +125,7 @@
 // -----------------------------------------------------------------------------
 template<typename MODEL>
 StateEnsemble4D<MODEL>::StateEnsemble4D(const Geometry_ & resol,
+                                        const Model_ & model,
                                         const eckit::Configuration & config)
   : states_() {
   // Abort if both "members" and "members from template" are specified
@@ -127,26 +136,33 @@
 
   // Loop over all ensemble members
   for (size_t jj = 0; jj < membersConfig.size(); ++jj) {
-    states_.emplace_back(StateSet_(resol, membersConfig[jj]));
+    states_.emplace_back(State4D_(membersConfig[jj], resol, model));
   }
+
+  // Define variables
+  variables_ = JediVariables(states_[0][0].state().variables().variablesList());
+
   Log::trace() << "StateEnsemble4D:contructor done" << std::endl;
 }
 
 // -----------------------------------------------------------------------------
 
 template<typename MODEL>
-StateSet<MODEL> StateEnsemble4D<MODEL>::mean() const {
+State4D<MODEL> StateEnsemble4D<MODEL>::mean() const {
   // Compute ensemble mean
-  Accumulator<MODEL, StateSet_, StateSet_> ensmean(states_[0]);
-
-  const double rr = 1.0/static_cast<double>(states_.size());
-  Log::info() << "calculating mean with states_.size() of " << states_.size() << std::endl;
-  for (size_t iens = 0; iens < states_.size(); ++iens) {
-    ensmean.accumul(rr, states_[iens]);
+  State4D_ ensmean4D;
+  for (size_t jt = 0; jt < states_[0].statesNumber(); ++jt) {
+    Accumulator<MODEL, State_, State_> ensmean(states_[0][jt]);
+
+    const double rr = 1.0/static_cast<double>(states_.size());
+    for (size_t iens = 0; iens < states_.size(); ++iens) {
+      ensmean.accumul(rr, states_[iens][jt]);
+    }
+    ensmean4D.push_back(ensmean);
   }
 
   Log::trace() << "StateEnsemble4D::mean done" << std::endl;
-  return std::move(ensmean);
+  return ensmean4D;
 }
 
 // -----------------------------------------------------------------------------
