--- /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/vader/CMakeLists.txt.tmp	2024-07-04 13:56:18.784081654 +0200
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/vader/CMakeLists.txt	2024-07-04 13:01:51.027050842 +0200
@@ -30,8 +30,6 @@
     SuperMoistIncrOp.cc
     SuperMoistIncrOp.h
     PressureParameters.h
-    WriteVariances.cc
-    WriteVariances.h
     )
 
     # Generic VADER blocks
@@ -41,4 +39,9 @@
     )
 endif()
 
+list(APPEND vader_src_files_list
+WriteVariances.cc
+WriteVariances.h
+)
+
 set( vader_src_files ${vader_src_files_list} PARENT_SCOPE )
