--- /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/vader/DryAirDensity.h.tmp.bak	2024-04-15 12:07:59.490440037 +0200
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/vader/DryAirDensity.h	2024-04-15 12:08:38.819393639 +0200
@@ -23,9 +23,7 @@
 #include "saber/blocks/SaberOuterBlockBase.h"
 
 namespace oops {
-  namespace patch{
-class Variables;
-}
+  namespace patch {class Variables;}
 }
 
 namespace saber {
