--- /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/vader/GpToHp.h.tmp.bak	2024-03-07 14:56:01.868890803 +0100
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/vader/GpToHp.h	2024-03-07 12:46:13.464972875 +0100
@@ -27,9 +27,7 @@
 #include "saber/vader/PressureParameters.h"
 
 namespace oops {
-  namespace patch{
-class Variables;
-}
+  namespace patch {class Variables;}
 }
 
 namespace saber {
