--- /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/vader/HydroBal.h.tmp.bak	2024-03-21 09:17:23.681575884 +0100
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/vader/HydroBal.h	2024-03-21 09:18:21.573604817 +0100
@@ -23,9 +23,7 @@
 #include "saber/blocks/SaberOuterBlockBase.h"
 
 namespace oops {
-  namespace patch{
-class Variables;
-}
+  namespace patch {class Variables;}
 }
 
 namespace saber {
