--- /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/vader/MoistureControl.cc.tmp	2024-04-04 15:19:51.833074133 +0200
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/vader/MoistureControl.cc	2024-04-04 15:04:30.063855636 +0200
@@ -63,7 +63,8 @@
   oops::Log::trace() << classname() << "::MoistureControl starting" << std::endl;
 
   // Covariance FieldSet
-  covFieldSet_ = createMuStats(outerGeometryData.fieldSet(),
+  covFieldSet_ = createMuStats(xb["air_temperature"].levels(),
+                               outerGeometryData.fieldSet(),
                                params.moistureControlParams.value());
 
   std::vector<std::string> requiredStateVariables{
@@ -189,7 +190,8 @@
 
 // -----------------------------------------------------------------------------
 
-atlas::FieldSet createMuStats(const atlas::FieldSet & fields,
+atlas::FieldSet createMuStats(const size_t & modelLevelsDefault,
+                              const atlas::FieldSet & fields,
                               const MoistureControlCovarianceParameters & params) {
   // Get necessary parameters
   // path to covariance file with gp covariance parameters.
@@ -199,7 +201,7 @@
   if (fields.has("height")) {
     modelLevels = fields["height"].shape(1);
   } else {
-    modelLevels = fields["vert_coord"].shape(1);
+    modelLevels = modelLevelsDefault;
   }
 
   // geostrophic pressure vertical regression statistics are grouped
