--- /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/vader/MoistureControl.h.tmp	2024-04-19 12:02:23.571181972 +0200
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/vader/MoistureControl.h	2024-04-04 15:35:09.294252488 +0200
@@ -26,9 +26,7 @@
 #include "saber/blocks/SaberOuterBlockBase.h"
 
 namespace oops {
-  namespace patch{
-class Variables;
-}
+  namespace patch {class Variables;}
 }
 
 namespace saber {
