--- /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/vader/SuperMoistIncrOp.cc.tmp	2024-06-12 18:33:04.894312316 +0200
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/vader/SuperMoistIncrOp.cc	2024-05-24 08:41:35.450951363 +0200
@@ -99,23 +99,6 @@
 
 // -----------------------------------------------------------------------------
 
-void SuperMoistIncrOp::directCalibration(const oops::FieldSets & fsets) {
-  oops::Log::trace() << classname() << "::directCalibration starting" << std::endl;
-
-  // Derive air temperature and total water perturbations first
-  oops::FieldSets fsets_copy(fsets);
-  for (size_t fset_index = 0; fset_index < fsets.ens_size(); fset_index++) {
-    auto & fset = fsets_copy[fset_index];
-    exnerThetaToTemp_->multiply(fset);
-    MIO_->leftInverseMultiply(fset);
-  }
-
-  // Direct calibration of MIO block
-  MIO_->directCalibration(fsets_copy);
-
-  oops::Log::trace() << classname() << "::directCalibration done" << std::endl;
-}
-
 void SuperMoistIncrOp::print(std::ostream & os) const {
   os << classname();
 }
