--- /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/vader/SuperMoistIncrOp.h.tmp.bak	2024-03-07 14:56:01.780890727 +0100
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/vader/SuperMoistIncrOp.h	2024-03-07 12:46:13.476972885 +0100
@@ -26,9 +26,7 @@
 #include "saber/vader/MoistIncrOpParameters.h"
 
 namespace oops {
-  namespace patch{
-class Variables;
-}
+  namespace patch {class Variables;}
 }
 
 namespace saber {
