--- /home/benjaminm/code/oops-bundle/ecsaber/saber/quench/mains/CMakeLists.txt.tmp.bak	2024-08-27 12:41:01.790470778 +0200
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/quench/mains/CMakeLists.txt	2024-08-02 15:43:44.906476901 +0200
@@ -1,4 +1,5 @@
 # (C) Copyright 2017-2019 UCAR
+# (C) Copyright 2024 Meteorologisk Institutt
 #
 # This software is licensed under the terms of the Apache Licence Version 2.0
 # which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
@@ -23,3 +24,53 @@
                         LIBS    quench
                                 vader
                                 saber )
+
+ecbuild_add_executable( TARGET  saber_quench_hofx.x
+                        SOURCES quenchHofX.cc
+                        LIBS    quench )
+
+ecbuild_add_executable( TARGET  saber_quench_makeobs.x
+                        SOURCES quenchMakeObs.cc
+                        LIBS    quench )
+
+ecbuild_add_executable( TARGET  saber_quench_variational.x
+                        SOURCES quenchVariational.cc
+                        LIBS    quench
+                                vader
+                                ${gsibec_LIBRARIES}
+                                saber )
+
+ecbuild_add_executable( TARGET  saber_quench_evil_update.x
+                        SOURCES quenchEvilUpdate.cc
+                        LIBS    quench
+                                vader
+                                ${gsibec_LIBRARIES}
+                                saber )
+
+ecbuild_add_executable( TARGET  saber_quench_assim_ensemble.x
+                        SOURCES quenchAssimEnsemble.cc
+                        LIBS    quench
+                                vader
+                                ${gsibec_LIBRARIES}
+                                saber )
+
+ecbuild_add_executable( TARGET  saber_quench_ensemble_variance.x
+                        SOURCES quenchEnsembleVariance.cc
+                        LIBS    quench
+                                vader
+                                ${gsibec_LIBRARIES}
+                                saber )
+
+ecbuild_add_executable( TARGET  saber_quench_bgos.x
+                        SOURCES quenchBGOS.cc
+                        LIBS    quench
+                                vader
+                                ${gsibec_LIBRARIES}
+                                saber )
+
+ecbuild_add_executable( TARGET  saber_quench_synthetic_data.x
+                        SOURCES quenchSyntheticData.cc
+                        LIBS    quench
+                                vader
+                                ${gsibec_LIBRARIES}
+                                saber )
