--- /home/benjaminm/code/oops-bundle/ecsaber/saber/quench/mains/CMakeLists.txt.tmp.bak	2024-07-04 11:14:00.219282111 +0200
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/quench/mains/CMakeLists.txt	2024-07-04 11:12:45.487034018 +0200
@@ -30,3 +30,18 @@
                         LIBS    quench
                                 ${vader_LIBRARIES}
                                 saber )
+
+ecbuild_add_executable( TARGET  saber_quench_hofx.x
+                        SOURCES quenchHofX.cc
+                        LIBS    quench )
+
+ecbuild_add_executable( TARGET  saber_quench_makeobs.x
+                        SOURCES quenchMakeObs.cc
+                        LIBS    quench )
+
+ecbuild_add_executable( TARGET  saber_quench_variational.x
+                        SOURCES quenchVariational.cc
+                        LIBS    quench
+                                ${vader_LIBRARIES}
+                                ${gsibec_LIBRARIES}
+                                saber )
