--- /home/benjaminm/code/oops-bundle/ecsaber/saber/quench/mains/CMakeLists.txt.tmp.bak	2024-07-08 13:13:54.378607772 +0200
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/quench/mains/CMakeLists.txt	2024-07-08 13:13:33.474536046 +0200
@@ -2,7 +2,7 @@
 #
 # This software is licensed under the terms of the Apache Licence Version 2.0
 # which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
- 
+
 # VADER libraries
 if( vader_FOUND )
     set( vader_LIBRARIES "vader" )
@@ -30,3 +30,39 @@
                         LIBS    quench
                                 ${vader_LIBRARIES}
                                 saber )
+
+ecbuild_add_executable( TARGET  saber_quench_hofx.x
+                        SOURCES quenchHofX.cc
+                        LIBS    quench )
+
+ecbuild_add_executable( TARGET  saber_quench_makeobs.x
+                        SOURCES quenchMakeObs.cc
+                        LIBS    quench )
+
+ecbuild_add_executable( TARGET  saber_quench_variational.x
+                        SOURCES quenchVariational.cc
+                        LIBS    quench
+                                ${vader_LIBRARIES}
+                                ${gsibec_LIBRARIES}
+                                saber )
+
+ecbuild_add_executable( TARGET  saber_quench_evil_update.x
+                        SOURCES quenchEvilUpdate.cc
+                        LIBS    quench
+                                ${vader_LIBRARIES}
+                                ${gsibec_LIBRARIES}
+                                saber )
+
+ecbuild_add_executable( TARGET  saber_quench_assim_ensemble.x
+                        SOURCES quenchAssimEnsemble.cc
+                        LIBS    quench
+                                ${vader_LIBRARIES}
+                                ${gsibec_LIBRARIES}
+                                saber )
+
+ecbuild_add_executable( TARGET  saber_quench_ensemble_variance.x
+                        SOURCES quenchEnsembleVariance.cc
+                        LIBS    quench
+                                ${vader_LIBRARIES}
+                                ${gsibec_LIBRARIES}
+                                saber )
