--- /home/benjaminm/code/oops-bundle/ecsaber/saber/quench/mains/quenchProcessPerts.cc.tmp	2024-07-19 08:41:31.923709966 +0200
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/quench/mains/quenchProcessPerts.cc	2024-07-15 14:03:30.090175293 +0200
@@ -1,18 +1,22 @@
 /*
- * (C) Crown Copyright 2023 Met Office
+ * (C) Copyright 2024 Meteorologisk Institutt.
  *
  * This software is licensed under the terms of the Apache Licence Version 2.0
  * which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
  */
 
 #include "oops/runs/Run.h"
-#include "saber/oops/instantiateCovarFactory.h"
 #include "saber/oops/ProcessPerts.h"
+#include "src/instantiateQuenchMatrices.h"
+#include "src/Logbook.h"
 #include "src/Traits.h"
 
 int main(int argc,  char ** argv) {
   oops::Run run(argc, argv);
-  saber::instantiateCovarFactory<quench::Traits>();
-  saber::ProcessPerts<quench::Traits> processperts;
-  return run.execute(processperts);
+  quench::instantiateQuenchMatrices();
+  saber::ProcessPerts<quench::Traits> pp;
+  quench::Logbook::start();
+  run.execute(pp);
+  quench::Logbook::stop();
+  return 0;
 }
