--- /home/benjaminm/code/oops-bundle/ecsaber/oops_patch/src/oops/generic/PseudoModelState4D.h.tmp	2024-11-21 11:22:57.306426901 +0100
+++ /home/benjaminm/code/oops-bundle/ecsaber/oops_patch/src/oops/generic/PseudoModelState4D.h	2024-11-20 14:41:00.795851266 +0100
@@ -11,10 +11,10 @@
 #include <string>
 #include <vector>
 
-#include "oops/base/Geometry.h"
-#include "oops/base/State.h"
+#include "oops/interface/Geometry.h"
+#include "oops/interface/State.h"
+#include "oops/base/PostProcessor.h"
 #include "oops/base/State4D.h"
-#include "oops/generic/ModelBase.h"
 #include "oops/interface/ModelAuxControl.h"
 #include "oops/util/Duration.h"
 #include "oops/util/Logger.h"
@@ -24,7 +24,9 @@
 /// Generic implementation of the pseudo model initialized with 4D State
 /// (steps through time by stepping through states in 4D state)
 template <typename MODEL>
-class PseudoModelState4D : public ModelBase<MODEL> {
+class PseudoModelState4D : public util::Printable,
+                           private eckit::NonCopyable,
+                           private util::ObjectCounter<Model<MODEL> > {
   typedef Geometry<MODEL>          Geometry_;
   typedef ModelAuxControl<MODEL>   ModelAux_;
   typedef State<MODEL>             State_;
@@ -38,18 +40,19 @@
   PseudoModelState4D(const State4D_ & state4d,
                      const util::Duration & tstep = util::Duration(0));
 
-  /// initialize forecast
-  void initialize(State_ &) const override;
-  /// one forecast step
-  void step(State_ &, const ModelAux_ &) const override;
-  /// finalize forecast
-  void finalize(State_ &) const override;
-
-  /// model time step
-  const util::Duration & timeResolution() const override {return tstep_;}
+  /// \brief Run the forecast from state \p xx for \p len time, with \p post postprocessors
+  /// Does not need to be implemented in the models
+  void forecast(State_ & xx, const ModelAux_ &, const util::Duration & len,
+                PostProcessor<State_> & post) const;
 
  private:
-  void print(std::ostream &) const override;
+  /// Initialize the model (prior to forecast)
+  void initialize(State_ &, const ModelAux_ &) const;
+  /// Model time step
+  void step(State_ &, const ModelAux_ &) const;
+  /// Finalize the model (after the forecast)
+  void finalize(State_ &) const;
+  void print(std::ostream &) const;
 
   /// Reference to 4D state that is used in the model
   const State4D_ & state4d_;
@@ -65,15 +68,42 @@
 PseudoModelState4D<MODEL>::PseudoModelState4D(const State4D_ & state4d,
                                               const util::Duration & tstep)
   : state4d_(state4d), tstep_(tstep) {
-  const std::vector<util::DateTime> validTimes = state4d_.validTimes();
+  const std::vector<util::DateTime> validTimes = state4d_.times();
   if (validTimes.size() > 1) tstep_ = validTimes[1] - validTimes[0];
+  std::cout << tstep_ << std::endl;
   Log::trace() << "PseudoModelState4D<MODEL>::PseudoModelState4D done" << std::endl;
 }
 
 // -----------------------------------------------------------------------------
 
-template<typename MODEL>
-void PseudoModelState4D<MODEL>::initialize(State_ & xx) const {
+template <typename MODEL>
+void PseudoModelState4D<MODEL>::forecast(State_ &xx, const ModelAux_ &maux,
+                                         const util::Duration &len,
+                                         PostProcessor<State_> &post) const {
+  Log::trace() << "PseudoModelState4D<MODEL>::forecast starting" << std::endl;
+  const util::DateTime end(xx.validTime() + len);
+  Log::info() << "PseudoModelState4D::forecast: forecast starting: " << xx << std::endl;
+  this->initialize(xx, maux);
+  post.initialize(xx, end, tstep_);
+  size_t step = 0;
+  while (xx.validTime() < end) {
+    this->step(xx, maux);
+    post.process(xx);
+    ++step;
+  }
+  post.finalize(xx);
+  this->finalize(xx);
+  Log::info() << "PseudoModelState4D::forecast: forecast finished: " << xx << std::endl;
+  ASSERT(xx.validTime() == end);
+
+  Log::trace() << "PseudoModelState4D<MODEL>::forecast done" << std::endl;
+}
+
+// -----------------------------------------------------------------------------
+
+template <typename MODEL>
+void PseudoModelState4D<MODEL>::initialize(State_ &xx, const ModelAux_ &maux) const {
+  Log::trace() << "PseudoModelState4D<MODEL>::initialize starting" << std::endl;
   currentstate_ = 0;
   xx = state4d_[currentstate_];
   Log::trace() << "PseudoModelState4D<MODEL>::initialize done" << std::endl;
@@ -81,17 +111,17 @@
 
 // -----------------------------------------------------------------------------
 
-template<typename MODEL>
-void PseudoModelState4D<MODEL>::step(State_ & xx, const ModelAux_ & merr) const {
-  Log::trace() << "PseudoModelState4D<MODEL>:step Starting " << std::endl;
+template <typename MODEL>
+void PseudoModelState4D<MODEL>::step(State_ &xx, const ModelAux_ &maux) const {
+  Log::trace() << "PseudoModelState4D<MODEL>::step starting" << std::endl;
   xx = state4d_[currentstate_++];
   Log::trace() << "PseudoModelState4D<MODEL>::step done" << std::endl;
 }
 
 // -----------------------------------------------------------------------------
 
-template<typename MODEL>
-void PseudoModelState4D<MODEL>::finalize(State_ & xx) const {
+template <typename MODEL>
+void PseudoModelState4D<MODEL>::finalize(State_ &xx) const {
   Log::trace() << "PseudoModelState4D<MODEL>::finalize done" << std::endl;
 }
 
