--- /home/benjaminm/code/oops-bundle/ecsaber/oops_patch/src/oops/generic/PseudoLinearModelIncrement4D.h.tmp.bak	2025-01-18 07:00:41.485515919 +0100
+++ /home/benjaminm/code/oops-bundle/ecsaber/oops_patch/src/oops/generic/PseudoLinearModelIncrement4D.h	2025-01-18 07:01:37.775802152 +0100
@@ -10,12 +10,12 @@
 #include <string>
 #include <vector>
 
-#include "oops/base/Geometry.h"
-#include "oops/base/Increment.h"
+#include "oops/interface/Geometry.h"
+#include "oops/interface/Increment.h"
 #include "oops/base/Increment4D.h"
-#include "oops/base/State.h"
+#include "oops/interface/State.h"
 #include "oops/base/State4D.h"
-#include "oops/generic/LinearModelBase.h"
+#include "oops/interface/LinearModelBase.h"
 #include "oops/interface/ModelAuxIncrement.h"
 #include "oops/util/Duration.h"
 #include "oops/util/Logger.h"
@@ -64,7 +64,7 @@
   const util::Duration & timeResolution() const override {return tstep_;}
   const util::Duration & stepTrajectory() const override {return tstep_;}
   /// linear model variables
-  const oops::JediVariables & variables() const {return vars_;}
+  const oops::Variables & variables() const {return vars_;}
 
  private:
   void print(std::ostream &) const override;
@@ -75,7 +75,7 @@
   util::Duration   tstep_;
   /// Index of the current increment
   mutable size_t currentinc_;
-  oops::JediVariables vars_;
+  oops::Variables vars_;
 };
 
 // -----------------------------------------------------------------------------
