--- /home/benjaminm/code/ecsaber/bundle/ecsaber/oops_patch/src/oops/generic/UnstructuredInterpolator.cc.tmp.bak	2026-06-11 09:26:33.621206380 +0200
+++ /home/benjaminm/code/ecsaber/bundle/ecsaber/oops_patch/src/oops/generic/UnstructuredInterpolator.cc	2026-06-11 09:34:39.542892801 +0200
@@ -26,10 +26,10 @@
 #include "eckit/config/Configuration.h"
 #include "eckit/exception/Exceptions.h"
 
-#include "oops/base/Geometry.h"
+#include "oops/interface/Geometry.h"
 #include "oops/base/GeometryData.h"
-#include "oops/base/Increment.h"
-#include "oops/base/State.h"
+#include "oops/interface/Increment.h"
+#include "oops/interface/State.h"
 #include "oops/base/Variables.h"
 #include "oops/util/Logger.h"
 #include "oops/util/missingValues.h"
@@ -84,7 +84,6 @@
   Log::trace() << "UnstructuredInterpolator::apply starting" << std::endl;
   util::Timer timer("oops::UnstructuredInterpolator", "apply");
 
-  if (nout_ == 0) { return; }
   ASSERT(target_mask.size() == nout_);
 
   size_t nflds = 0;
