--- /home/benjaminm/code/oops-bundle/ecsaber/oops_patch/src/oops/generic/UnstructuredInterpolator.cc.tmp.bak	2026-01-14 08:16:57.003511480 +0100
+++ /home/benjaminm/code/oops-bundle/ecsaber/oops_patch/src/oops/generic/UnstructuredInterpolator.cc	2026-01-14 08:17:55.002343703 +0100
@@ -26,10 +26,10 @@
 #include "eckit/config/Configuration.h"
 #include "eckit/exception/Exceptions.h"
 
-#include "oops/base/Geometry.h"
+#include "oops/interface/Geometry.h"
 #include "oops/base/GeometryData.h"
-#include "oops/base/Increment.h"
-#include "oops/base/State.h"
+#include "oops/interface/Increment.h"
+#include "oops/interface/State.h"
 #include "oops/base/Variables.h"
 #include "oops/util/Logger.h"
 #include "oops/util/missingValues.h"
