--- /home/benjaminm/code/oops-bundle/ecsaber/oops_patch/src/oops/generic/VerticalLocEV.h.tmp	2025-01-31 15:41:05.107511073 +0100
+++ /home/benjaminm/code/oops-bundle/ecsaber/oops_patch/src/oops/generic/VerticalLocEV.h	2025-01-30 14:41:33.226452916 +0100
@@ -18,13 +18,14 @@
 
 #include "eckit/config/Configuration.h"
 
-#include "oops/base/Geometry.h"
+#include "oops/interface/Geometry.h"
 #include "oops/base/Increment4D.h"
 #include "oops/base/IncrementEnsemble.h"
 #include "oops/base/IncrementEnsemble4D.h"
 #include "oops/base/LocalIncrement.h"
 #include "oops/generic/gc99.h"
 #include "oops/interface/GeometryIterator.h"
+#include "oops/util/ECUtilities.h"
 #include "oops/util/Logger.h"
 #include "oops/util/ObjectCounter.h"
 #include "oops/util/parameters/Parameter.h"
@@ -67,7 +68,7 @@
 class VerticalLocEV: public util::Printable,
                      private util::ObjectCounter<VerticalLocEV<MODEL>> {
   typedef Geometry<MODEL>            Geometry_;
-  typedef GeometryIterator<MODEL>    GeometryIterator_;
+  typedef typename MODEL::GeometryIterator  GeometryIterator__;
   typedef Increment4D<MODEL>         Increment4D_;
   typedef IncrementEnsemble<MODEL>   IncrementEnsemble_;
   typedef IncrementEnsemble4D<MODEL> IncrementEnsemble4D_;
@@ -83,7 +84,7 @@
 
 // modulate an incrementEnsemble at a {gridPoint, timeSlice}
   Eigen::MatrixXd modulateIncrement(const IncrementEnsemble4D_ &,
-                                    const GeometryIterator_ &, size_t) const;
+                                    const GeometryIterator__ &, size_t) const;
 
 // returns number of retained eigen modes
   size_t neig() const {return neig_;}
@@ -159,11 +160,11 @@
     const Geometry_ & geom = (*sqrtVertLoc_)[0].geometry();
     for (size_t ieig=0; ieig < neig_; ++ieig) {
       std::vector<double> EvecRepl = replicateEigenVector(ieig);
-      (*sqrtVertLoc_)[ieig].ones();
-      for (GeometryIterator_ gpi = geom.begin(); gpi != geom.end(); ++gpi) {
-        oops::LocalIncrement gp = (*sqrtVertLoc_)[ieig].getLocal(gpi);
+      util::ones((*sqrtVertLoc_)[ieig]);
+      for (GeometryIterator__ gpi = geom.geometry().begin(); gpi != geom.geometry().end(); ++gpi) {
+        oops::LocalIncrement gp = (*sqrtVertLoc_)[ieig].increment().getLocal(gpi);
         gp *= EvecRepl;
-        (*sqrtVertLoc_)[ieig].setLocal(gp, gpi);
+        (*sqrtVertLoc_)[ieig].increment().setLocal(gp, gpi);
       }
     }
   }
@@ -197,7 +198,7 @@
 template<typename MODEL>
   Eigen::MatrixXd VerticalLocEV<MODEL>::computeCorrMatrix(const Geometry_ & geom) {
     std::string locUnits = options_.VertLocUnits;
-    std::vector<double> vCoord = geom.verticalCoord(locUnits);
+    std::vector<double> vCoord = geom.geometry().verticalCoord(locUnits);
     size_t nlevs = vCoord.size();
 
     // compute vertical correlations and eigen vectors
@@ -314,17 +315,17 @@
 
   for (size_t ieig=0; ieig < neig_; ++ieig) {
     // modulate an increment
-    for (size_t itime=0; itime < incr.size(); ++itime) {
+    for (size_t itime=incr.first(); itime <= incr.last(); ++itime) {
       if (EVsStoredAs3D_) {  // if EVs stored as 3D use oops operation for schur_product
         incrsOut[ieig][itime] = (*sqrtVertLoc_)[ieig];
         incrsOut[ieig][itime].schur_product_with(incr[itime]);
       } else {  // if EV is only available as a 1D column use grid eterator
         std::vector<double> EvecRepl = replicateEigenVector(ieig);
         const Geometry_ & geom = incr[itime].geometry();
-        for (GeometryIterator_ gpi = geom.begin(); gpi != geom.end(); ++gpi) {
-          oops::LocalIncrement gp = incr[itime].getLocal(gpi);
+        for (GeometryIterator__ gpi = geom.geometry().begin(); gpi != geom.geometry().end(); ++gpi) {
+          oops::LocalIncrement gp = incr[itime].increment().getLocal(gpi);
           gp *= EvecRepl;
-          incrsOut[ieig][itime].setLocal(gp, gpi);
+          incrsOut[ieig][itime].increment().setLocal(gp, gpi);
         }
       }
     }
@@ -335,14 +336,14 @@
 template<typename MODEL>
 Eigen::MatrixXd VerticalLocEV<MODEL>::modulateIncrement(
                                   const IncrementEnsemble4D_ & incrs,
-                                  const GeometryIterator_ & gi,
+                                  const GeometryIterator__ & gi,
                                   size_t itime) const {
   // modulate an increment at grid point
 
   size_t nv = 0;
   std::vector<double> EvecRepl;
   if (EVsStoredAs3D_) {
-    nv = (*sqrtVertLoc_)[0].getLocal(gi).getVals().size();
+    nv = (*sqrtVertLoc_)[0].increment().getLocal(gi).getVals().size();
   } else {
     EvecRepl = replicateEigenVector(0);
     nv = EvecRepl.size();
@@ -353,10 +354,10 @@
 
   size_t ii = 0;
   for (size_t iens=0; iens < nens; ++iens) {
-    etmp2 =  incrs[iens][itime].getLocal(gi).getVals();
+    etmp2 =  incrs[iens][itime].increment().getLocal(gi).getVals();
     for (size_t ieig=0; ieig < neig_; ++ieig) {
       if (EVsStoredAs3D_) {
-        EvecRepl = (*sqrtVertLoc_)[ieig].getLocal(gi).getVals();
+        EvecRepl = (*sqrtVertLoc_)[ieig].increment().getLocal(gi).getVals();
       } else {
         EvecRepl = replicateEigenVector(ieig);
       }
