--- /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/spectralb/SpectralCovariance.cc.tmp	2024-11-29 15:35:50.510555944 +0100
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/spectralb/SpectralCovariance.cc	2024-11-29 11:25:18.895274642 +0100
@@ -83,7 +83,7 @@
                      << " is not supported. Instead please use 'ID' central block"
                      << " and 'square root of spectral covariance' outer block."
                      << std::endl;
-  throw(eckit::FunctionalityNotSupported(
+  throw(eckit::NotImplemented(
         "use ID and square root of spectral covariance instead.", Here()));
   oops::Log::trace() << classname() << "::randomize done" << std::endl;
 }
