--- /home/benjaminm/code/oops-bundle/ecsaber/vader/vader-import.cmake.in.tmp	2024-07-19 08:40:20.747663551 +0200
+++ /home/benjaminm/code/oops-bundle/ecsaber/vader/vader-import.cmake.in	2024-07-19 08:41:24.307704999 +0200
@@ -2,10 +2,6 @@
 
 include(CMakeFindDependencyMacro)
 
-if( @jedicmake_FOUND@ AND NOT jedicmake_FOUND )
-    find_dependency(jedicmake REQUIRED)
-endif()
-
 if(NOT Boost_FOUND)
     find_dependency(Boost REQUIRED)
 endif()
@@ -26,6 +22,10 @@
     find_dependency(oops REQUIRED)
 endif()
 
+if(NOT oops-patch_FOUND)
+    find_dependency(oops-patch REQUIRED)
+endif()
+
 # Optional dependencies
 
 if(@gsw_FOUND@ AND NOT gsw_FOUND)
