--- /home/benjaminm/code/oops-bundle/ecsaber/vader/vader-import.cmake.in.tmp	2025-11-05 15:15:52.838783591 +0100
+++ /home/benjaminm/code/oops-bundle/ecsaber/vader/vader-import.cmake.in	2025-11-05 15:15:31.936701833 +0100
@@ -2,8 +2,10 @@
 
 include(CMakeFindDependencyMacro)
 
-if( @jedicmake_FOUND@ AND NOT jedicmake_FOUND )
-    find_dependency(jedicmake REQUIRED)
+if(ENABLE_QUENCH)
+    if(NOT jedicmake_FOUND)
+        find_dependency(jedicmake REQUIRED)
+    endif()
 endif()
 
 if(NOT Boost_FOUND)
