--- /home/benjaminm/code/oops-bundle/ecsaber/oops_patch/src/oops/assimilation/GETKFSolver.h.tmp.bak	2025-02-27 14:15:29.615112857 +0100
+++ /home/benjaminm/code/oops-bundle/ecsaber/oops_patch/src/oops/assimilation/GETKFSolver.h	2025-02-27 11:07:10.199179385 +0100
@@ -20,18 +20,19 @@
 #include "oops/assimilation/LocalEnsembleSolver.h"
 #include "oops/base/Departures.h"
 #include "oops/base/DeparturesEnsemble.h"
-#include "oops/base/Geometry.h"
+#include "oops/interface/Geometry.h"
 #include "oops/base/IncrementEnsemble4D.h"
 #include "oops/base/ObsEnsemble.h"
 #include "oops/base/ObsErrors.h"
 #include "oops/base/Observations.h"
-#include "oops/base/ObsSpaces.h"
+#include "oops/base/ObservationSpaces.h"
 #include "oops/base/StateEnsemble4D.h"
-#include "oops/base/StateSet.h"
+#include "oops/base/State4D.h"
 #include "oops/generic/VerticalLocEV.h"
 #include "oops/interface/GeometryIterator.h"
-#include "oops/interface/ObsDataVector.h"
+//#include "oops/interface/ObsDataVector.h"
 #include "oops/util/ConfigFunctions.h"
+#include "oops/util/ECUtilities.h"
 #include "oops/util/Logger.h"
 #include "oops/util/printRunStats.h"
 #include "oops/util/Timer.h"
@@ -47,12 +48,12 @@
  * ensemble Kalman filter. Journal of Advances in Modeling Earth Systems, 10, 
  * 3221– 3232. https://doi.org/10.1029/2018MS001468
  */
-template <typename MODEL, typename OBS>
-class GETKFSolver : public LocalEnsembleSolver<MODEL, OBS> {
-  typedef Departures<OBS>             Departures_;
-  typedef DeparturesEnsemble<OBS>     DeparturesEnsemble_;
+template <typename MODEL>
+class GETKFSolver : public LocalEnsembleSolver<MODEL> {
+  typedef Departures<MODEL>           Departures_;
+  typedef DeparturesEnsemble<MODEL>   DeparturesEnsemble_;
   typedef Geometry<MODEL>             Geometry_;
-  typedef GeometryIterator<MODEL>     GeometryIterator_;
+  typedef typename MODEL::GeometryIterator  GeometryIterator__;
   typedef Increment<MODEL>            Increment_;
   typedef Increment4D<MODEL>          Increment4D_;
   typedef IncrementEnsemble4D<MODEL>  IncrementEnsemble4D_;
@@ -60,19 +61,17 @@
   typedef Model<MODEL>                Model_;
   typedef ModelAuxControl<MODEL>      ModelAux_;
   typedef ModelAuxIncrement<MODEL>    ModelAuxInc_;
-  typedef ObsAuxControls<OBS>         ObsAux_;
-  typedef ObsAuxIncrements<OBS>       ObsAuxInc_;
-  typedef ObsDataVector<OBS, int>     ObsDataInt_;
-  typedef ObsEnsemble<OBS>            ObsEnsemble_;
-  typedef ObsErrors<OBS>              ObsErrors_;
-  typedef Observations<OBS>           Observations_;
-  typedef Observers<MODEL, OBS>       Observers_;
-  typedef ObserversTLAD<MODEL, OBS>   ObserversTLAD_;
-  typedef ObsSpaces<OBS>              ObsSpaces_;
+  typedef ObsAuxControls<MODEL>       ObsAux_;
+  typedef ObsAuxIncrements<MODEL>     ObsAuxInc_;
+//  typedef ObsDataVector<MODEL>        ObsDataInt_;
+  typedef ObsEnsemble<MODEL>          ObsEnsemble_;
+  typedef ObsErrors<MODEL>            ObsErrors_;
+  typedef Observations<MODEL>         Observations_;
+  typedef ObservationSpaces<MODEL>    ObsSpaces_;
   typedef PseudoModelState4D<MODEL>   PseudoModel_;
   typedef PseudoLinearModelIncrement4D<MODEL> PseudoLinearModel_;
   typedef State<MODEL>                State_;
-  typedef StateSet<MODEL>             StateSet_;
+  typedef State4D<MODEL>              State4D_;
   typedef StateEnsemble4D<MODEL>      StateEnsemble4D_;
   typedef VerticalLocEV<MODEL>        VerticalLocEV_;
 
@@ -82,12 +81,13 @@
   /// Constructor (allocates Wa, wa, HZb_,
   /// saves options from the config, computes VerticalLocEV_)
   GETKFSolver(ObsSpaces_ &, const Geometry_ &, const eckit::Configuration &, size_t,
-              const StateSet_ &, const JediVariables &);
+              const State4D_ &, const JediVariables &);
 
-  Observations_ computeHofX(const StateEnsemble4D_ &, size_t, bool) override;
+  Observations_ computeHofX(const StateEnsemble4D_ &, size_t, bool, const Model_ &,
+                            const Observations_ &, const ObsAux_ &);
 
-  /// entire KF update (computeWeights+applyWeights) for a grid point GeometryIterator_
-  void measurementUpdate(const IncrementEnsemble4D_ &, const GeometryIterator_ &,
+  /// entire KF update (computeWeights+applyWeights) for a grid point GeometryIterator__
+  void measurementUpdate(const IncrementEnsemble4D_ &, const GeometryIterator__ &,
                          IncrementEnsemble4D_ &) override;
 
  private:
@@ -102,7 +102,7 @@
 
   /// Applies weights and adds posterior inflation
   void applyWeights(const IncrementEnsemble4D_ &, IncrementEnsemble4D_ &,
-                    const GeometryIterator_ &);
+                    const GeometryIterator__ &);
 
  protected:
   // parameters
@@ -120,11 +120,11 @@
 
 // -----------------------------------------------------------------------------
 
-template <typename MODEL, typename OBS>
-GETKFSolver<MODEL, OBS>::GETKFSolver(ObsSpaces_ & obspaces, const Geometry_ & geometry,
+template <typename MODEL>
+GETKFSolver<MODEL>::GETKFSolver(ObsSpaces_ & obspaces, const Geometry_ & geometry,
                                 const eckit::Configuration & config, size_t nens,
-                                const StateSet_ & xbmean, const JediVariables & incvars)
-  : LocalEnsembleSolver<MODEL, OBS>(obspaces, geometry, config, nens, xbmean, incvars),
+                                const State4D_ & xbmean, const JediVariables & incvars)
+  : LocalEnsembleSolver<MODEL>(obspaces, geometry, config, nens, xbmean, incvars),
     nens_(nens), geometry_(geometry),
     vertloc_(config.getSubConfiguration("local ensemble DA.vertical localization"), xbmean[0],
     incvars), neig_(vertloc_.neig()), nanal_(neig_*nens_), HZb_(obspaces, nanal_)
@@ -135,22 +135,25 @@
 }
 
 // -----------------------------------------------------------------------------
-template <typename MODEL, typename OBS>
-Observations<OBS> GETKFSolver<MODEL, OBS>::computeHofX(const StateEnsemble4D_ & ens_xx,
-                                                       size_t iteration, bool readFromFile) {
+template <typename MODEL>
+Observations<MODEL> GETKFSolver<MODEL>::computeHofX(const StateEnsemble4D_ & ens_xx,
+                                                    size_t iteration,
+                                                    bool readFromFile, const Model_ & model,
+                                                    const Observations_ & yobs, 
+                                                    const ObsAux_ & ybias) {
   util::Timer timer(classname(), "computeHofX");
 
-  ModelAux_ moderr(geometry_, eckit::LocalConfiguration());
+  ModelAux_ moderr(geometry_, model, eckit::LocalConfiguration());
   ModelAuxInc_  moderrinc(geometry_, eckit::LocalConfiguration());
-  ObsAux_  obsaux(this->obspaces_, this->observersconf_);
-  ObsAuxInc_  obsauxinc(this->obspaces_, this->observersconf_);
-
+  ObsAux_  obsaux(this->obspaces_, this->obsconf_);
+  ObsAuxInc_  obsauxinc(this->obspaces_, this->obsconf_);
   Observations_ yb_mean(this->obspaces_);
 
   if (readFromFile) {
     // compute/read H(x) for the original ensemble members
     // also computes omb_
-    yb_mean = LocalEnsembleSolver<MODEL, OBS>::computeHofX(ens_xx, iteration, readFromFile);
+    yb_mean = LocalEnsembleSolver<MODEL>::computeHofX(ens_xx, iteration, readFromFile, model,
+                                                      yobs, ybias);
 
     // read modulated ensemble
     Observations_ ytmp(yb_mean);
@@ -159,20 +162,24 @@
       Log::info() << " GETKFSolver::computeHofX starting ensemble member " << iens+1 << std::endl;
       util::printRunStats("GETKFSolver read hofx");
       for (size_t ieig = 0; ieig < neig_; ++ieig) {
-        ytmp.read("hofxm"+std::to_string(iteration)+"_"+std::to_string(ieig+1)+
-                      "_"+std::to_string(iens+1));
-        HZb_[ii] = ytmp - yb_mean;
+        ytmp.read(util::setObsValue(this->obsconf_, "hofxm"+std::to_string(iteration)+"_"+
+                      std::to_string(ieig+1)+"_"+std::to_string(iens+1)));
+        for (size_t jj = 0; jj < HZb_[ii].size(); ++jj) {
+          HZb_[ii][jj] = ytmp[jj];
+          HZb_[ii][jj] -= yb_mean[jj];
+        }
         ii = ii + 1;
       }
     }
   } else {
     const util::Duration default_tstep = (this->obspaces_.windowEnd()
                                         - this->obspaces_.windowStart()) * 2;
-    const std::vector<util::DateTime> times = ens_xx[0].validTimes();
+    const std::vector<util::DateTime> times = ens_xx[0].times();
     const util::Duration flength = times[times.size()-1] - times[0];
 
     // Setup PseudoLinearModelIncrement4D to run on ensemble perturbation
-    Increment4D_ dx(geometry_, this->incvars_, times);
+    const oops::Variables<MODEL> varsT(templatedVarsConf(this->incvars_));
+    Increment4D_ dx(geometry_, varsT, times);
 
     // modulate ensemble of obs
     IncrementEnsemble4D_ Ztmp(geometry_, this->incvars_, times, neig_);
@@ -184,6 +191,7 @@
     size_t ii = 0;
 
     if (this->useLinearObserver()) {
+/*
       this->R_.reset(new ObsErrors_(this->observersconf_, this->obspaces_));
 
       // Setup pseudo model to run on ensemble mean
@@ -236,15 +244,15 @@
 
       // mask H(x) ensemble perturbations
       for (size_t iens = 0; iens < ens_xx.size(); ++iens) {
-        this->invVarR_->mask(this->Yb_[iens]);
-        this->Yb_[iens].mask(*this->invVarR_);
+        util::mask(*this->invVarR_, this->Yb_[iens]);
+        util::mask(this->Yb_[iens], (this->invVarR_));
       }
 
       // calculate obs departures and mask with qc flag
       Observations_ yobs(this->obspaces_, "ObsValue");
       this->omb_ = yobs - yb_mean;
-      this->invVarR_->mask(this->omb_);
-      this->omb_.mask(*this->invVarR_);
+      util::mask(*this->invVarR_, this->omb_);
+      util::mask(this->omb_, *this->invVarR_);
 
       // add linearized H(x) to the linear model postprocessor
       linear_hofx.initializeTL(posttrajtl);
@@ -285,10 +293,12 @@
           ii = ii + 1;
         }
       }
+*/
     } else {
       // compute/read H(x) for the original ensemble members
       // also computes omb_
-      yb_mean = LocalEnsembleSolver<MODEL, OBS>::computeHofX(ens_xx, iteration, readFromFile);
+      yb_mean = LocalEnsembleSolver<MODEL>::computeHofX(ens_xx, iteration, readFromFile, model,
+                                                        yobs, ybias);
 
       for (size_t iens = 0; iens < nens_; ++iens) {
         Log::info() << " GETKFSolver::computeHofX starting ensemble member " << iens+1 << std::endl;
@@ -296,13 +306,18 @@
         dx.diff(ens_xx[iens], this->xbmean_);
         vertloc_.modulateIncrement(dx, Ztmp);
         for (size_t ieig = 0; ieig < neig_; ++ieig) {
-          StateSet_ tmpState = this->xbmean_;
-          tmpState += Ztmp[ieig];
+          State4D_ tmpState = this->xbmean_;
+          for (size_t jj = 0; jj < tmpState.statesNumber(); ++jj) {
+            tmpState[jj] += Ztmp[ieig][jj];
+          }
           Observations_ tmpObs(this->obspaces_);
-          this->computeHofX4DNonLinear(config, tmpState, tmpObs);
-          HZb_[ii] = tmpObs - yb_mean;
-          tmpObs.save("hofxm"+std::to_string(iteration)+"_"+std::to_string(ieig+1)+
-                        "_"+std::to_string(iens+1));
+          this->computeHofX4DNonLinear(config, model, tmpState, ybias, tmpObs);
+          for (size_t jj = 0; jj < HZb_[ii].size(); ++jj) {
+            HZb_[ii][jj] = tmpObs[jj];
+            HZb_[ii][jj] -= yb_mean[jj];
+          }
+          tmpObs.save(util::setObsValue(this->obsconf_, "hofxm"+std::to_string(iteration)+"_"
+                        +std::to_string(ieig+1)+"_"+std::to_string(iens+1)));
           ii = ii + 1;
         }
       }
@@ -313,11 +328,11 @@
 
 // -----------------------------------------------------------------------------
 
-template <typename MODEL, typename OBS>
-void GETKFSolver<MODEL, OBS>::computeWeights(const Eigen::VectorXd & dy,
-                                             const Eigen::MatrixXd & Yb,
-                                             const Eigen::MatrixXd & YbOrig,
-                                             const Eigen::VectorXd & R_invvar) {
+template <typename MODEL>
+void GETKFSolver<MODEL>::computeWeights(const Eigen::VectorXd & dy,
+                                        const Eigen::MatrixXd & Yb,
+                                        const Eigen::MatrixXd & YbOrig,
+                                        const Eigen::VectorXd & R_invvar) {
   // compute transformation matrix, save in Wa_, wa_
   // Yb(nobs,neig*nens), YbOrig(nobs,nens)
   // uses GSI GETKF code
@@ -350,10 +365,10 @@
 
 // -----------------------------------------------------------------------------
 
-template <typename MODEL, typename OBS>
-void GETKFSolver<MODEL, OBS>::applyWeights(const IncrementEnsemble4D_ & bkg_pert,
-                                           IncrementEnsemble4D_ & ana_pert,
-                                           const GeometryIterator_ & i) {
+template <typename MODEL>
+void GETKFSolver<MODEL>::applyWeights(const IncrementEnsemble4D_ & bkg_pert,
+                                      IncrementEnsemble4D_ & ana_pert,
+                                      const GeometryIterator__ & i) {
   // apply Wa_, wa_
   util::Timer timer(classname(), "applyWeights");
 
@@ -362,7 +377,7 @@
   Eigen::MatrixXd XbOriginal;   // original perturbations
 
   // loop through analysis times and ens. members
-  for (unsigned itime=0; itime < bkg_pert[0].size(); ++itime) {
+  for (unsigned itime=bkg_pert[0].first(); itime <= bkg_pert[0].last(); ++itime) {
     // cast bkg_pert ensemble at grid point i as an Eigen matrix Xb
     // modulates Xb
     XbModulated = vertloc_.modulateIncrement(bkg_pert, i, itime);
@@ -387,21 +402,21 @@
 
 // -----------------------------------------------------------------------------
 
-template <typename MODEL, typename OBS>
-void GETKFSolver<MODEL, OBS>::measurementUpdate(const IncrementEnsemble4D_ & bkg_pert,
-                                                const GeometryIterator_ & i,
-                                                IncrementEnsemble4D_ & ana_pert) {
+template <typename MODEL>
+void GETKFSolver<MODEL>::measurementUpdate(const IncrementEnsemble4D_ & bkg_pert,
+                                           const GeometryIterator__ & i,
+                                           IncrementEnsemble4D_ & ana_pert) {
   util::Timer timer(classname(), "measurementUpdate");
 
   // create the local subset of observations
   Departures_ locvector(this->obspaces_);
-  locvector.ones();
+  util::ones(locvector);
   this->obsloc().computeLocalization(i, locvector);
   for (size_t iens = 0; iens < nanal_; ++iens) {
-     (this->invVarR_)->mask(this->HZb_[iens]);
+    util::mask(*this->invVarR_, this->HZb_[iens]);
   }
-  locvector.mask(*(this->invVarR_));
-  const Eigen::VectorXd local_omb_vec = this->omb_.packEigen(locvector);
+  util::mask(locvector, (*this->invVarR_));
+  Eigen::VectorXd local_omb_vec = util::packEigen(this->omb_, locvector);
 
   if (local_omb_vec.size() == 0) {
     // no obs. so no need to update Wa_ and wa_
@@ -413,9 +428,9 @@
     const Eigen::MatrixXd local_Yb_mat = this->Yb_.packEigen(locvector);
     const Eigen::MatrixXd local_HZ_mat = this->HZb_.packEigen(locvector);
     // create local obs errors and apply localization
-    const Eigen::VectorXd localization = locvector.packEigen(locvector);
-    const Eigen::VectorXd local_invVarR_vec = this->invVarR_->packEigen(locvector).array()
-                                              * localization.array();
+    const Eigen::VectorXd localization = util::packEigen(locvector, locvector);
+    Eigen::VectorXd local_invVarR_vec = util::packEigen(*this->invVarR_, locvector);
+    local_invVarR_vec.array() *= localization.array();
     computeWeights(local_omb_vec, local_HZ_mat, local_Yb_mat, local_invVarR_vec);
     applyWeights(bkg_pert, ana_pert, i);
   }
