--- /home/benjaminm/code/oops-bundle/ecsaber/oops_patch/src/oops/assimilation/instantiateLocalEnsembleSolverFactory.h.tmp.bak	2025-01-31 15:41:04.540507487 +0100
+++ /home/benjaminm/code/oops-bundle/ecsaber/oops_patch/src/oops/assimilation/instantiateLocalEnsembleSolverFactory.h	2025-01-30 08:03:17.421747809 +0100
@@ -10,15 +10,15 @@
 
 #include "oops/assimilation/GETKFSolver.h"
 #include "oops/assimilation/LETKFSolver.h"
-#include "oops/assimilation/LETKFSolverGSI.h"
+//#include "oops/assimilation/LETKFSolverGSI.h"
 #include "oops/assimilation/LocalEnsembleSolver.h"
 
 namespace oops {
 
-template <typename MODEL, typename OBS> void instantiateLocalEnsembleSolverFactory() {
-  static LocalEnsembleSolverMaker<MODEL, OBS, LETKFSolver<MODEL, OBS> > makerLETKF_("LETKF");
-  static LocalEnsembleSolverMaker<MODEL, OBS, LETKFSolverGSI<MODEL, OBS> > makerGSI_("GSI LETKF");
-  static LocalEnsembleSolverMaker<MODEL, OBS, GETKFSolver<MODEL, OBS> > makerGETKF_("GETKF");
+template <typename MODEL> void instantiateLocalEnsembleSolverFactory() {
+  static LocalEnsembleSolverMaker<MODEL, LETKFSolver<MODEL> > makerLETKF_("LETKF");
+//  static LocalEnsembleSolverMaker<MODEL, LETKFSolverGSI<MODEL> > makerGSI_("GSI LETKF");
+  static LocalEnsembleSolverMaker<MODEL, GETKFSolver<MODEL> > makerGETKF_("GETKF");
 }
 
 }  // namespace oops
