--- /home/benjaminm/code/oops-bundle/ecsaber/oops_patch/src/oops/mpi/mpi.h.tmp	2024-11-06 13:06:33.708619003 +0100
+++ /home/benjaminm/code/oops-bundle/ecsaber/oops_patch/src/oops/mpi/mpi.h	2024-11-06 13:00:07.442920503 +0100
@@ -69,22 +69,6 @@
 // ------------------------------------------------------------------------------------------------
 
 template <typename SERIALIZABLE>
-void sendReceiveReplace(const eckit::mpi::Comm & comm, SERIALIZABLE & sendrecvobj,
-                        const int dest, const int sendtag, const int source, const int recvtag) {
-  util::Timer timer("oops::mpi", "sendReceiveReplace");
-  size_t sz = sendrecvobj.serialSize();
-  std::vector<double> sendrecvbuf;
-  sendrecvobj.serialize(sendrecvbuf);
-  eckit::mpi::Status status = comm.sendReceiveReplace(sendrecvbuf.data(), sz,
-                                                      dest, sendtag, source, recvtag);
-  size_t ii = 0;
-  sendrecvobj.deserialize(sendrecvbuf, ii);
-  ASSERT(ii == sz);
-}
-
-// ------------------------------------------------------------------------------------------------
-
-template <typename SERIALIZABLE>
 void broadcast(const eckit::mpi::Comm & comm, SERIALIZABLE & obj, const size_t root) {
   util::Timer timer("oops::mpi", "broadcast");
   size_t sz = obj.serialSize();
@@ -101,22 +85,6 @@
 }
 
 // ------------------------------------------------------------------------------------------------
-
-template <typename SERIALIZABLE>
-void reduceInPlace(const eckit::mpi::Comm & comm, SERIALIZABLE & obj, const size_t root) {
-  util::Timer timer("oops::mpi", "reduceInPlace");
-  size_t sz = obj.serialSize();
-  std::vector<double> buf;
-  obj.serialize(buf);
-  comm.reduceInPlace(buf.data(), sz, eckit::mpi::sum(), root);
-  size_t ii = 0;
-  if (comm.rank() == root) {
-    obj.deserialize(buf, ii);
-    ASSERT(ii == sz);
-  }
-}
-
-// ------------------------------------------------------------------------------------------------
 
 template <typename SERIALIZABLE>
 void allReduceInPlace(const eckit::mpi::Comm & comm, SERIALIZABLE & obj) {
