--- /home/benjaminm/code/oops-bundle/ecsaber/oops-patch/src/oops/util/ConfigFunctions.cc.tmp	2024-07-08 13:40:29.363495112 +0200
+++ /home/benjaminm/code/oops-bundle/ecsaber/oops-patch/src/oops/util/ConfigFunctions.cc	2024-07-08 13:39:59.379397059 +0200
@@ -158,9 +158,6 @@
     } else if (isVector(config)) {
        // seekAndReplace should not be called on a vector of configurations
        ABORT("This is a vector, shouldn't be here...");
-    } else {
-       // seekAndReplace should not be called on a final pair
-       ABORT("This is a final pair, shouldn't be here...");
     }
   }
 
