--- /home/benjaminm/code/oops-bundle/ecsaber/oops-patch/src/oops/util/FieldSetHelpers.cc.tmp	2024-03-07 14:55:05.920841830 +0100
+++ /home/benjaminm/code/oops-bundle/ecsaber/oops-patch/src/oops/util/FieldSetHelpers.cc	2024-03-07 14:49:58.412577780 +0100
@@ -17,6 +17,7 @@
 #include "atlas/field.h"
 #include "atlas/util/function/VortexRollup.h"
 
+#include "eckit/exception/Exceptions.h"
 #include "eckit/utils/Hash.h"
 
 #include "oops/util/abor1_cpp.h"
@@ -1538,8 +1539,6 @@
   readFieldSet(comm, fspace, variableSizes, vars.variables(), config, fset);
 }
 
-
-
 // -----------------------------------------------------------------------------
 
 }  // namespace util
