--- /home/benjaminm/code/oops-bundle/ecsaber/oops-patch/src/oops/util/FieldSetHelpers.h.tmp	2024-03-07 14:55:05.932841841 +0100
+++ /home/benjaminm/code/oops-bundle/ecsaber/oops-patch/src/oops/util/FieldSetHelpers.h	2024-03-07 13:26:21.319642905 +0100
@@ -99,7 +99,6 @@
                   const eckit::Configuration &,
                   atlas::FieldSet &);
 
-
 // -----------------------------------------------------------------------------
 
 }  // namespace util
