--- /home/benjaminm/code/oops-bundle/ecsaber/oops-patch/src/oops/util/FunctionSpaceHelpers.h.tmp.bak	2024-04-04 15:06:30.559973193 +0200
+++ /home/benjaminm/code/oops-bundle/ecsaber/oops-patch/src/oops/util/FunctionSpaceHelpers.h	2024-04-04 15:12:04.180325409 +0200
@@ -70,4 +70,13 @@
 
 // -----------------------------------------------------------------------------
 
+// Define a distribution and a mesh from a custom partition
+void setupStructuredMeshWithCustomPartition(const eckit::mpi::Comm &,
+    const atlas::Grid &,
+    const std::vector<int> &,
+    atlas::grid::Distribution &,
+    atlas::Mesh &);
+
+// -----------------------------------------------------------------------------
+
 }  // namespace util
