--- /home/benjaminm/code/oops-bundle/ecsaber/oops_patch/src/oops/util/FieldSetOperations.cc.tmp	2024-11-27 12:20:21.321075774 +0100
+++ /home/benjaminm/code/oops-bundle/ecsaber/oops_patch/src/oops/util/FieldSetOperations.cc	2024-11-26 15:07:09.491531984 +0100
@@ -177,57 +177,68 @@
     // Add contributions
     auto view1 = atlas::array::make_view<double, 2>(field1);
     auto view2 = atlas::array::make_view<double, 2>(field2);
-    if (field1.functionspace().type() == "Spectral") {
-      const atlas::functionspace::Spectral fs(field1.functionspace());
-      const atlas::idx_t N = fs.truncation();
-      const auto zonal_wavenumbers = fs.zonal_wavenumbers();
-      const atlas::idx_t nb_zonal_wavenumbers = zonal_wavenumbers.size();
-      int jnode = 0;
-      for (int jm=0; jm < nb_zonal_wavenumbers; ++jm) {
-        const atlas::idx_t m1 = zonal_wavenumbers(jm);
-        for (std::size_t n1 = m1; n1 <= static_cast<std::size_t>(N); ++n1) {
-          if (m1 == 0) {
-            // Real part only
-            for (atlas::idx_t jlevel = 0; jlevel < field1.shape(1); ++jlevel) {
-              if (view1(jnode, jlevel) != util::missingValue<double>()
-                && view2(jnode, jlevel) != util::missingValue<double>()) {
-                dp += view1(jnode, jlevel)*view2(jnode, jlevel);
+    if (field1.functionspace()) {
+      if (field1.functionspace().type() == "Spectral") {
+        const atlas::functionspace::Spectral fs(field1.functionspace());
+        const atlas::idx_t N = fs.truncation();
+        const auto zonal_wavenumbers = fs.zonal_wavenumbers();
+        const atlas::idx_t nb_zonal_wavenumbers = zonal_wavenumbers.size();
+        int jnode = 0;
+        for (int jm=0; jm < nb_zonal_wavenumbers; ++jm) {
+          const atlas::idx_t m1 = zonal_wavenumbers(jm);
+          for (std::size_t n1 = m1; n1 <= static_cast<std::size_t>(N); ++n1) {
+            if (m1 == 0) {
+              // Real part only
+              for (atlas::idx_t jlevel = 0; jlevel < field1.shape(1); ++jlevel) {
+                if (view1(jnode, jlevel) != util::missingValue<double>()
+                  && view2(jnode, jlevel) != util::missingValue<double>()) {
+                  dp += view1(jnode, jlevel)*view2(jnode, jlevel);
+                }
               }
-            }
-            ++jnode;
+              ++jnode;
 
-            // No imaginary part
-            ++jnode;
-          } else {
-            // Real part
-            for (atlas::idx_t jlevel = 0; jlevel < field1.shape(1); ++jlevel) {
-              if (view1(jnode, jlevel) != util::missingValue<double>()
-                && view2(jnode, jlevel) != util::missingValue<double>()) {
-                dp += 2.0*view1(jnode, jlevel)*view2(jnode, jlevel);
+              // No imaginary part
+              ++jnode;
+            } else {
+              // Real part
+              for (atlas::idx_t jlevel = 0; jlevel < field1.shape(1); ++jlevel) {
+                if (view1(jnode, jlevel) != util::missingValue<double>()
+                  && view2(jnode, jlevel) != util::missingValue<double>()) {
+                  dp += 2.0*view1(jnode, jlevel)*view2(jnode, jlevel);
+                }
               }
-            }
-            ++jnode;
+              ++jnode;
 
-            // Imaginary part
+              // Imaginary part
+              for (atlas::idx_t jlevel = 0; jlevel < field1.shape(1); ++jlevel) {
+                if (view1(jnode, jlevel) != util::missingValue<double>()
+                  && view2(jnode, jlevel) != util::missingValue<double>()) {
+                  dp += 2.0*view1(jnode, jlevel)*view2(jnode, jlevel);
+                }
+              }
+              ++jnode;
+            }
+          }
+        }
+      } else {
+        const auto ghostView = atlas::array::make_view<int, 1>(field1.functionspace().ghost());
+        for (atlas::idx_t jnode = 0; jnode < field1.shape(0); ++jnode) {
+          if (ghostView(jnode) == 0) {
             for (atlas::idx_t jlevel = 0; jlevel < field1.shape(1); ++jlevel) {
               if (view1(jnode, jlevel) != util::missingValue<double>()
                 && view2(jnode, jlevel) != util::missingValue<double>()) {
-                dp += 2.0*view1(jnode, jlevel)*view2(jnode, jlevel);
+                dp += view1(jnode, jlevel)*view2(jnode, jlevel);
               }
             }
-            ++jnode;
           }
         }
       }
     } else {
-      const auto ghostView = atlas::array::make_view<int, 1>(field1.functionspace().ghost());
       for (atlas::idx_t jnode = 0; jnode < field1.shape(0); ++jnode) {
-        if (ghostView(jnode) == 0) {
-          for (atlas::idx_t jlevel = 0; jlevel < field1.shape(1); ++jlevel) {
-            if (view1(jnode, jlevel) != util::missingValue<double>()
-              && view2(jnode, jlevel) != util::missingValue<double>()) {
-              dp += view1(jnode, jlevel)*view2(jnode, jlevel);
-            }
+        for (atlas::idx_t jlevel = 0; jlevel < field1.shape(1); ++jlevel) {
+          if (view1(jnode, jlevel) != util::missingValue<double>()
+            && view2(jnode, jlevel) != util::missingValue<double>()) {
+            dp += view1(jnode, jlevel)*view2(jnode, jlevel);
           }
         }
       }
