--- /home/benjaminm/code/oops-bundle/ecsaber/oops_patch/src/oops/util/FieldSetHelpers.cc.tmp	2024-11-29 15:34:24.736190241 +0100
+++ /home/benjaminm/code/oops-bundle/ecsaber/oops_patch/src/oops/util/FieldSetHelpers.cc	2024-11-29 15:35:41.526517595 +0100
@@ -361,9 +361,17 @@
   oops::Log::trace() << "copyFieldSet starting" << std::endl;
   fset.clear();
   for (const auto & otherField : otherFset) {
-    // Create Field
-    atlas::Field field = otherField.functionspace().createField<double>(
-      atlas::option::name(otherField.name()) | atlas::option::levels(otherField.shape(1)));
+    // Check whether the input Field is associated with a FunctionSpace
+    atlas::Field field;
+    if (otherField.functionspace()) {
+      // Create Field from FunctionSpace
+      field = otherField.functionspace().createField<double>(
+        atlas::option::name(otherField.name()) | atlas::option::levels(otherField.shape(1)));
+    } else {
+      // Create Field without FunctionSpace
+      field = atlas::Field(otherField.name(), atlas::array::make_datatype<double>(),
+        atlas::array::make_shape(otherField.shape(0), otherField.shape(1)));
+    }
 
     // Copy data
     if (field.rank() == 2) {
@@ -384,6 +392,9 @@
     // Add field
     fset.add(field);
   }
+
+  // Copy fieldset name
+  fset.name() = otherFset.name();
 }
 
 // -----------------------------------------------------------------------------
