--- /home/benjaminm/code/oops-bundle/ecsaber/oops_patch/src/oops/util/FieldSetOperations.cc.tmp	2025-01-31 15:41:05.324512445 +0100
+++ /home/benjaminm/code/oops-bundle/ecsaber/oops_patch/src/oops/util/FieldSetOperations.cc	2025-01-28 13:35:16.822180733 +0100
@@ -10,7 +10,6 @@
 #include <cmath>
 
 #include "atlas/array.h"
-#include "atlas/field.h"
 #include "atlas/functionspace.h"
 
 #include "eckit/exception/Exceptions.h"
@@ -33,7 +32,7 @@
       auto view = atlas::array::make_view<double, 2>(field);
       view.assign(0.0);
     } else {
-      ABORT("zeroFieldSet: wrong rank");
+      throw eckit::Exception("zeroFieldSet: wrong rank", Here());
     }
   }
 
@@ -64,7 +63,7 @@
         }
       }
     } else {
-      ABORT("addFieldSets: wrong rank");
+      throw eckit::Exception("addFieldSets: wrong rank", Here());
     }
 
     // If either term in the sum is out-of-date, then the result will be out-of-date
@@ -96,7 +95,7 @@
         }
       }
     } else {
-      ABORT("subFieldSets: wrong rank");
+      throw eckit::Exception("subFieldSets: wrong rank", Here());
     }
 
     // If either term in the subtraction is out-of-date, then the result will be out-of-date
@@ -123,7 +122,7 @@
         }
       }
     } else {
-      ABORT("multiplyFieldSet: wrong rank");
+      throw eckit::Exception("multiplyFieldSet: wrong rank", Here());
     }
   }
   oops::Log::trace() << "multiplyFieldSet done" << std::endl;
@@ -151,7 +150,7 @@
         }
       }
     } else {
-      ABORT("multiplyFieldSets: wrong rank");
+      throw eckit::Exception("multiplyFieldSets: wrong rank", Here());
     }
 
     // If either term in the product is out-of-date, then the result will be out-of-date
@@ -166,9 +165,13 @@
 double dotProductFieldsLocal(const atlas::Field & field1,
                              const atlas::Field & field2) {
   oops::Log::trace() << "dotProductFieldsLocal starting" << std::endl;
+
+  // Check fields names
   ASSERT(field1.name() == field2.name());
-  // Compute dot product
+
+  // Compute local dot product
   double dp = 0.0;
+
   if (field1.rank() == 2 && field2.rank() == 2) {
     // Check fields consistency
     ASSERT(field1.shape(0) == field2.shape(0));
@@ -233,11 +236,11 @@
       }
     }
   } else {
-    ABORT("dotProductFieldsLocal: wrong rank");
+    throw eckit::Exception("dotProductFieldsLocal: wrong rank", Here());
   }
 
+  // Return local dot product
   oops::Log::trace() << "dotProductFieldsLocal done" << std::endl;
-  // Return dot product
   return dp;
 }
 
@@ -246,9 +249,13 @@
 double dotProductFields(const atlas::Field & field1,
                         const atlas::Field & field2,
                         const eckit::mpi::Comm & comm) {
+  // Compute local dot product
   double dp = dotProductFieldsLocal(field1, field2);
+
   // Allreduce
   comm.allReduceInPlace(dp, eckit::mpi::sum());
+
+  // Return dot product
   return dp;
 }
 
@@ -262,17 +269,19 @@
 
   // Compute dot product
   double dp = 0.0;
+
   for (const auto & var : vars) {
     // Check fields presence
     if (fset1.has(var) && fset2.has(var)) {
       dp += dotProductFieldsLocal(fset1.field(var), fset2.field(var));
     }
   }
+
   // Allreduce
   comm.allReduceInPlace(dp, eckit::mpi::sum());
 
-  oops::Log::trace() << "dotProductFieldSets done" << std::endl;
   // Return dot product
+  oops::Log::trace() << "dotProductFieldSets done" << std::endl;
   return dp;
 }
 
@@ -315,12 +324,12 @@
             // If the numerator is 0, then it's ok for the denominator to be 0; this is probably
             // a case of 0/0 in the halo, and we opt to return 0 (i.e., no change to the field).
             // If the numerator is finite (= this else branch), this is a divide-by-zero error:
-            ABORT("divideFieldSets: divide by zero");
+            throw eckit::Exception("divideFieldSets: divide by zero", Here());
           }
         }
       }
     } else {
-      ABORT("divideFieldSets: wrong rank");
+      throw eckit::Exception("divideFieldSets: wrong rank", Here());
     }
 
     // If either term in the division is out-of-date, then the result will be out-of-date
@@ -357,13 +366,13 @@
             if (std::abs(divView(jnode, jlevel)) > 0.0) {
               view(jnode, jlevel) /= divView(jnode, jlevel);
             } else {
-                ABORT("divideFieldSets with mask: divide by zero");
+                throw eckit::Exception("divideFieldSets with mask: divide by zero", Here());
             }
           }
         }
       }
     } else {
-      ABORT("divideFieldSets with mask: wrong rank");
+      throw eckit::Exception("divideFieldSets with mask: wrong rank", Here());
     }
 
     // If either term in the division is out-of-date, then the result will be out-of-date
@@ -390,7 +399,7 @@
         }
       }
     } else {
-      ABORT("sqrtFieldSet: wrong rank");
+      throw eckit::Exception("sqrtFieldSet: wrong rank", Here());
     }
   }
 
