--- /home/benjaminm/code/oops-bundle/ecsaber/oops_patch/src/oops/util/printRunStats.cc.tmp	2025-01-28 11:16:56.113118842 +0100
+++ /home/benjaminm/code/oops-bundle/ecsaber/oops_patch/src/oops/util/printRunStats.cc	2025-01-28 09:24:52.824043420 +0100
@@ -15,7 +15,7 @@
 
 #include "oops/mpi/mpi.h"
 #include "oops/util/Logger.h"
-#include "oops/util/Timer.h"
+#include "oops/util/ECUtilities.h"
 
 namespace util {
 
