--- /home/benjaminm/code/ecsaber/bundle/ecsaber/saber/test/CMakeLists.txt.tmp.bak	2026-03-02 15:52:59.556926813 +0100
+++ /home/benjaminm/code/ecsaber/bundle/ecsaber/saber/test/CMakeLists.txt	2026-03-02 15:54:53.198026248 +0100
@@ -200,9 +200,10 @@
       file( MAKE_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/testdata/${test} )
 
       # Link to yaml file
-      execute_process( COMMAND ${CMAKE_COMMAND} -E create_symlink
-                       ${CMAKE_CURRENT_SOURCE_DIR}/testinput/${test}.yaml
-                       ${CMAKE_CURRENT_BINARY_DIR}/testinput/${test}.yaml )
+      execute_process(COMMAND ${Python3_EXECUTABLE}
+                      ${CMAKE_BINARY_DIR}/bin/ecsaber_yaml_update.py
+                      ${CMAKE_CURRENT_SOURCE_DIR}/testinput/${test}.yaml
+                      ${CMAKE_CURRENT_BINARY_DIR}/testinput/${test}.yaml )
 
       # Link to reference file
       if( EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/testref/${test}.ref )
@@ -313,10 +314,11 @@
 
                       # Add test
                       ecbuild_add_test( TARGET saber_test_${test}_${mpi}-${omp}
-                                        MPI ${mpi}
-                                        OMP ${omp}
-                                        COMMAND ${CMAKE_BINARY_DIR}/bin/saber_quench_${exename}.x
-                                        ARGS testinput/${test}.yaml
+                                        COMMAND ${CMAKE_BINARY_DIR}/bin/ecsaber_compare.py
+                                        ARGS ${mpi}
+                                             ${omp}
+                                             ${CMAKE_BINARY_DIR}/bin/saber_quench_${exename}.x
+                                             testinput/${test}.yaml
                                         DEPENDS saber_quench_${exename}.x
                                         TEST_DEPENDS ${deps_list} )
                   endif()
@@ -341,10 +343,11 @@
 
                   # Add test
                   ecbuild_add_test( TARGET saber_test_${test}_${mpi}-${omp}
-                                    MPI ${mpi}
-                                    OMP ${omp}
-                                    COMMAND ${CMAKE_BINARY_DIR}/bin/saber_quench_error_covariance_toolbox.x
-                                    ARGS testinput/${yaml_prefix}${test}.yaml
+                                    COMMAND ${CMAKE_BINARY_DIR}/bin/ecsaber_compare.py
+                                    ARGS ${mpi}
+                                         ${omp}
+                                         ${CMAKE_BINARY_DIR}/bin/saber_quench_error_covariance_toolbox.x
+                                         testinput/${yaml_prefix}${test}.yaml
                                     DEPENDS saber_quench_error_covariance_toolbox.x
                                     TEST_DEPENDS ${deps_list} )
               endif()
@@ -381,12 +384,3 @@
                 SOURCES         fctest/fctest_nicas_sqrt.F90
                 LIBS            saber )
 endif()
-
-# A bespoke test for parallel B
-set( mpi 5 )
-set( omp 1 )
-ecbuild_add_test( TARGET saber_test_dirac_parallel_hybrid_ratio_stddev_5-1
-                  MPI ${mpi}
-                  OMP ${omp}
-                  COMMAND ${CMAKE_BINARY_DIR}/bin/saber_quench_error_covariance_toolbox.x
-                  ARGS testinput/dirac_parallel_hybrid_ratio_stddev.yaml)
