--- /home/benjaminm/code/oops-bundle/ecsaber/saber/test/CMakeLists.txt.tmp	2024-07-04 13:56:22.276091445 +0200
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/test/CMakeLists.txt	2024-07-04 13:48:12.738714286 +0200
@@ -7,9 +7,6 @@
 # Tests activated
 message( STATUS "SABER tests activated:" )
 
-# Use find_branch function in jedi-cmake
-include( git_functions )
-
 # Default SABER_TEST_TIER
 set( SABER_TEST_TIER 1 )
 
@@ -148,9 +145,10 @@
     file( MAKE_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/testdata/${test} )
 
     # Link to yaml file
-    execute_process( COMMAND ${CMAKE_COMMAND} -E create_symlink
-                     ${CMAKE_CURRENT_SOURCE_DIR}/testinput/${test}.yaml
-                     ${CMAKE_CURRENT_BINARY_DIR}/testinput/${test}.yaml )
+    execute_process(COMMAND ${Python3_EXECUTABLE}
+                    ${CMAKE_CURRENT_SOURCE_DIR}/ecmwf_yaml_update.py
+                    ${CMAKE_CURRENT_SOURCE_DIR}/testinput/${test}.yaml
+                    ${CMAKE_CURRENT_BINARY_DIR}/testinput/${test}.yaml )
 
     # Link to reference file
     if( EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/testref/${test}.ref )
@@ -217,8 +215,8 @@
                 endif()
 
                 # Special check for 4D tests
-                string( FIND ${test} "_4d" fourd_result )
-                if( fourd_result EQUAL -1 OR ( fourd_result GREATER_EQUAL 0 AND ${mpi} GREATER 1 ) )
+                string( FIND ${test} "_4d" found_result )
+                if( found_result EQUAL -1 )
                     set( 4dCheck true CACHE BOOL "" FORCE )
                 else()
                     set( 4dCheck false CACHE BOOL "" FORCE )
