--- /home/benjaminm/code/ecsaber/bundle/ecsaber/saber/test/CMakeLists.txt.tmp.bak	2026-04-01 16:29:45.966306318 +0200
+++ /home/benjaminm/code/ecsaber/bundle/ecsaber/saber/test/CMakeLists.txt	2026-04-01 16:31:52.966777510 +0200
@@ -236,9 +236,10 @@
       file( MAKE_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/testdata/${test} )
 
       # Link to yaml file
-      execute_process( COMMAND ${CMAKE_COMMAND} -E create_symlink
-                       ${CMAKE_CURRENT_SOURCE_DIR}/testinput/${test}.yaml
-                       ${CMAKE_CURRENT_BINARY_DIR}/testinput/${test}.yaml )
+      execute_process(COMMAND ${Python3_EXECUTABLE}
+                      ${CMAKE_BINARY_DIR}/bin/ecsaber_yaml_update.py
+                      ${CMAKE_CURRENT_SOURCE_DIR}/testinput/${test}.yaml
+                      ${CMAKE_CURRENT_BINARY_DIR}/testinput/${test}.yaml )
 
       # Link to reference file
       if( EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/testref/${test}.ref )
@@ -354,11 +355,12 @@
                       endif()
 
                       # Add test
-                      ecbuild_add_test( TARGET saber_${test}_${mpi}-${omp}
-                                        MPI ${mpi}
-                                        OMP ${omp}
-                                        COMMAND ${CMAKE_BINARY_DIR}/bin/saber_quench_${exename}.x
-                                        ARGS testinput/${test}.yaml
+                      ecbuild_add_test( TARGET saber_test_${test}_${mpi}-${omp}
+                                        COMMAND ${CMAKE_BINARY_DIR}/bin/ecsaber_compare.py
+                                        ARGS ${mpi}
+                                             ${omp}
+                                             ${CMAKE_BINARY_DIR}/bin/saber_quench_${exename}.x
+                                             testinput/${test}.yaml
                                         DEPENDS saber_quench_${exename}.x
                                         TEST_DEPENDS ${deps_list} )
                   endif()
@@ -382,11 +384,12 @@
                   endif()
 
                   # Add test
-                  ecbuild_add_test( TARGET saber_${test}_${mpi}-${omp}
-                                    MPI ${mpi}
-                                    OMP ${omp}
-                                    COMMAND ${CMAKE_BINARY_DIR}/bin/saber_quench_error_covariance_toolbox.x
-                                    ARGS testinput/${yaml_prefix}${test}.yaml
+                  ecbuild_add_test( TARGET saber_test_${test}_${mpi}-${omp}
+                                    COMMAND ${CMAKE_BINARY_DIR}/bin/ecsaber_compare.py
+                                    ARGS ${mpi}
+                                         ${omp}
+                                         ${CMAKE_BINARY_DIR}/bin/saber_quench_error_covariance_toolbox.x
+                                         testinput/${yaml_prefix}${test}.yaml
                                     DEPENDS saber_quench_error_covariance_toolbox.x
                                     TEST_DEPENDS ${deps_list} )
               endif()
