--- /home/benjaminm/code/oops-bundle/ecsaber/vader/src/CMakeLists.txt.tmp.bak	2024-03-21 09:30:25.005966097 +0100
+++ /home/benjaminm/code/oops-bundle/ecsaber/vader/src/CMakeLists.txt	2024-03-21 09:30:56.625981870 +0100
@@ -156,7 +156,7 @@
 # Mark Boost as a system library to silence compiler warnings from Boost headers
 target_include_directories( ${PROJECT_NAME} SYSTEM PUBLIC ${Boost_INCLUDE_DIRS} )
 
-target_link_libraries( ${PROJECT_NAME} PUBLIC ${oops_LIBRARIES} ) #TODO: Change to "oops::oops" once oops adds namespace support
+target_link_libraries( ${PROJECT_NAME} PUBLIC ${oops_LIBRARIES} oops-patch ) #TODO: Change to "oops::oops" once oops adds namespace support
 
 if( gsw_FOUND )
   target_link_libraries( ${PROJECT_NAME} PUBLIC gsw )
