--- /home/benjaminm/code/oops-bundle/ecsaber/vader/src/CMakeLists.txt.tmp.bak	2024-08-27 12:39:59.561233177 +0200
+++ /home/benjaminm/code/oops-bundle/ecsaber/vader/src/CMakeLists.txt	2024-08-27 12:40:52.635435758 +0200
@@ -177,7 +177,7 @@
 # Mark Boost as a system library to silence compiler warnings from Boost headers
 target_include_directories( ${PROJECT_NAME} SYSTEM PUBLIC ${Boost_INCLUDE_DIRS} )
 
-target_link_libraries( ${PROJECT_NAME} PUBLIC ${oops_LIBRARIES} ) #TODO: Change to "oops::oops" once oops adds namespace support
+target_link_libraries( ${PROJECT_NAME} PUBLIC ${oops_LIBRARIES} oops_patch ) #TODO: Change to "oops::oops" once oops adds namespace support
 
 if( gsw_FOUND )
   target_link_libraries( ${PROJECT_NAME} PUBLIC gsw )
