--- /home/benjaminm/code/oops-bundle/ecsaber/saber/test/testref/convertstate_lam.ref.tmp	2024-02-14 13:20:33.635916395 +0100
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/test/testref/convertstate_lam.ref	2024-02-13 14:26:07.023649184 +0100
@@ -1,5 +1,5 @@
 Input state: 
-  Valid time: 2010-01-01T12:00:00Z
+Valid time:2010-01-01T12:00:00Z
 Quench geometry grid:
 - name: structured
 - size: 3763
@@ -25,7 +25,7 @@
   velocity_potential: 3.8796907093220716e+06
   surface_pressure: 1.2268659258452002e+06
 Output state: 
-  Valid time: 2010-01-01T12:00:00Z
+Valid time:2010-01-01T12:00:00Z
 Quench geometry grid:
 - name: structured
 - size: 3763
@@ -51,7 +51,7 @@
   velocity_potential: 3.8796907093220716e+06
   surface_pressure: 1.2268659258452002e+06
 Input state: 
-  Valid time: 2010-01-01T12:00:00Z
+Valid time:2010-01-01T12:00:00Z
 Quench geometry grid:
 - name: structured
 - size: 3763
@@ -77,7 +77,7 @@
   velocity_potential: 9.6992267733051791e+05
   surface_pressure: 3.0671648146130005e+05
 Output state: 
-  Valid time: 2010-01-01T12:00:00Z
+Valid time:2010-01-01T12:00:00Z
 Quench geometry grid:
 - name: structured
 - size: 3763
@@ -103,7 +103,7 @@
   velocity_potential: 9.6992267733051791e+05
   surface_pressure: 3.0671648146130005e+05
 Input state: 
-  Valid time: 2010-01-01T12:00:00Z
+Valid time:2010-01-01T12:00:00Z
 Quench geometry grid:
 - name: structured
 - size: 3763
@@ -129,7 +129,7 @@
   velocity_potential: 7.7593814186441432e+02
   surface_pressure: 0.0000000000000000e+00
 Output state: 
-  Valid time: 2010-01-01T12:00:00Z
+Valid time:2010-01-01T12:00:00Z
 Quench geometry grid:
 - name: structured
 - size: 3763
@@ -155,7 +155,7 @@
   velocity_potential: 7.7593814186441432e+02
   surface_pressure: 0.0000000000000000e+00
 Input state: 
-  Valid time: 2010-01-01T12:00:00Z
+Valid time:2010-01-01T12:00:00Z
 Quench geometry grid:
 - name: structured
 - size: 3763
@@ -181,7 +181,7 @@
   velocity_potential: 1.9398453546610358e+02
   surface_pressure: 0.0000000000000000e+00
 Output state: 
-  Valid time: 2010-01-01T12:00:00Z
+Valid time:2010-01-01T12:00:00Z
 Quench geometry grid:
 - name: structured
 - size: 3763
