--- /home/benjaminm/code/oops-bundle/ecsaber/saber/test/testref/dirac_bump_2.ref.tmp	2025-02-21 09:53:34.976518995 +0100
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/test/testref/dirac_bump_2.ref	2025-02-21 09:43:49.636265300 +0100
@@ -44,7 +44,7 @@
           nc2 =      250
           Independent levels:  1[2]
 Adjoint test for block BUMP_VerticalBalance passed
-Norm of input parameter StdDev: 5.7539391781678255e+01
+Norm of input parameter StdDev: 5.7539391781678347e+01
 Adjoint test for block StdDev passed
        Independent levels:  1[2]
           Subset Sc0 size:              762
@@ -61,7 +61,7 @@
              Level   2 ~> 100.0%
              Level   1 ~>  0.100E+01 vert. coord.
              Level   2 ~>  0.200E+01 vert. coord.
-Norm of input parameter stddev - 1: 5.7539391781678255e+01
+Norm of input parameter stddev - 1: 5.7539391781678347e+01
 Adjoint test for block BUMP_StdDev passed
        Independent levels:  1[2]
           Subset Sc0 size:              762
@@ -72,7 +72,7 @@
              Level   2 ~>  0.200E+01 vert. coord.
 Adjoint test for block BUMP_NICAS passed
 Square-root test for block BUMP_NICAS passed
-Norm of input parameter inflation: 6.9558608381710329e+01
+Norm of input parameter inflation: 6.9558608381710044e+01
 Norm of input parameter StdDev: 1.7389652095427326e+02
        Independent levels:  1[2]
           Subset Sc0 size:              762
@@ -84,17 +84,17 @@
 Square-root test for block BUMP_NICAS passed
 Adjoint test for block Ensemble passed
 Square-root test for block Ensemble passed
-Covariance(hybrid) * Increment:
+Covariance(SABER) * Increment:
 - Valid time: 2010-01-01T12:00:00Z
   Geometry: L40x21 [840]
   Fields:
-    eastward_wind: 1.5885846717183985e+00
-    northward_wind: 5.0803567566774199e-01
+    eastward_wind: 1.5885846717183993e+00
+    northward_wind: 5.0803567566774221e-01
 - Valid time: 2010-01-01T18:00:00Z
   Geometry: L40x21 [840]
   Fields:
     eastward_wind: 3.6332544284472684e-01
-    northward_wind: 2.8798332323318876e-01
+    northward_wind: 2.8798332323318893e-01
        Independent levels:  1[2]
           Subset Sc0 size:              762
           Domain area (% of Earth area): 0.100E+03%
@@ -130,7 +130,7 @@
           nc2 =      250
           Independent levels:  1[2]
 Adjoint test for block BUMP_VerticalBalance passed
-Norm of input parameter StdDev: 5.7539391781678255e+01
+Norm of input parameter StdDev: 5.7539391781678347e+01
 Adjoint test for block StdDev passed
        Independent levels:  1[2]
           Subset Sc0 size:              762
@@ -147,7 +147,7 @@
              Level   2 ~> 100.0%
              Level   1 ~>  0.100E+01 vert. coord.
              Level   2 ~>  0.200E+01 vert. coord.
-Norm of input parameter stddev - 1: 5.7539391781678255e+01
+Norm of input parameter stddev - 1: 5.7539391781678347e+01
 Adjoint test for block BUMP_StdDev passed
        Independent levels:  1[2]
           Subset Sc0 size:              762
@@ -158,18 +158,18 @@
              Level   2 ~>  0.200E+01 vert. coord.
 Adjoint test for block BUMP_NICAS passed
 Square-root test for block BUMP_NICAS passed
-Covariance(hybrid1_SABER) * Increment:
+Covariance(SABER1_SABER) * Increment:
 - Valid time: 2010-01-01T12:00:00Z
   Geometry: L40x21 [840]
   Fields:
-    eastward_wind: 3.7259415926535306e-12
-    northward_wind: 1.3202009593235781e-12
+    eastward_wind: 3.7259415926535379e-12
+    northward_wind: 1.3202009593235821e-12
 - Valid time: 2010-01-01T18:00:00Z
   Geometry: L40x21 [840]
   Fields:
-    eastward_wind: 3.7259415926535306e-12
-    northward_wind: 1.3202009593235781e-12
-Norm of input parameter inflation: 6.9558608381710329e+01
+    eastward_wind: 3.7259415926535379e-12
+    northward_wind: 1.3202009593235821e-12
+Norm of input parameter inflation: 6.9558608381710044e+01
 Norm of input parameter StdDev: 1.7389652095427326e+02
        Independent levels:  1[2]
           Subset Sc0 size:              762
@@ -181,14 +181,14 @@
 Square-root test for block BUMP_NICAS passed
 Adjoint test for block Ensemble passed
 Square-root test for block Ensemble passed
-Covariance(hybrid2_SABER) * Increment:
+Covariance(SABER2_SABER) * Increment:
 - Valid time: 2010-01-01T12:00:00Z
   Geometry: L40x21 [840]
   Fields:
-    eastward_wind: 3.1771693434345925e+00
-    northward_wind: 1.0160713513355568e+00
+    eastward_wind: 3.1771693434345938e+00
+    northward_wind: 1.0160713513355570e+00
 - Valid time: 2010-01-01T18:00:00Z
   Geometry: L40x21 [840]
   Fields:
-    eastward_wind: 7.2665088568883618e-01
+    eastward_wind: 7.2665088568883629e-01
     northward_wind: 5.7596664646668794e-01
