--- /home/benjaminm/code/oops-bundle/ecsaber/saber/test/testref/dirac_stddev_1.ref.tmp	2024-07-10 17:22:29.492397506 +0200
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/test/testref/dirac_stddev_1.ref	2024-07-10 17:21:39.527692159 +0200
@@ -22,9 +22,9 @@
 Adjoint test for block ID passed
 Covariance(SABER) diagnostics:
 - Variances at Dirac points:
-  + Value for variable stream_function, subwindow 0, at (longitude, latitude, vertical index) point (0.00000, 0.00000, 1): 1.0356656962519009e+00
+  + Value for variable stream_function, subwindow 0, at (longitude, latitude, vertical index) point (0.00000, 0.00000, 1): 1.0356656962519004e+00
 - Covariances at diagnostic points:
-  + Value for variable stream_function, subwindow 0, at (longitude, latitude, vertical index) point (0.00000, 0.00000, 1): 1.0356656962519009e+00
+  + Value for variable stream_function, subwindow 0, at (longitude, latitude, vertical index) point (0.00000, 0.00000, 1): 1.0356656962519004e+00
 Covariance(SABER) * Increment:
 Valid time:2010-01-01T12:00:00Z
 Quench geometry grid:
