--- /home/benjaminm/code/oops-bundle/ecsaber/saber/test/testref/dirac_interpolation_3.ref.tmp	2024-07-10 17:22:29.200393384 +0200
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/test/testref/dirac_interpolation_3.ref	2024-07-10 17:21:39.523692102 +0200
@@ -37,4 +37,4 @@
   - vert_coord: [1.0000000000000000e+00]
   Mask size: 100%
 Fields:
-  stream_function: 4.9500000000000066e-01
+  stream_function: 5.3270000000000139e-01
