--- /home/benjaminm/code/ecsaber/bundle/ecsaber/saber/test/testref/dirac_localization_bump_1.ref.tmp.bak	2026-02-11 21:46:03.942899314 +0100
+++ /home/benjaminm/code/ecsaber/bundle/ecsaber/saber/test/testref/dirac_localization_bump_1.ref	2026-02-11 21:26:16.836692746 +0100
@@ -15,7 +15,7 @@
     + max    = 1.0000000000000000e+00
     + mean   = 6.5616797900262466e-04
     + stddev = 2.5615775978927613e-02
-Covariance(hybrid) dirac test:
+Covariance(SABER-hybrid) dirac test:
        Independent levels:  1[2]
           Subset Sc0 size:              762
           Domain area (% of Earth area): 0.100E+03%
@@ -94,7 +94,7 @@
 Square-root test for the multivariate central block passed
 Adjoint test for block Ensemble passed
 Square-root test for block Ensemble passed
-Covariance(hybrid) * Increment:
+Covariance(SABER-hybrid) * Increment:
 - Valid time: 2010-01-01T12:00:00Z
   Geometry: L40x21 [840]
   Fields:
@@ -111,7 +111,7 @@
     + max    = 1.2673127220262930e+00
     + mean   = 4.1361283797479581e-03
     + stddev = 4.8405861487388492e-02
-Covariance(hybrid-cmp1_ensemble) dirac test:
+Covariance(SABER-hybrid-cmp1_SABER-ensemble) dirac test:
        Independent levels:  1[2]
           Subset Sc0 size:              762
           Domain area (% of Earth area): 0.100E+03%
@@ -143,7 +143,7 @@
              ns =                             1178
              v%n_s =                             2
              c%n_s[global] =                 28436
-Covariance(hybrid-cmp1_ensemble) * Increment:
+Covariance(SABER-hybrid-cmp1_SABER-ensemble) * Increment:
 - Valid time: 2010-01-01T12:00:00Z
   Geometry: L40x21 [840]
   Fields:
@@ -160,7 +160,7 @@
     + max    = 1.2673127220262930e+00
     + mean   = 4.1361283797479581e-03
     + stddev = 4.8405861487388492e-02
-Localization(hybrid-cmp1_ensemble_localization) dirac test:
+Localization(SABER-hybrid-cmp1_SABER-ensemble_localization) dirac test:
        Independent levels:  1[2]
           Subset Sc0 size:              762
           Domain area (% of Earth area): 0.100E+03%
@@ -192,7 +192,7 @@
              ns =                             1178
              v%n_s =                             2
              c%n_s[global] =                 28436
-Localization(hybrid-cmp1_ensemble_localization) * Increment:
+Localization(SABER-hybrid-cmp1_SABER-ensemble_localization) * Increment:
 - Valid time: 2010-01-01T12:00:00Z
   Geometry: L40x21 [840]
   Fields:
@@ -209,11 +209,11 @@
     + max    = 1.0000000000000002e+00
     + mean   = 6.7588404393921202e-03
     + stddev = 5.4149030644156722e-02
-Localization(hybrid-cmp1_ensemble_localization) diagnostics:
+Localization(SABER-hybrid-cmp1_SABER-ensemble_localization) diagnostics:
 - Localization at zero separation:
   + Value for variable eastward_wind, subwindow 0, at (longitude, latitude, vertical index) point (0.00000, 0.00000, 1): 9.9999999999999956e-01
   + Value for variable air_horizontal_streamfunction, subwindow 0, at (longitude, latitude, vertical index) point (0.00000, 0.00000, 1): 1.0000000000000002e+00
-Covariance(hybrid-cmp2_ensemble) dirac test:
+Covariance(SABER-hybrid-cmp2_SABER-ensemble) dirac test:
        Independent levels:  1[2]
           Subset Sc0 size:              762
           Domain area (% of Earth area): 0.100E+03%
@@ -228,7 +228,7 @@
              Level   2 ~> 100.0%
              Level   1 ~>  0.100E+01 vert. coord.
              Level   2 ~>  0.200E+01 vert. coord.
-Covariance(hybrid-cmp2_ensemble) * Increment:
+Covariance(SABER-hybrid-cmp2_SABER-ensemble) * Increment:
 - Valid time: 2010-01-01T12:00:00Z
   Geometry: L40x21 [840]
   Fields:
@@ -245,7 +245,7 @@
     + max    = 1.2673127220262930e+00
     + mean   = 4.1361283797479581e-03
     + stddev = 4.8405861487388492e-02
-Localization(hybrid-cmp2_ensemble_localization) dirac test:
+Localization(SABER-hybrid-cmp2_SABER-ensemble_localization) dirac test:
        Independent levels:  1[2]
           Subset Sc0 size:              762
           Domain area (% of Earth area): 0.100E+03%
@@ -260,7 +260,7 @@
              Level   2 ~> 100.0%
              Level   1 ~>  0.100E+01 vert. coord.
              Level   2 ~>  0.200E+01 vert. coord.
-Localization(hybrid-cmp2_ensemble_localization) * Increment:
+Localization(SABER-hybrid-cmp2_SABER-ensemble_localization) * Increment:
 - Valid time: 2010-01-01T12:00:00Z
   Geometry: L40x21 [840]
   Fields:
@@ -277,11 +277,11 @@
     + max    = 1.0000000000000002e+00
     + mean   = 6.7588404393921202e-03
     + stddev = 5.4149030644156722e-02
-Localization(hybrid-cmp2_ensemble_localization) diagnostics:
+Localization(SABER-hybrid-cmp2_SABER-ensemble_localization) diagnostics:
 - Localization at zero separation:
   + Value for variable eastward_wind, subwindow 0, at (longitude, latitude, vertical index) point (0.00000, 0.00000, 1): 9.9999999999999956e-01
   + Value for variable air_horizontal_streamfunction, subwindow 0, at (longitude, latitude, vertical index) point (0.00000, 0.00000, 1): 1.0000000000000002e+00
-Covariance(hybrid-cmp3_SABER-ensemble) dirac test:
+Covariance(SABER-hybrid-cmp3_SABER-ensemble) dirac test:
        Independent levels:  1[2]
           Subset Sc0 size:              762
           Domain area (% of Earth area): 0.100E+03%
@@ -293,7 +293,7 @@
 Square-root test for block BUMP_NICAS passed
 Adjoint test for block Ensemble passed
 Square-root test for block Ensemble passed
-Covariance(hybrid-cmp3_SABER-ensemble) * Increment:
+Covariance(SABER-hybrid-cmp3_SABER-ensemble) * Increment:
 - Valid time: 2010-01-01T12:00:00Z
   Geometry: L40x21 [840]
   Fields:
@@ -310,7 +310,7 @@
     + max    = 1.2673127220262930e+00
     + mean   = 4.1361283797479581e-03
     + stddev = 4.8405861487388492e-02
-Localization(hybrid-cmp3_SABER-ensemble_localization) dirac test:
+Localization(SABER-hybrid-cmp3_SABER-ensemble_localization) dirac test:
        Independent levels:  1[2]
           Subset Sc0 size:              762
           Domain area (% of Earth area): 0.100E+03%
@@ -320,7 +320,7 @@
              Level   2 ~>  0.200E+01 vert. coord.
 Adjoint test for block BUMP_NICAS passed
 Square-root test for block BUMP_NICAS passed
-Localization(hybrid-cmp3_SABER-ensemble_localization) * Increment:
+Localization(SABER-hybrid-cmp3_SABER-ensemble_localization) * Increment:
 - Valid time: 2010-01-01T12:00:00Z
   Geometry: L40x21 [840]
   Fields:
@@ -337,11 +337,11 @@
     + max    = 1.0000000000000002e+00
     + mean   = 6.7588404393921202e-03
     + stddev = 5.4149030644156722e-02
-Localization(hybrid-cmp3_SABER-ensemble_localization) diagnostics:
+Localization(SABER-hybrid-cmp3_SABER-ensemble_localization) diagnostics:
 - Localization at zero separation:
   + Value for variable eastward_wind, subwindow 0, at (longitude, latitude, vertical index) point (0.00000, 0.00000, 1): 9.9999999999999956e-01
   + Value for variable air_horizontal_streamfunction, subwindow 0, at (longitude, latitude, vertical index) point (0.00000, 0.00000, 1): 1.0000000000000002e+00
-Covariance(hybrid-cmp4_SABER-ensemble) dirac test:
+Covariance(SABER-hybrid-cmp4_SABER-ensemble) dirac test:
        Independent levels:  1[2]
           Subset Sc0 size:              762
           Domain area (% of Earth area): 0.100E+03%
@@ -364,7 +364,7 @@
 Square-root test for the multivariate central block passed
 Adjoint test for block Ensemble passed
 Square-root test for block Ensemble passed
-Covariance(hybrid-cmp4_SABER-ensemble) * Increment:
+Covariance(SABER-hybrid-cmp4_SABER-ensemble) * Increment:
 - Valid time: 2010-01-01T12:00:00Z
   Geometry: L40x21 [840]
   Fields:
@@ -381,7 +381,7 @@
     + max    = 1.2673127220262930e+00
     + mean   = 4.1361283797479581e-03
     + stddev = 4.8405861487388492e-02
-Localization(hybrid-cmp4_SABER-ensemble_localization) dirac test:
+Localization(SABER-hybrid-cmp4_SABER-ensemble_localization) dirac test:
        Independent levels:  1[2]
           Subset Sc0 size:              762
           Domain area (% of Earth area): 0.100E+03%
@@ -402,7 +402,7 @@
 Square-root test for block BUMP_NICAS passed
 Square-root test for block BUMP_NICAS passed
 Square-root test for the multivariate central block passed
-Localization(hybrid-cmp4_SABER-ensemble_localization) * Increment:
+Localization(SABER-hybrid-cmp4_SABER-ensemble_localization) * Increment:
 - Valid time: 2010-01-01T12:00:00Z
   Geometry: L40x21 [840]
   Fields:
@@ -419,7 +419,7 @@
     + max    = 1.0000000000000002e+00
     + mean   = 6.7588404393921202e-03
     + stddev = 5.4149030644156722e-02
-Localization(hybrid-cmp4_SABER-ensemble_localization) diagnostics:
+Localization(SABER-hybrid-cmp4_SABER-ensemble_localization) diagnostics:
 - Localization at zero separation:
   + Value for variable eastward_wind, subwindow 0, at (longitude, latitude, vertical index) point (0.00000, 0.00000, 1): 9.9999999999999956e-01
   + Value for variable air_horizontal_streamfunction, subwindow 0, at (longitude, latitude, vertical index) point (0.00000, 0.00000, 1): 1.0000000000000002e+00
