--- /home/benjaminm/code/ecsaber/bundle/ecsaber/saber/test/testref/dirac_bump_2.ref.tmp.bak	2026-02-02 17:25:41.592963281 +0100
+++ /home/benjaminm/code/ecsaber/bundle/ecsaber/saber/test/testref/dirac_bump_2.ref	2026-02-02 17:27:56.130959280 +0100
@@ -97,7 +97,7 @@
 Square-root test for block BUMP_NICAS passed
 Adjoint test for block Ensemble passed
 Square-root test for block Ensemble passed
-Covariance(hybrid) * Increment:
+Covariance(SABER) * Increment:
 - Valid time: 2010-01-01T12:00:00Z
   Geometry: L40x21 [840]
   Fields:
@@ -147,7 +147,6 @@
           Wavenumber:  5
              - psi/chi to u:  39.37007%
              - psi/chi to v:  39.37007%
-Adjoint test for block BUMP_PsiChiToUV passed
        Independent levels:  1[2]
           Subset Sc0 size:              762
           Domain area (% of Earth area): 0.100E+03%
@@ -158,9 +157,7 @@
           nc1 =      500
           nc2 =      250
           Independent levels:  1[2]
-Adjoint test for block BUMP_VerticalBalance passed
 Norm of input parameter StdDev: 5.4937700734311314e+01
-Adjoint test for block StdDev passed
        Independent levels:  1[2]
           Subset Sc0 size:              762
           Domain area (% of Earth area): 0.100E+03%
@@ -168,7 +165,6 @@
              Level   2 ~> 100.0%
              Level   1 ~>  0.100E+01 vert. coord.
              Level   2 ~>  0.200E+01 vert. coord.
-Adjoint test for block BUMP_StdDev passed
        Independent levels:  1[2]
           Subset Sc0 size:              762
           Domain area (% of Earth area): 0.100E+03%
@@ -177,7 +173,6 @@
              Level   1 ~>  0.100E+01 vert. coord.
              Level   2 ~>  0.200E+01 vert. coord.
 Norm of input parameter stddev - 1: 5.4937700734311314e+01
-Adjoint test for block BUMP_StdDev passed
        Independent levels:  1[2]
           Subset Sc0 size:              762
           Domain area (% of Earth area): 0.100E+03%
@@ -185,9 +180,7 @@
              Level   2 ~> 100.0%
              Level   1 ~>  0.100E+01 vert. coord.
              Level   2 ~>  0.200E+01 vert. coord.
-Adjoint test for block BUMP_NICAS passed
-Square-root test for block BUMP_NICAS passed
-Covariance(hybrid1_SABER) * Increment:
+Covariance(SABER1_SABER) * Increment:
 - Valid time: 2010-01-01T12:00:00Z
   Geometry: L40x21 [840]
   Fields:
@@ -216,8 +209,6 @@
     + stddev = 3.2868334756832761e-14
 Norm of input parameter inflation: 6.6250433960843253e+01
 Norm of input parameter StdDev: 1.6562608490210712e+02
-Adjoint test for block StdDev passed
-Adjoint test for block ID passed
        Independent levels:  1[2]
           Subset Sc0 size:              762
           Domain area (% of Earth area): 0.100E+03%
@@ -225,11 +216,7 @@
              Level   2 ~> 100.0%
              Level   1 ~>  0.100E+01 vert. coord.
              Level   2 ~>  0.200E+01 vert. coord.
-Adjoint test for block BUMP_NICAS passed
-Square-root test for block BUMP_NICAS passed
-Adjoint test for block Ensemble passed
-Square-root test for block Ensemble passed
-Covariance(hybrid2_SABER) * Increment:
+Covariance(SABER2_SABER) * Increment:
 - Valid time: 2010-01-01T12:00:00Z
   Geometry: L40x21 [840]
   Fields:
