--- /home/benjaminm/code/ecsaber/bundle/ecsaber/saber/test/testref/dirac_localization_bump_4.ref.tmp.bak	2026-02-11 15:00:27.431003708 +0100
+++ /home/benjaminm/code/ecsaber/bundle/ecsaber/saber/test/testref/dirac_localization_bump_4.ref	2026-02-11 15:17:09.339409283 +0100
@@ -66,7 +66,7 @@
 Square-root test for the multivariate central block passed
 Adjoint test for block Ensemble passed
 Square-root test for block Ensemble passed
-Covariance(hybrid) * Increment:
+Covariance(SABER) * Increment:
 - Valid time: 2010-01-01T12:00:00Z
   Geometry: L40x21 [840]
   Fields:
@@ -105,7 +105,7 @@
              ns =                              522
              v%n_s =                             2
              c%n_s[global] =                 13688
-Covariance(hybrid1_ensemble) * Increment:
+Covariance(SABER1_SABER) * Increment:
 - Valid time: 2010-01-01T12:00:00Z
   Geometry: L40x21 [840]
   Fields:
