--- /home/benjaminm/code/oops-bundle/ecsaber/saber/test/testref/dirac_fastlam_9.ref.tmp	2024-05-22 11:25:08.814620620 +0200
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/test/testref/dirac_fastlam_9.ref	2024-05-22 10:51:10.555416943 +0200
@@ -24,12 +24,12 @@
   stream_function: 3.0000000000000000e+00
   velocity_potential: 0.0000000000000000e+00
   surface_pressure: 3.0000000000000000e+00
-Norm of input parameter weight_0: 5.1821246270964352e+01
-Norm of input parameter weight_1: 1.3660485222955450e+02
-Norm of input parameter weight_2: 5.1115031705121055e+01
-Norm of input parameter normalization_0: 2.1701285379844464e+02
-Norm of input parameter normalization_1: 2.3177645116279072e+02
-Norm of input parameter normalization_2: 2.3173024180072034e+02
+Norm of input parameter weight_0: 7.2930744333457682e+01
+Norm of input parameter weight_1: 1.2620653673036158e+02
+Norm of input parameter weight_2: 4.5290975324119849e+01
+Norm of input parameter normalization_0: 2.1351287226377931e+02
+Norm of input parameter normalization_1: 2.2559151288033380e+02
+Norm of input parameter normalization_2: 2.3041565899377736e+02
     FastLAM interpolation accuracy test passed
     FastLAM interpolation adjoint test passed
     FastLAM interpolation accuracy test passed
@@ -67,6 +67,6 @@
   - vert_coord: [1.0000000000000000e+00]
   Mask size: 100%
 Fields:
-  stream_function: 1.3667526069849375e+01
-  velocity_potential: 1.3667526069849375e+01
-  surface_pressure: 1.1833883108358981e+01
+  stream_function: 1.2835522825802885e+01
+  velocity_potential: 1.2835522825802885e+01
+  surface_pressure: 1.1190234305617500e+01
