--- /home/benjaminm/code/oops-bundle/ecsaber/saber/test/testref/dirac_bump_5.ref.tmp	2024-05-06 14:16:31.275700211 +0200
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/test/testref/dirac_bump_5.ref	2024-05-06 13:02:19.559697703 +0200
@@ -46,4 +46,4 @@
   - vert_coord: [1.0000000000000000e+00,2.0000000000000000e+00]
   Mask size: 100%
 Fields:
-  stream_function: 2.0497561324834241e+00
+  stream_function: 2.0496834509272146e+00
