--- /home/benjaminm/code/oops-bundle/ecsaber/saber/test/testref/dirac_bump_2.ref.tmp	2024-05-06 14:08:02.842279981 +0200
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/test/testref/dirac_bump_2.ref	2024-05-06 14:03:37.101476530 +0200
@@ -92,7 +92,7 @@
              Level   2 ~>  0.200E+01 vert. coord.
 Adjoint test for block BUMP_NICAS passed
 Square-root test for block BUMP_NICAS passed
-Norm of input parameter inflation: 5.7319844819175614e+01
+Norm of input parameter inflation: 5.7319844819175621e+01
 Norm of input parameter StdDev: 5.3046693215269400e+01
        Independent levels:  1[2]
           Subset Sc0 size:              762
@@ -106,7 +106,7 @@
 Square-root test for block BUMP_NICAS passed
 Adjoint test for block Ensemble passed
 Square-root test for block Ensemble passed
-Covariance(hybrid) * Increment:
+Covariance(SABER) * Increment:
 Valid time:2010-01-01T12:00:00Z
 Quench geometry grid:
 - name: L40x21
@@ -123,8 +123,8 @@
   - vert_coord: [1.0000000000000000e+00,2.0000000000000000e+00]
   Mask size: 100%
 Fields:
-  eastward_wind: 4.6277913662317277e-01
-  northward_wind: 1.2275820379462828e-01
+  eastward_wind: 3.1497228077222345e-01
+  northward_wind: 1.2579405910183361e-01
 Valid time:2010-01-01T18:00:00Z
 Quench geometry grid:
 - name: L40x21
@@ -141,8 +141,8 @@
   - vert_coord: [1.0000000000000000e+00,2.0000000000000000e+00]
   Mask size: 100%
 Fields:
-  eastward_wind: 9.1227416041006398e-02
-  northward_wind: 7.7720086254917012e-02
+  eastward_wind: 8.7692143605848441e-02
+  northward_wind: 9.0014058992322843e-02
        Independent levels:  1[2]
           Subset Sc0 size:              762
           Min. / max. longitudes: -171.0 /   180.0
@@ -200,7 +200,7 @@
              Level   2 ~>  0.200E+01 vert. coord.
 Adjoint test for block BUMP_NICAS passed
 Square-root test for block BUMP_NICAS passed
-Covariance(hybrid1_SABER) * Increment:
+Covariance(SABER1_SABER) * Increment:
 Valid time:2010-01-01T12:00:00Z
 Quench geometry grid:
 - name: L40x21
@@ -237,7 +237,7 @@
 Fields:
   eastward_wind: 9.0462803982807654e-14
   northward_wind: 2.7442608459841396e-14
-Norm of input parameter inflation: 5.7319844819175614e+01
+Norm of input parameter inflation: 5.7319844819175621e+01
 Norm of input parameter StdDev: 5.3046693215269400e+01
        Independent levels:  1[2]
           Subset Sc0 size:              762
@@ -251,7 +251,7 @@
 Square-root test for block BUMP_NICAS passed
 Adjoint test for block Ensemble passed
 Square-root test for block Ensemble passed
-Covariance(hybrid2_SABER) * Increment:
+Covariance(SABER2_SABER) * Increment:
 Valid time:2010-01-01T12:00:00Z
 Quench geometry grid:
 - name: L40x21
@@ -268,7 +268,7 @@
   - vert_coord: [1.0000000000000000e+00,2.0000000000000000e+00]
   Mask size: 100%
 Fields:
-  eastward_wind: 6.2994456154442680e-01
+  eastward_wind: 6.2994456154442668e-01
   northward_wind: 2.5158811820366722e-01
 Valid time:2010-01-01T18:00:00Z
 Quench geometry grid:
@@ -286,5 +286,5 @@
   - vert_coord: [1.0000000000000000e+00,2.0000000000000000e+00]
   Mask size: 100%
 Fields:
-  eastward_wind: 1.7538428721168914e-01
+  eastward_wind: 1.7538428721168911e-01
   northward_wind: 1.8002811798464560e-01
