--- /home/benjaminm/code/ecsaber/bundle/ecsaber/saber/test/testref/dirac_localization_mixed_1.ref.tmp.bak	2026-02-11 15:00:26.918931540 +0100
+++ /home/benjaminm/code/ecsaber/bundle/ecsaber/saber/test/testref/dirac_localization_mixed_1.ref	2026-02-11 15:14:32.816992154 +0100
@@ -38,7 +38,7 @@
              ns =                             1178
              v%n_s =                             2
              c%n_s[global] =                 35280
-Covariance(ensemble) * Increment:
+Covariance(SABER) * Increment:
 - Valid time: 2010-01-01T12:00:00Z
   Geometry: F12 [1152]
   Fields:
