--- /home/benjaminm/code/ecsaber/bundle/ecsaber/saber/test/testref/dirac_bump_6.ref.tmp.bak	2026-02-11 21:46:03.381897840 +0100
+++ /home/benjaminm/code/ecsaber/bundle/ecsaber/saber/test/testref/dirac_bump_6.ref	2026-02-11 21:25:53.550527133 +0100
@@ -10,27 +10,27 @@
   - northward_wind (2 levels):
     + min    = 0.0000000000000000e+00
     + max    = 0.0000000000000000e+00
-Covariance(hybrid) dirac test:
+Covariance(SABER-hybrid) dirac test:
 Adjoint test for block Ensemble passed
 Square-root test for block Ensemble passed
-Covariance(hybrid) * Increment:
+Covariance(SABER-hybrid) * Increment:
 - Valid time: 2010-01-01T12:00:00Z
   Geometry: L40x21 [840]
   Fields:
   - eastward_wind (2 levels):
     + min    = -6.0918753181843444e-01
-    + max    = 9.0019572469053788e-01
-    + mean   = 4.9944483467209110e-02
-    + stddev = 1.9720463112584824e-01
+    + max    = 9.0019572469053810e-01
+    + mean   = 4.9944483467209130e-02
+    + stddev = 1.9720463112584827e-01
   - northward_wind (2 levels):
-    + min    = -5.9337315609450869e-01
-    + max    = 5.9924954825054466e-01
-    + mean   = -1.5150635424641333e-02
-    + stddev = 1.9061663485918223e-01
-Covariance(hybrid-cmp1_SABER-ensemble) dirac test:
+    + min    = -5.9337315609450847e-01
+    + max    = 5.9924954825054488e-01
+    + mean   = -1.5150635424641342e-02
+    + stddev = 1.9061663485918226e-01
+Covariance(SABER-hybrid-cmp1_SABER-ensemble) dirac test:
 Adjoint test for block Ensemble passed
 Square-root test for block Ensemble passed
-Covariance(hybrid-cmp1_SABER-ensemble) * Increment:
+Covariance(SABER-hybrid-cmp1_SABER-ensemble) * Increment:
 - Valid time: 2010-01-01T12:00:00Z
   Geometry: L40x21 [840]
   Fields:
@@ -44,8 +44,8 @@
     + max    = 5.9924954825054466e-01
     + mean   = -1.5150635424641333e-02
     + stddev = 1.9061663485918223e-01
-Covariance(hybrid-cmp2_ensemble) dirac test:
-Covariance(hybrid-cmp2_ensemble) * Increment:
+Covariance(SABER-hybrid-cmp2_SABER-ensemble) dirac test:
+Covariance(SABER-hybrid-cmp2_SABER-ensemble) * Increment:
 - Valid time: 2010-01-01T12:00:00Z
   Geometry: L40x21 [840]
   Fields:
