--- /home/benjaminm/code/oops-bundle/ecsaber/saber/test/testref/dirac_fastlam_6.ref.tmp	2024-04-04 15:19:54.705078235 +0200
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/test/testref/dirac_fastlam_6.ref	2024-04-04 14:55:56.451400159 +0200
@@ -24,12 +24,12 @@
   stream_function: 3.0000000000000000e+00
   velocity_potential: 0.0000000000000000e+00
   surface_pressure: 3.0000000000000000e+00
-Norm of input parameter weight_0: 5.1821246270964352e+01
-Norm of input parameter weight_1: 1.3660485222955450e+02
-Norm of input parameter weight_2: 5.1115031705121055e+01
+Norm of input parameter weight_0: 5.1821246270964323e+01
+Norm of input parameter weight_1: 1.3660485222955452e+02
+Norm of input parameter weight_2: 5.1115031705121069e+01
 Norm of input parameter normalization_0: 2.1701285379844464e+02
-Norm of input parameter normalization_1: 2.3177645116279072e+02
-Norm of input parameter normalization_2: 2.3173024180072034e+02
+Norm of input parameter normalization_1: 2.3177645116279069e+02
+Norm of input parameter normalization_2: 2.3173024180072025e+02
     FastLAM interpolation accuracy test passed
     FastLAM interpolation adjoint test passed
     FastLAM redToRows test passed
