--- /home/benjaminm/code/oops-bundle/ecsaber/saber/test/testref/dirac_fastlam_8.ref.tmp	2024-04-04 15:19:54.721078258 +0200
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/test/testref/dirac_fastlam_8.ref	2024-04-04 14:55:57.059400633 +0200
@@ -38,15 +38,15 @@
     FastLAM interpolation adjoint test passed
     FastLAM interpolation accuracy test passed
     FastLAM interpolation adjoint test passed
-Norm of output parameter : 1.1644890209873171e+04
-Norm of output parameter : 8.7234826187710150e+03
-Norm of output parameter : 1.6992922050928357e+03
-Norm of output parameter : 5.1821246270964352e+01
-Norm of output parameter : 1.3660485222955450e+02
-Norm of output parameter : 5.1115031705121055e+01
-Norm of output parameter : 2.1701285379844464e+02
-Norm of output parameter : 2.3177645116279072e+02
-Norm of output parameter : 2.3173024180072034e+02
+Norm of output parameter index i: 1.1644890209873171e+04
+Norm of output parameter index j: 8.7234826187710150e+03
+Norm of output parameter normalized horizontal length-scale: 1.6992922050928357e+03
+Norm of output parameter weight - 0: 5.1821246270964323e+01
+Norm of output parameter weight - 1: 1.3660485222955452e+02
+Norm of output parameter weight - 2: 5.1115031705121069e+01
+Norm of output parameter normalization - 0: 2.1701285379844464e+02
+Norm of output parameter normalization - 1: 2.3177645116279069e+02
+Norm of output parameter normalization - 2: 2.3173024180072025e+02
 Adjoint test for block FastLAM passed
 Square-root test for block FastLAM passed
 Covariance(SABER) * Increment:
