--- /home/benjaminm/code/oops-bundle/ecsaber/saber/test/testref/dirac_bump_1.ref.tmp	2024-05-06 14:08:02.854280016 +0200
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/test/testref/dirac_bump_1.ref	2024-05-06 14:04:11.077582887 +0200
@@ -118,8 +118,8 @@
   - vert_coord: [1.0000000000000000e+00,2.0000000000000000e+00]
   Mask size: 100%
 Fields:
-  eastward_wind: 4.6277913662317277e-01
-  northward_wind: 1.2275820379462828e-01
+  eastward_wind: 7.6469517419453437e-01
+  northward_wind: 2.4507263685615482e-01
 Valid time:2010-01-01T18:00:00Z
 Quench geometry grid:
 - name: L40x21
@@ -136,8 +136,8 @@
   - vert_coord: [1.0000000000000000e+00,2.0000000000000000e+00]
   Mask size: 100%
 Fields:
-  eastward_wind: 9.1227416041006412e-02
-  northward_wind: 7.7720086254916998e-02
+  eastward_wind: 1.7245707129062748e-01
+  northward_wind: 1.4105636151765230e-01
        Independent levels:  1[2]
           Subset Sc0 size:              762
           Min. / max. longitudes: -171.0 /   180.0
@@ -212,7 +212,7 @@
   - vert_coord: [1.0000000000000000e+00,2.0000000000000000e+00]
   Mask size: 100%
 Fields:
-  eastward_wind: 9.0462803982807667e-14
+  eastward_wind: 9.0462803982807654e-14
   northward_wind: 2.7442608459841396e-14
 Valid time:2010-01-01T18:00:00Z
 Quench geometry grid:
@@ -230,7 +230,7 @@
   - vert_coord: [1.0000000000000000e+00,2.0000000000000000e+00]
   Mask size: 100%
 Fields:
-  eastward_wind: 9.0462803982807667e-14
+  eastward_wind: 9.0462803982807654e-14
   northward_wind: 2.7442608459841396e-14
        Independent levels:  1[2]
           Subset Sc0 size:              762
@@ -258,8 +258,8 @@
   - vert_coord: [1.0000000000000000e+00,2.0000000000000000e+00]
   Mask size: 100%
 Fields:
-  eastward_wind: 6.2994456154442668e-01
-  northward_wind: 2.5158811820366722e-01
+  eastward_wind: 1.5293903483890439e+00
+  northward_wind: 4.9014527371230965e-01
 Valid time:2010-01-01T18:00:00Z
 Quench geometry grid:
 - name: L40x21
@@ -276,8 +276,8 @@
   - vert_coord: [1.0000000000000000e+00,2.0000000000000000e+00]
   Mask size: 100%
 Fields:
-  eastward_wind: 1.7538428721168914e-01
-  northward_wind: 1.8002811798464560e-01
+  eastward_wind: 3.4491414258125613e-01
+  northward_wind: 2.8211272303530460e-01
        Independent levels:  1[2]
           Subset Sc0 size:              762
           Min. / max. longitudes: -171.0 /   180.0
@@ -289,7 +289,9 @@
              Level   2 ~>  0.200E+01 vert. coord.
 Localization(hybrid2_ensemble_localization) diagnostics:
 - Localization at zero separation:
-  + Value for variable eastward_wind, subwindow 0, at (longitude, latitude, vertical index) point (0.00000, 0.00000, 1): 1.0000000000000002e+00
+  + Value for variable eastward_wind, subwindow 0, at (longitude, latitude, vertical index) point (0.00000, 0.00000, 1): 1.0000000000000000e+00
+  + Value for variable eastward_wind, subwindow 0, at (longitude, latitude, vertical index) point (0.00000, 0.00000, 1): 1.0000000000000000e+00
+  + Value for variable northward_wind, subwindow 0, at (longitude, latitude, vertical index) point (0.00000, 0.00000, 1): 1.0000000000000000e+00
 Localization(hybrid2_ensemble) * Increment:
 Valid time:2010-01-01T12:00:00Z
 Quench geometry grid:
