--- /home/benjaminm/code/oops-bundle/ecsaber/saber/test/testref/dirac_bump_4.ref.tmp	2024-05-06 14:16:30.951699335 +0200
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/test/testref/dirac_bump_4.ref	2024-05-06 13:02:08.499665707 +0200
@@ -40,7 +40,7 @@
           nc2 =      250
           Independent levels:  1[2]
 Adjoint test for block BUMP_VerticalBalance passed
-Norm of input parameter StdDev: 5.7539391781676251e+01
+Norm of input parameter StdDev: 5.7539391781678255e+01
 Adjoint test for block StdDev passed
        Independent levels:  1[2]
           Subset Sc0 size:              762
@@ -82,5 +82,5 @@
   - vert_coord: [1.0000000000000000e+00,2.0000000000000000e+00]
   Mask size: 100%
 Fields:
-  eastward_wind: 2.3187376237098562e-13
-  northward_wind: 2.1956469456184360e-13
+  eastward_wind: 2.1734216281358465e-13
+  northward_wind: 2.0283339969344209e-13
