--- /home/benjaminm/code/ecsaber/bundle/ecsaber/saber/test/testref/dirac_spectralb_gauss_vader_3.ref.tmp	2026-02-11 21:46:03.666898588 +0100
+++ /home/benjaminm/code/ecsaber/bundle/ecsaber/saber/test/testref/dirac_spectralb_gauss_vader_3.ref	2026-02-11 21:27:29.522206958 +0100
@@ -28,8 +28,8 @@
   - water_vapor_mixing_ratio_wrt_moist_air_and_condensed_water (70 levels):
     + min    = 0.0000000000000000e+00
     + max    = 0.0000000000000000e+00
-Covariance(hybrid) dirac test:
-Covariance(hybrid) * Increment:
+Covariance(SABER-hybrid) dirac test:
+Covariance(SABER-hybrid) * Increment:
 - Valid time: 2016-01-01T16:00:00Z
   Geometry: F12 [1152]
   Fields:
@@ -73,15 +73,15 @@
     + max    = 2.7997032594450769e-08
     + mean   = 4.5094718781909049e-14
     + stddev = 4.3893637202727890e-10
-Covariance(hybrid) diagnostics:
+Covariance(SABER-hybrid) diagnostics:
 - Variances at Dirac points:
   + Value for variable dry_air_density_levels_minus_one, subwindow 0, at (longitude, latitude, vertical index) point (180.00000, 3.67270, 1): 1.0355549127483617e-07
 - Covariances at diagnostic points:
   + Value for variable dry_air_density_levels_minus_one, subwindow 0, at (longitude, latitude, vertical index) point (0.00000, 84.37646, 1): -8.5299495350436338e-10
   + Value for variable dry_air_density_levels_minus_one, subwindow 0, at (longitude, latitude, vertical index) point (0.00000, 3.67270, 70): 7.2264863000533227e-16
   + Value for variable dry_air_density_levels_minus_one, subwindow 0, at (longitude, latitude, vertical index) point (180.00000, 3.67270, 1): 1.0355549127483617e-07
-Covariance(hybrid-cmp1_SABER-ensemble) dirac test:
-Covariance(hybrid-cmp1_SABER-ensemble) * Increment:
+Covariance(SABER-hybrid-cmp1_SABER-ensemble) dirac test:
+Covariance(SABER-hybrid-cmp1_SABER-ensemble) * Increment:
 - Valid time: 2016-01-01T16:00:00Z
   Geometry: F12 [1152]
   Fields:
@@ -125,7 +125,7 @@
     + max    = 2.7997032594450769e-08
     + mean   = 4.5094718781909049e-14
     + stddev = 4.3893637202727890e-10
-Covariance(hybrid-cmp1_SABER-ensemble) diagnostics:
+Covariance(SABER-hybrid-cmp1_SABER-ensemble) diagnostics:
 - Variances at Dirac points:
   + Value for variable dry_air_density_levels_minus_one, subwindow 0, at (longitude, latitude, vertical index) point (180.00000, 3.67270, 1): 1.0355549127483617e-07
 - Covariances at diagnostic points:
