--- /home/benjaminm/code/oops-bundle/ecsaber/saber/test/testref/dirac_fastlam_1.ref.tmp	2024-04-04 15:19:54.641078143 +0200
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/test/testref/dirac_fastlam_1.ref	2024-04-04 14:55:55.511399428 +0200
@@ -32,11 +32,11 @@
     FastLAM interpolation adjoint test passed
     FastLAM redToRows test passed
     FastLAM rowsToCols test passed
-Norm of output parameter : 1.1644890209873171e+04
-Norm of output parameter : 8.7234826187710150e+03
-Norm of output parameter : 1.6276513347796993e+03
-Norm of output parameter : 2.0345269720502603e+02
-Norm of output parameter : 2.1747388285958212e+02
+Norm of output parameter index i: 1.1644890209873171e+04
+Norm of output parameter index j: 8.7234826187710150e+03
+Norm of output parameter normalized horizontal length-scale: 1.6276513347796993e+03
+Norm of output parameter weight - 0: 2.0345269720502603e+02
+Norm of output parameter normalization - 0: 2.1747388285958212e+02
 Adjoint test for block FastLAM passed
 Square-root test for block FastLAM passed
 Covariance(SABER) * Increment:
