--- /home/benjaminm/code/oops-bundle/ecsaber/saber/test/testref/error_covariance_training_bump_nicas_6.ref.tmp	2024-05-06 14:16:30.779698870 +0200
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/test/testref/error_covariance_training_bump_nicas_6.ref	2024-05-06 13:02:01.411645136 +0200
@@ -19,11 +19,11 @@
 Norm of input parameter rh1 - 1: 2.6600928484758756e+08
 Norm of input parameter rh1 - 2: 2.1469901889690262e+08
 Norm of input parameter rh2 - 1: 2.8142686892415738e+08
-Norm of input parameter rh2 - 2: 2.3890998875146008e+08
+Norm of input parameter rh2 - 2: 2.3890998875146011e+08
 Norm of input parameter rhc - 1: 9.6404864228226437e+00
 Norm of input parameter rhc - 2: 9.6404864228226437e+00
-Norm of input parameter rv - 1: 2.0602447479815532e+02
-Norm of input parameter rv - 2: 5.3394022282251328e+01
+Norm of input parameter rv - 1: 2.0602447479815535e+02
+Norm of input parameter rv - 2: 5.3394022282251335e+01
              Effective levels:   1   2 
                 Horizontal support radius:    4041.73 km (   3452.48 km -    4415.02 km)
                 Estimated nc1 from horizontal support radius:      576
