--- /home/benjaminm/code/oops-bundle/ecsaber/saber/test/testref/error_covariance_training_bump_nicas_3.ref.tmp	2024-05-06 14:16:30.991699443 +0200
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/test/testref/error_covariance_training_bump_nicas_3.ref	2024-05-06 13:02:10.315670970 +0200
@@ -7,9 +7,9 @@
              Level   2 ~> 100.0%
              Level   1 ~>  0.100E+01 vert. coord.
              Level   2 ~>  0.200E+01 vert. coord.
-Norm of input parameter nicas_a - 1: 4.3839794469441046e+01
-Norm of input parameter nicas_a - 2: 1.4136489898515414e+01
-Norm of input parameter nicas_norm - 1: 5.9379940338858695e+01
+Norm of input parameter nicas_a - 1: 4.3839794469441969e+01
+Norm of input parameter nicas_a - 2: 1.4136489898515467e+01
+Norm of input parameter nicas_norm - 1: 5.9379940338858738e+01
              nc1(  1) =                        567
              nc1(  2) =                        515
              ns =                             1082
