--- /home/benjaminm/code/oops-bundle/ecsaber/saber/test/testref/error_covariance_training_bump_stddev_1.ref.tmp	2024-05-22 11:25:08.306619734 +0200
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/test/testref/error_covariance_training_bump_stddev_1.ref	2024-05-22 10:50:49.943380011 +0200
@@ -1,6 +1,6 @@
        Independent levels:  1[2]
           Subset Sc0 size:              762
-          Min. / max. longitudes: -171.0 /   180.0
+          Min. / max. longitudes: -180.0 /   180.0
           Min. / max. latitudes:   -90.0 /    90.0
           Domain area (% of Earth area): 0.100E+03%
              Level   1 ~> 100.0%
@@ -31,6 +31,6 @@
                 Level   2: rhflt =   40000.00 km, rel. diff. =  0.19375E-01
              Level   1 ~>  0.10169E+01 /  0.10169E+01
              Level   2 ~>  0.10195E+01 /  0.10195E+01
-Norm of output parameter stddev - 1: 4.1359270513803665e+01
-Norm of output parameter var - 1: 4.3289362578921313e+01
-Norm of output parameter m4 - 1: 1.3993375181143369e+02
+Norm of output parameter stddev - 1: 4.1359270513803672e+01
+Norm of output parameter var - 1: 4.3289362578921306e+01
+Norm of output parameter m4 - 1: 1.3993375181143315e+02
