--- /home/benjaminm/code/oops-bundle/ecsaber/saber/test/testref/error_covariance_training_bump_hdiag-nicas_1.ref.tmp	2024-05-06 14:16:31.203700016 +0200
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/test/testref/error_covariance_training_bump_hdiag-nicas_1.ref	2024-05-06 13:02:17.303691186 +0200
@@ -260,16 +260,16 @@
 Norm of output parameter dirac_mom_single_obs - 1: 9.4957515532060626e+00
 Norm of output parameter dirac_mom - 1: 1.1810192991624243e+01
 Norm of output parameter dirac_diag_cor - 1: 2.2185808499236241e+00
-Norm of output parameter cor_a - 1: 4.3839794469441046e+01
-Norm of output parameter cor_a - 2: 1.4136489898515414e+01
-Norm of output parameter cor_rh1 - 1: 1.9902399622449389e+08
-Norm of output parameter cor_rh1 - 2: 1.1191941778167167e+08
-Norm of output parameter cor_rh2 - 1: 2.9308846208048666e+08
-Norm of output parameter cor_rh2 - 2: 1.6481575416450593e+08
-Norm of output parameter cor_rhc - 1: 2.3896222621905441e+00
-Norm of output parameter cor_rhc - 2: 2.3896222621905441e+00
-Norm of output parameter cor_rh - 1: 2.4138376564442521e+08
-Norm of output parameter cor_rh - 2: 1.3574006665205994e+08
-Norm of output parameter cor_rv - 1: 2.3932521332512050e+02
-Norm of output parameter cor_rv - 2: 2.3932521332606790e+01
-Norm of output parameter nicas_norm - 1: 5.9379940338858695e+01
+Norm of output parameter cor_a - 1: 4.3839794469441969e+01
+Norm of output parameter cor_a - 2: 1.4136489898515467e+01
+Norm of output parameter cor_rh1 - 1: 1.9902399622449541e+08
+Norm of output parameter cor_rh1 - 2: 1.1191941778167243e+08
+Norm of output parameter cor_rh2 - 1: 2.9308846208048773e+08
+Norm of output parameter cor_rh2 - 2: 1.6481575416450891e+08
+Norm of output parameter cor_rhc - 1: 2.3896222621905969e+00
+Norm of output parameter cor_rhc - 2: 2.3896222621905969e+00
+Norm of output parameter cor_rh - 1: 2.4138376564442578e+08
+Norm of output parameter cor_rh - 2: 1.3574006665205804e+08
+Norm of output parameter cor_rv - 1: 2.3932521332511897e+02
+Norm of output parameter cor_rv - 2: 2.3932521332606786e+01
+Norm of output parameter nicas_norm - 1: 5.9379940338858738e+01
