--- /home/benjaminm/code/oops-bundle/ecsaber/saber/test/testref/error_covariance_training_bump_hdiag-nicas_5.ref.tmp	2024-05-22 11:25:08.882620738 +0200
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/test/testref/error_covariance_training_bump_hdiag-nicas_5.ref	2024-05-22 10:51:14.215423622 +0200
@@ -1,6 +1,6 @@
        Independent levels:  1[2]
           Subset Sc0 size:              173
-          Min. / max. longitudes: -167.7 /   180.0
+          Min. / max. longitudes: -180.0 /   180.0
           Min. / max. latitudes:   -90.0 /    90.0
           Domain area (% of Earth area): 0.100E+03%
              Level   1 ~> 100.0%
@@ -88,55 +88,55 @@
           Independent levels for angular sector   4 / class  10:  1[2]
                 Level:   1 ~> cov. at class zero:  0.97E+00
                 Level:   2 ~> cov. at class zero:  0.96E+00
-             Block stream_function:  0.18702613E+00 for      144 diagnostic points
+             Block stream_function:  0.19261743E+00 for      144 diagnostic points
                    Level:   1 ~> amplitude:        0.58
-                                 cor. hor. support:      4742.88 km
-                                 cor. ver. support:         7.90 vertical units
-                   Level:   2 ~> amplitude:        0.62
-                                 cor. hor. support:      4428.39 km
-                                 cor. ver. support:         6.54 vertical units
+                                 cor. hor. support:      4827.95 km
+                                 cor. ver. support:         8.17 vertical units
+                   Level:   2 ~> amplitude:        0.61
+                                 cor. hor. support:      4521.23 km
+                                 cor. ver. support:         7.14 vertical units
                    Level:   1 ~> amplitude:        0.42
-                                 cor. hor. support:      2667.12 km
-                                 cor. ver. support:         0.79 vertical units
-                   Level:   2 ~> amplitude:        0.38
-                                 cor. hor. support:      2490.27 km
-                                 cor. ver. support:         0.65 vertical units
+                                 cor. hor. support:      2714.96 km
+                                 cor. ver. support:         0.82 vertical units
+                   Level:   2 ~> amplitude:        0.39
+                                 cor. hor. support:      2542.47 km
+                                 cor. ver. support:         0.71 vertical units
              Effective levels:   1   2 
-                Horizontal support radius:    4742.88 km (   4742.88 km -    4742.88 km)
-                Estimated nc1 from horizontal support radius:      418
+                Horizontal support radius:    4827.95 km (   4827.95 km -    4827.95 km)
+                Estimated nc1 from horizontal support radius:      404
                    Decimate full grid, at least      173 points required,     173 valid points found
                    Subgrid hash:   1239477474
                 Final nc1:      173
-                Effective horizontal resolution:  2.57
-                Horizontal support radius:    4428.39 km (   4428.39 km -    4428.39 km)
-                Estimated nc1 from horizontal support radius:      480
+                Effective horizontal resolution:  2.62
+                Horizontal support radius:    4521.23 km (   4521.23 km -    4521.23 km)
+                Estimated nc1 from horizontal support radius:      461
                    Decimate full grid, at least      173 points required,     173 valid points found
                    Subgrid hash:   1239477474
                 Final nc1:      173
-                Effective horizontal resolution:  2.40
+                Effective horizontal resolution:  2.45
              nc1(  1) =                        173
              nc1(  2) =                        173
              ns =                              346
              v%n_s =                             2
-             c%n_s[global] =                  4253
+             c%n_s[global] =                  4337
              Effective levels:   1   2 
-                Horizontal support radius:    2667.12 km (   2667.12 km -    2667.12 km)
-                Estimated nc1 from horizontal support radius:     1324
+                Horizontal support radius:    2714.96 km (   2714.96 km -    2714.96 km)
+                Estimated nc1 from horizontal support radius:     1278
                    Decimate full grid, at least      173 points required,     173 valid points found
                    Subgrid hash:   1239477474
                 Final nc1:      173
-                Effective horizontal resolution:  1.45
-                Horizontal support radius:    2490.27 km (   2490.27 km -    2490.27 km)
-                Estimated nc1 from horizontal support radius:     1519
+                Effective horizontal resolution:  1.47
+                Horizontal support radius:    2542.47 km (   2542.47 km -    2542.47 km)
+                Estimated nc1 from horizontal support radius:     1457
                    Decimate full grid, at least      173 points required,     173 valid points found
                    Subgrid hash:   1239477474
                 Final nc1:      173
-                Effective horizontal resolution:  1.35
+                Effective horizontal resolution:  1.38
              nc1(  1) =                        173
              nc1(  2) =                        173
              ns =                              346
              v%n_s =                             2
-             c%n_s[global] =                   385
+             c%n_s[global] =                   399
     Full NICAS adjoint test result:                        T
        Block NICAS adjoint test result:                    T
           Interpolation adjoint test (horizontal) result:  T
@@ -158,19 +158,19 @@
              Min / max: 1.0000000 /  1.0000000 over     10 tests
              0.0 /    0.0:  1.0000000
           Level   1:  0.0000000 -  1.0000000
-          Level   2:  0.0000000 -  0.5687413
-Norm of output parameter dirac_mom - 1: 4.5847594118047885e+00
-Norm of output parameter dirac_diag_cor - 1: 1.4049010083234903e+00
-Norm of output parameter cor_a - 1: 1.1193645188076077e+01
-Norm of output parameter cor_a - 2: 7.4246181896717696e+00
-Norm of output parameter cor_rh1 - 1: 7.3652032943927839e+07
-Norm of output parameter cor_rh1 - 2: 4.1417581808692828e+07
-Norm of output parameter cor_rh2 - 1: 9.9135796551314577e+07
-Norm of output parameter cor_rh2 - 2: 5.5748155206522353e+07
-Norm of output parameter cor_rhc - 1: 1.2373360854270123e+00
-Norm of output parameter cor_rhc - 2: 1.2373360854270123e+00
-Norm of output parameter cor_rh - 1: 8.5347899082850680e+07
-Norm of output parameter cor_rh - 2: 4.7994650672510058e+07
-Norm of output parameter cor_rv - 1: 1.3487852705213120e+02
-Norm of output parameter cor_rv - 2: 1.3487852705266574e+01
+          Level   2:  0.0000000 -  0.5641340
+Norm of output parameter dirac_mom - 1: 4.5847594118047867e+00
+Norm of output parameter dirac_diag_cor - 1: 1.4180657532595164e+00
+Norm of output parameter cor_a - 1: 1.1013230643696351e+01
+Norm of output parameter cor_a - 2: 7.5974522080458646e+00
+Norm of output parameter cor_rh1 - 1: 7.5360157820866063e+07
+Norm of output parameter cor_rh1 - 2: 4.2378131015581556e+07
+Norm of output parameter cor_rh2 - 1: 1.0108144699950317e+08
+Norm of output parameter cor_rh2 - 2: 5.6842274857914969e+07
+Norm of output parameter cor_rhc - 1: 2.0496983542874103e+00
+Norm of output parameter cor_rhc - 2: 2.0496983542874103e+00
+Norm of output parameter cor_rh - 1: 8.6999193669104576e+07
+Norm of output parameter cor_rh - 2: 4.8923241858424522e+07
+Norm of output parameter cor_rv - 1: 1.4278561234113781e+02
+Norm of output parameter cor_rv - 2: 1.4278561234170407e+01
 Norm of output parameter nicas_norm - 1: 1.8601075237738275e+01
