--- /home/benjaminm/code/oops-bundle/ecsaber/saber/test/testref/error_covariance_training_bump_nicas_5.ref.tmp	2024-06-12 18:33:08.430324341 +0200
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/test/testref/error_covariance_training_bump_nicas_5.ref	2024-06-11 11:30:59.454152990 +0200
@@ -1,22 +1,22 @@
        Independent levels:  1[2]
           Subset Sc0 size:              875
-          Domain area (% of Earth area): 0.616E+01%
+          Domain area (% of Earth area): 0.591E+01%
              Level   1 ~> 100.0%
              Level   2 ~> 100.0%
              Level   1 ~>  0.100E+01 vert. coord.
              Level   2 ~>  0.200E+01 vert. coord.
              Effective levels:   1   2 
                 Horizontal support radius:    1000.00 km (   1000.00 km -    1000.00 km)
-                Estimated nc1 from horizontal support radius:      581
-                   Decimate full grid, at least      581 points required,     875 valid points found
-                   Subgrid hash:   -540767041
-                Final nc1:      581
+                Estimated nc1 from horizontal support radius:      556
+                   Decimate full grid, at least      556 points required,     848 valid points found
+                   Subgrid hash:   -142159505
+                Final nc1:      556
                 Effective horizontal resolution:  4.00
-             nc1(  1) =                        581
-             nc1(  2) =                        581
-             ns =                             1162
+             nc1(  1) =                        556
+             nc1(  2) =                        556
+             ns =                             1112
              v%n_s =                             2
-             c%n_s[global] =                 33596
+             c%n_s[global] =                 39528
     Full NICAS adjoint test result:                        T
        Block NICAS adjoint test result:                    T
           Interpolation adjoint test (horizontal) result:  T
