--- /home/benjaminm/code/oops-bundle/ecsaber/saber/test/testref/error_covariance_training_bump_nicas_2.ref.tmp	2024-05-22 11:25:09.018620976 +0200
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/test/testref/error_covariance_training_bump_nicas_2.ref	2024-05-22 10:51:19.731433760 +0200
@@ -1,34 +1,34 @@
        Independent levels:  1[2]
           Subset Sc0 size:              762
-          Min. / max. longitudes: -171.0 /   180.0
+          Min. / max. longitudes: -180.0 /   180.0
           Min. / max. latitudes:   -90.0 /    90.0
           Domain area (% of Earth area): 0.100E+03%
              Level   1 ~> 100.0%
              Level   2 ~> 100.0%
              Level   1 ~>  0.100E+01 vert. coord.
              Level   2 ~>  0.200E+01 vert. coord.
-Norm of input parameter nicas_a - 1: 4.3839794469441046e+01
-Norm of input parameter nicas_a - 2: 1.4136489898515414e+01
-             nc1(  1) =                        567
-             nc1(  2) =                        515
-             ns =                             1082
+Norm of input parameter nicas_a - 1: 4.2336977193041456e+01
+Norm of input parameter nicas_a - 2: 1.5657837636639481e+01
+             nc1(  1) =                        553
+             nc1(  2) =                        500
+             ns =                             1053
              v%n_s =                             2
-             c%n_s[global] =                 30955
+             c%n_s[global] =                 29826
              nc1(  1) =                        762
              nc1(  2) =                        762
              ns =                             1524
              v%n_s =                             2
-             c%n_s[global] =                 13158
-             nc1(  1) =                        530
-             nc1(  2) =                        563
-             ns =                             1093
+             c%n_s[global] =                 13960
+             nc1(  1) =                        542
+             nc1(  2) =                        565
+             ns =                             1107
              v%n_s =                             2
-             c%n_s[global] =                 30912
+             c%n_s[global] =                 30525
              nc1(  1) =                        762
              nc1(  2) =                        762
              ns =                             1524
              v%n_s =                             2
-             c%n_s[global] =                 13478
+             c%n_s[global] =                 13718
     Full NICAS adjoint test result:                        T
        Block NICAS adjoint test result:                    T
           Interpolation adjoint test (horizontal) result:  T
@@ -62,8 +62,8 @@
           Component NICAS adjoint test result:             T
              0.0 /    0.0:  1.0000000
           Level   1:  0.0000000 -  1.0000000
-          Level   2:  0.0000000 -  0.6163033
+          Level   2:  0.0000000 -  0.6334295
              0.0 /    0.0:  0.0000000
           Level   1:  0.0000000 -  0.0000000
           Level   2:  0.0000000 -  0.0000000
-Norm of output parameter dirac_nicas - 1: 2.2529052849239979e+00
+Norm of output parameter dirac_nicas - 1: 2.2516673611810836e+00
