--- /home/benjaminm/code/oops-bundle/ecsaber/saber/test/testref/error_covariance_training_bump_nicas_3.ref.tmp	2024-05-22 11:25:08.334619782 +0200
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/test/testref/error_covariance_training_bump_nicas_3.ref	2024-05-22 10:50:51.035381938 +0200
@@ -1,35 +1,37 @@
        Independent levels:  1[2]
           Subset Sc0 size:              762
-          Min. / max. longitudes: -171.0 /   180.0
+          Min. / max. longitudes: -180.0 /   180.0
           Min. / max. latitudes:   -90.0 /    90.0
           Domain area (% of Earth area): 0.100E+03%
              Level   1 ~> 100.0%
              Level   2 ~> 100.0%
              Level   1 ~>  0.100E+01 vert. coord.
              Level   2 ~>  0.200E+01 vert. coord.
-Norm of input parameter nicas_a - 1: 4.3839794469441046e+01
-Norm of input parameter nicas_a - 2: 1.4136489898515414e+01
-Norm of input parameter nicas_norm - 1: 5.9379940338858695e+01
-             nc1(  1) =                        567
-             nc1(  2) =                        515
-             ns =                             1082
+Norm of input parameter nicas_a - 1: 4.2336977193041456e+01
+Norm of input parameter nicas_a - 2: 1.5657837636639481e+01
+Norm of input parameter rh - 1: 2.4235287450180620e+08
+Norm of input parameter rh - 2: 1.3628503661116999e+08
+Norm of input parameter nicas_norm - 1: 5.9399923574089613e+01
+             nc1(  1) =                        553
+             nc1(  2) =                        500
+             ns =                             1053
              v%n_s =                             2
-             c%n_s[global] =                 30955
+             c%n_s[global] =                 29826
              nc1(  1) =                        762
              nc1(  2) =                        762
              ns =                             1524
              v%n_s =                             2
-             c%n_s[global] =                 13158
-             nc1(  1) =                        530
-             nc1(  2) =                        563
-             ns =                             1093
+             c%n_s[global] =                 13960
+             nc1(  1) =                        542
+             nc1(  2) =                        565
+             ns =                             1107
              v%n_s =                             2
-             c%n_s[global] =                 30912
+             c%n_s[global] =                 30525
              nc1(  1) =                        762
              nc1(  2) =                        762
              ns =                             1524
              v%n_s =                             2
-             c%n_s[global] =                 13478
+             c%n_s[global] =                 13718
     Full NICAS adjoint test result:                        T
        Block NICAS adjoint test result:                    T
           Interpolation adjoint test (horizontal) result:  T
