--- /home/benjaminm/code/oops-bundle/ecsaber/saber/test/testref/error_covariance_training_bump_hdiag_7.ref.tmp	2024-05-06 14:16:31.539700925 +0200
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/test/testref/error_covariance_training_bump_hdiag_7.ref	2024-05-06 13:02:28.755724211 +0200
@@ -95,7 +95,7 @@
                 Level:   1 ~> hybrid coefficients: 0.56 / 0.42
                 Level:   2 ~> hybrid coefficients: 0.53 / 0.43
 Norm of output parameter loc_rh - 1: 2.1711165838792601e+08
-Norm of output parameter loc_rv - 1: 2.0044142458539639e+02
+Norm of output parameter loc_rv - 1: 2.0044142458539631e+02
 Norm of output parameter hyb_coef_ens - 1: 2.6732620444699819e+01
 Norm of output parameter hyb_coef_ens_lr - 1: 2.0822510714431008e+01
 Norm of output parameter loc_rh_lr - 1: 2.3694160476838171e+08
