--- /home/benjaminm/code/oops-bundle/ecsaber/saber/test/testref/error_covariance_training_bump_hdiag_1.ref.tmp	2024-05-06 14:16:30.791698902 +0200
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/test/testref/error_covariance_training_bump_hdiag_1.ref	2024-05-06 13:02:02.035646948 +0200
@@ -185,10 +185,10 @@
 Norm of output parameter cor_rh1 - 1: 2.6600928484758756e+08
 Norm of output parameter cor_rh1 - 2: 2.1469901889690262e+08
 Norm of output parameter cor_rh2 - 1: 2.8142686892415738e+08
-Norm of output parameter cor_rh2 - 2: 2.3890998875146008e+08
+Norm of output parameter cor_rh2 - 2: 2.3890998875146011e+08
 Norm of output parameter cor_rhc - 1: 9.6404864228226437e+00
 Norm of output parameter cor_rhc - 2: 9.6404864228226437e+00
 Norm of output parameter cor_rh - 1: 2.7111930440737945e+08
 Norm of output parameter cor_rh - 2: 2.2453671134278226e+08
-Norm of output parameter cor_rv - 1: 2.0602447479815532e+02
-Norm of output parameter cor_rv - 2: 5.3394022282251328e+01
+Norm of output parameter cor_rv - 1: 2.0602447479815535e+02
+Norm of output parameter cor_rv - 2: 5.3394022282251335e+01
