--- /home/benjaminm/code/oops-bundle/ecsaber/saber/test/testref/error_covariance_training_bump_nicas_6.ref.tmp	2024-02-14 13:20:33.203916154 +0100
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/test/testref/error_covariance_training_bump_nicas_6.ref	2024-02-13 13:47:13.442181683 +0100
@@ -14,16 +14,16 @@
           Domain area (% of Earth area): 0.100E+03%
              Level   1 ~> 100.0%
              Level   1 ~>  0.100E+01 vert. coord.
-Norm of input parameter a - 1: 5.7794891779592362e+01
-Norm of input parameter a - 2: 1.5961224548534769e+01
-Norm of input parameter rh1 - 1: 2.5693529569307017e+08
-Norm of input parameter rh1 - 2: 2.1161007184646529e+08
-Norm of input parameter rh2 - 1: 2.7200154827823085e+08
-Norm of input parameter rh2 - 2: 2.3197434612096420e+08
-Norm of input parameter rhc - 1: 9.6404864228226383e+00
-Norm of input parameter rhc - 2: 9.6404864228226383e+00
-Norm of input parameter rv - 1: 1.9343010342287178e+02
-Norm of input parameter rv - 2: 5.3913950859878383e+01
+Norm of input parameter a - 1: 5.1143329023232248e+01
+Norm of input parameter a - 2: 1.4094159742674634e+01
+Norm of input parameter rh1 - 1: 2.5693529569307232e+08
+Norm of input parameter rh1 - 2: 2.1161007184647402e+08
+Norm of input parameter rh2 - 1: 2.7200154827823156e+08
+Norm of input parameter rh2 - 2: 2.3197434612096655e+08
+Norm of input parameter rhc - 1: 9.6404864228226437e+00
+Norm of input parameter rhc - 2: 9.6404864228226437e+00
+Norm of input parameter rv - 1: 1.9343010342287204e+02
+Norm of input parameter rv - 2: 5.3913950859878355e+01
              Effective levels:   1   2 
                 Horizontal support radius:    3986.64 km (   3408.50 km -    4365.10 km)
                 Estimated nc1 from horizontal support radius:      592
@@ -127,9 +127,9 @@
           Communication AC adjoint test result:            T
           Convolution / communication adjoint test result: T
           Component NICAS adjoint test result:             T
-             0.0 /    0.0:  1.1284765
-          Level   1:  0.0000000 -  1.1284765
-          Level   2:  0.0000000 -  0.6769744
+             0.0 /    0.0:  1.0000000
+          Level   1:  0.0000000 -  1.0000000
+          Level   2:  0.0000000 -  0.5972193
              0.0 /    0.0:  0.0000000
           Level   1:  0.0000000 -  0.0000000
           Level   2:  0.0000000 -  0.0000000
@@ -171,5 +171,5 @@
           Communication AC adjoint test result:            T
           Convolution / communication adjoint test result: T
           Component NICAS adjoint test result:             T
-             0.0 /    0.0:  1.3672066
-          Level   1:  0.0000000 -  1.3672066
+             0.0 /    0.0:  1.0000000
+          Level   1:  0.0000000 -  1.0000000
