--- /home/benjaminm/code/oops-bundle/ecsaber/saber/test/testref/error_covariance_training_bump_nicas_5.ref.tmp	2024-05-06 14:16:31.319700330 +0200
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/test/testref/error_covariance_training_bump_nicas_5.ref	2024-05-06 13:02:21.523703371 +0200
@@ -14,17 +14,11 @@
                    Subgrid hash:   -540767041
                 Final nc1:      581
                 Effective horizontal resolution:  4.00
-                Horizontal support radius:    1000.00 km (   1000.00 km -    1000.00 km)
-                Estimated nc1 from horizontal support radius:      581
-                   Decimate full grid, at least      581 points required,     875 valid points found
-                   Subgrid hash:   -873112754
-                Final nc1:      581
-                Effective horizontal resolution:  4.00
              nc1(  1) =                        581
              nc1(  2) =                        581
              ns =                             1162
              v%n_s =                             2
-             c%n_s[global] =                 33490
+             c%n_s[global] =                 33596
     Full NICAS adjoint test result:                        T
        Block NICAS adjoint test result:                    T
           Interpolation adjoint test (horizontal) result:  T
@@ -36,4 +30,4 @@
           Component NICAS adjoint test result:             T
            149.4 /  -30.0:  1.0000000
           Level   1:  0.0000000 -  1.0000000
-          Level   2:  0.0000000 -  0.5688720
+          Level   2:  0.0000000 -  0.6000000
