--- /home/benjaminm/code/oops-bundle/ecsaber/saber/test/testref/error_covariance_training_bump_nicas_5.ref.tmp	2024-05-22 11:25:08.750620508 +0200
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/test/testref/error_covariance_training_bump_nicas_5.ref	2024-05-22 10:51:08.267412785 +0200
@@ -2,29 +2,23 @@
           Subset Sc0 size:              875
           Min. / max. longitudes:  100.0 /   170.0
           Min. / max. latitudes:   -50.0 /   -10.0
-          Domain area (% of Earth area): 0.616E+01%
+          Domain area (% of Earth area): 0.591E+01%
              Level   1 ~> 100.0%
              Level   2 ~> 100.0%
              Level   1 ~>  0.100E+01 vert. coord.
              Level   2 ~>  0.200E+01 vert. coord.
              Effective levels:   1   2 
                 Horizontal support radius:    1000.00 km (   1000.00 km -    1000.00 km)
-                Estimated nc1 from horizontal support radius:      581
-                   Decimate full grid, at least      581 points required,     875 valid points found
-                   Subgrid hash:   -540767041
-                Final nc1:      581
+                Estimated nc1 from horizontal support radius:      556
+                   Decimate full grid, at least      556 points required,     848 valid points found
+                   Subgrid hash:   1333588489
+                Final nc1:      556
                 Effective horizontal resolution:  4.00
-                Horizontal support radius:    1000.00 km (   1000.00 km -    1000.00 km)
-                Estimated nc1 from horizontal support radius:      581
-                   Decimate full grid, at least      581 points required,     875 valid points found
-                   Subgrid hash:   -873112754
-                Final nc1:      581
-                Effective horizontal resolution:  4.00
-             nc1(  1) =                        581
-             nc1(  2) =                        581
-             ns =                             1162
+             nc1(  1) =                        556
+             nc1(  2) =                        556
+             ns =                             1112
              v%n_s =                             2
-             c%n_s[global] =                 33490
+             c%n_s[global] =                 31444
     Full NICAS adjoint test result:                        T
        Block NICAS adjoint test result:                    T
           Interpolation adjoint test (horizontal) result:  T
@@ -36,4 +30,4 @@
           Component NICAS adjoint test result:             T
            149.4 /  -30.0:  1.0000000
           Level   1:  0.0000000 -  1.0000000
-          Level   2:  0.0000000 -  0.5688720
+          Level   2:  0.0000000 -  0.6000000
