--- /home/benjaminm/code/oops-bundle/ecsaber/saber/test/testref/error_covariance_training_bump_hdiag_9.ref.tmp	2024-05-06 14:16:30.927699270 +0200
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/test/testref/error_covariance_training_bump_hdiag_9.ref	2024-05-06 13:02:08.127664629 +0200
@@ -120,5 +120,5 @@
                 Level:   2 ~> hybrid coefficients: 0.31 / 0.64
 Norm of output parameter loc_rh - 1: 4.8989794855663562e+08
 Norm of output parameter loc_rv - 1: 9.7979589711327122e+01
-Norm of output parameter hyb_coef_ens - 1: 1.8058337359365844e+01
+Norm of output parameter hyb_coef_ens - 1: 1.8058337359365840e+01
 Norm of output parameter hyb_coef_sta - 1: 2.3721874592371822e+01
