--- /home/benjaminm/code/oops-bundle/ecsaber/saber/test/testref/error_covariance_training_bump_hdiag_1.ref.tmp	2024-05-22 11:25:08.054619294 +0200
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/test/testref/error_covariance_training_bump_hdiag_1.ref	2024-05-22 10:50:41.519365255 +0200
@@ -1,6 +1,6 @@
        Independent levels:  1[2]
           Subset Sc0 size:              762
-          Min. / max. longitudes: -171.0 /   180.0
+          Min. / max. longitudes: -180.0 /   180.0
           Min. / max. latitudes:   -90.0 /    90.0
           Domain area (% of Earth area): 0.100E+03%
              Level   1 ~> 100.0%
@@ -147,48 +147,48 @@
                 Level:   1 ~> cov. at class zero:  0.98E+00
                 Level:   2 ~> cov. at class zero:  0.10E+01
                 Level:   1 ~> cov. at class zero:  0.97E+00
-             Block stream_function:  0.35263264E+00 for    37356 diagnostic points
-             Block velocity_potential:  0.35667579E+00 for    37356 diagnostic points
-             Block surface_pressure:  0.35321006E+00 for     9339 diagnostic points
+             Block stream_function:  0.35220293E+00 for    37356 diagnostic points
+             Block velocity_potential:  0.35630961E+00 for    37356 diagnostic points
+             Block surface_pressure:  0.35343236E+00 for     9339 diagnostic points
+                   Level:   1 ~> amplitude:        0.72
+                                 cor. hor. support:      4128.04 km
+                                 cor. ver. support:         4.94 vertical units
+                   Level:   2 ~> amplitude:        0.71
+                                 cor. hor. support:      4339.49 km
+                                 cor. ver. support:         5.19 vertical units
+                   Level:   1 ~> amplitude:        0.28
+                                 cor. hor. support:      2321.37 km
+                                 cor. ver. support:         0.49 vertical units
+                   Level:   2 ~> amplitude:        0.29
+                                 cor. hor. support:      2440.28 km
+                                 cor. ver. support:         0.52 vertical units
                    Level:   1 ~> amplitude:        0.75
-                                 cor. hor. support:      4076.21 km
-                                 cor. ver. support:         4.12 vertical units
+                                 cor. hor. support:      4169.66 km
+                                 cor. ver. support:         4.23 vertical units
                    Level:   2 ~> amplitude:        0.74
-                                 cor. hor. support:      4274.19 km
-                                 cor. ver. support:         4.36 vertical units
+                                 cor. hor. support:      4082.23 km
+                                 cor. ver. support:         4.65 vertical units
                    Level:   1 ~> amplitude:        0.25
-                                 cor. hor. support:      2292.22 km
-                                 cor. ver. support:         0.41 vertical units
+                                 cor. hor. support:      2344.77 km
+                                 cor. ver. support:         0.42 vertical units
                    Level:   2 ~> amplitude:        0.26
-                                 cor. hor. support:      2403.55 km
-                                 cor. ver. support:         0.44 vertical units
-                   Level:   1 ~> amplitude:        0.77
-                                 cor. hor. support:      4213.17 km
-                                 cor. ver. support:         3.92 vertical units
-                   Level:   2 ~> amplitude:        0.76
-                                 cor. hor. support:      4090.16 km
-                                 cor. ver. support:         4.10 vertical units
-                   Level:   1 ~> amplitude:        0.23
-                                 cor. hor. support:      2369.24 km
-                                 cor. ver. support:         0.39 vertical units
-                   Level:   2 ~> amplitude:        0.24
-                                 cor. hor. support:      2300.07 km
-                                 cor. ver. support:         0.41 vertical units
-                   Level:   1 ~> amplitude:        0.73
-                                 cor. hor. support:      3939.75 km
+                                 cor. hor. support:      2295.61 km
+                                 cor. ver. support:         0.47 vertical units
+                   Level:   1 ~> amplitude:        0.71
+                                 cor. hor. support:      4020.80 km
                                  cor. ver. support:         0.00 vertical units
-                   Level:   1 ~> amplitude:        0.27
-                                 cor. hor. support:      2215.49 km
+                   Level:   1 ~> amplitude:        0.29
+                                 cor. hor. support:      2261.06 km
                                  cor. ver. support:         0.00 vertical units
-Norm of output parameter cor_a - 1: 5.0419080737424800e+01
-Norm of output parameter cor_a - 2: 1.4831352041415670e+01
-Norm of output parameter cor_rh1 - 1: 2.6600928484758756e+08
-Norm of output parameter cor_rh1 - 2: 2.1469901889690262e+08
-Norm of output parameter cor_rh2 - 1: 2.8142686892415738e+08
-Norm of output parameter cor_rh2 - 2: 2.3890998875146008e+08
-Norm of output parameter cor_rhc - 1: 9.6404864228226437e+00
-Norm of output parameter cor_rhc - 2: 9.6404864228226437e+00
-Norm of output parameter cor_rh - 1: 2.7111930440737945e+08
-Norm of output parameter cor_rh - 2: 2.2453671134278226e+08
-Norm of output parameter cor_rv - 1: 2.0602447479815532e+02
-Norm of output parameter cor_rv - 2: 5.3394022282251328e+01
+Norm of output parameter cor_a - 1: 5.0397044108460548e+01
+Norm of output parameter cor_a - 2: 1.4827377479622088e+01
+Norm of output parameter cor_rh1 - 1: 2.6085575211410484e+08
+Norm of output parameter cor_rh1 - 2: 2.1722208650635940e+08
+Norm of output parameter cor_rh2 - 1: 2.8562552415214670e+08
+Norm of output parameter cor_rh2 - 2: 2.3650597200702778e+08
+Norm of output parameter cor_rhc - 1: 8.2572773007379308e+00
+Norm of output parameter cor_rhc - 2: 8.2572773007379308e+00
+Norm of output parameter cor_rh - 1: 2.7058418651145124e+08
+Norm of output parameter cor_rh - 2: 2.2495411957443380e+08
+Norm of output parameter cor_rv - 1: 2.0695580221409611e+02
+Norm of output parameter cor_rv - 2: 6.3556661870891986e+01
