--- /home/benjaminm/code/oops-bundle/ecsaber/saber/test/testref/error_covariance_training_bump_hdiag_3.ref.tmp	2024-05-22 11:25:08.174619503 +0200
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/test/testref/error_covariance_training_bump_hdiag_3.ref	2024-05-22 10:50:45.615372404 +0200
@@ -1,6 +1,6 @@
        Independent levels:  1[2]
           Subset Sc0 size:              762
-          Min. / max. longitudes: -171.0 /   180.0
+          Min. / max. longitudes: -180.0 /   180.0
           Min. / max. latitudes:   -90.0 /    90.0
           Domain area (% of Earth area): 0.100E+03%
              Level   1 ~> 100.0%
@@ -64,43 +64,43 @@
           Independent levels for angular sector   4 / class  10:  1[2]
                 Level:   1 ~> cov. at class zero:  0.99E+00
                 Level:   2 ~> cov. at class zero:  0.10E+01
-             Block common:  0.19480025E+00 for    38768 diagnostic points
-                   Level:   1 ~> amplitude:        0.76
-                                 cor. hor. support:      4141.73 km
-                                 cor. ver. support:         3.98 vertical units
-                   Level:   2 ~> amplitude:        0.75
-                                 cor. hor. support:      4176.13 km
-                                 cor. ver. support:         4.20 vertical units
-                   Level:   1 ~> amplitude:        0.24
-                                 cor. hor. support:      2329.06 km
-                                 cor. ver. support:         0.40 vertical units
-                   Level:   2 ~> amplitude:        0.25
-                                 cor. hor. support:      2348.41 km
-                                 cor. ver. support:         0.42 vertical units
-             Block common:  0.33219410E+00 for    32873 diagnostic points
+             Block common:  0.19235575E+00 for    38768 diagnostic points
+                   Level:   1 ~> amplitude:        0.73
+                                 cor. hor. support:      4156.82 km
+                                 cor. ver. support:         4.60 vertical units
+                   Level:   2 ~> amplitude:        0.72
+                                 cor. hor. support:      4216.19 km
+                                 cor. ver. support:         4.93 vertical units
+                   Level:   1 ~> amplitude:        0.27
+                                 cor. hor. support:      2337.55 km
+                                 cor. ver. support:         0.46 vertical units
+                   Level:   2 ~> amplitude:        0.28
+                                 cor. hor. support:      2370.94 km
+                                 cor. ver. support:         0.49 vertical units
+             Block common:  0.33185664E+00 for    32283 diagnostic points
                 Level:   1 ~> loc. at class zero:       0.92
                 Level:   2 ~> loc. at class zero:       0.92
                    Level:   1 ~> amplitude:        1.00
-                                 loc. hor. support:      6164.52 km
+                                 loc. hor. support:      6074.23 km
                                  loc. ver. support:         6.06 vertical units
                    Level:   2 ~> amplitude:        1.00
-                                 loc. hor. support:      6198.03 km
+                                 loc. hor. support:      6295.38 km
                                  loc. ver. support:         6.06 vertical units
                    Level:   1 ~> amplitude:        0.00
-                                 loc. hor. support:      6164.52 km
+                                 loc. hor. support:      6074.23 km
                                  loc. ver. support:         6.06 vertical units
                    Level:   2 ~> amplitude:        0.00
-                                 loc. hor. support:      6198.03 km
+                                 loc. hor. support:      6295.38 km
                                  loc. ver. support:         6.06 vertical units
 Norm of output parameter loc_a - 1: 5.7965506984757752e+01
 Norm of output parameter loc_a - 2: 0.0000000000000000e+00
-Norm of output parameter loc_rh1 - 1: 3.8073581413247824e+08
-Norm of output parameter loc_rh1 - 2: 3.6301623352715296e+08
-Norm of output parameter loc_rh2 - 1: 4.3465691104227388e+08
-Norm of output parameter loc_rh2 - 2: 4.2140881693857831e+08
-Norm of output parameter loc_rhc - 1: 9.9968465768626782e+00
-Norm of output parameter loc_rhc - 2: 9.9968465768626782e+00
-Norm of output parameter loc_rh - 1: 3.9645107749844116e+08
-Norm of output parameter loc_rh - 2: 3.8151512149124235e+08
-Norm of output parameter loc_rv - 1: 3.6409176397622628e+02
-Norm of output parameter loc_rv - 2: 3.2962218254258033e+02
+Norm of output parameter loc_rh1 - 1: 3.6658526146261334e+08
+Norm of output parameter loc_rh1 - 2: 3.5810781247868115e+08
+Norm of output parameter loc_rh2 - 1: 4.4559965204550660e+08
+Norm of output parameter loc_rh2 - 2: 4.2599075859478521e+08
+Norm of output parameter loc_rhc - 1: 1.2106079285509049e+01
+Norm of output parameter loc_rhc - 2: 1.2106079285509049e+01
+Norm of output parameter loc_rh - 1: 3.9508459207662690e+08
+Norm of output parameter loc_rh - 2: 3.8062456526007646e+08
+Norm of output parameter loc_rv - 1: 3.6134833792529798e+02
+Norm of output parameter loc_rv - 2: 3.2492230617556066e+02
