--- /home/benjaminm/code/oops-bundle/ecsaber/saber/test/testref/randomization_bump_nicas_L10L2_static.ref.tmp	2024-05-06 14:16:31.231700092 +0200
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/test/testref/randomization_bump_nicas_L10L2_static.ref	2024-05-06 13:02:18.663695115 +0200
@@ -14,12 +14,6 @@
                    Subgrid hash:    762717970
                 Final nc1:      192
                 Effective horizontal resolution:  4.00
-                Horizontal support radius:    7000.00 km (   7000.00 km -    7000.00 km)
-                Estimated nc1 from horizontal support radius:      192
-                   Decimate full grid, at least      192 points required,     714 valid points found
-                   Subgrid hash:    762717970
-                Final nc1:      192
-                Effective horizontal resolution:  4.00
              nc1(  1) =                        192
              nc1(  2) =                        192
              ns =                              384
@@ -96,7 +90,7 @@
   - vert_coord: [1.0000000000000000e+00,2.0000000000000000e+00]
   Mask size: 100%
 Fields:
-  stream_function: 4.2865203381967447e+01
+  stream_function: 4.2865203381967440e+01
 Member 4: 
 Valid time:2010-01-01T12:00:00Z
 Quench geometry grid:
@@ -132,7 +126,7 @@
   - vert_coord: [1.0000000000000000e+00,2.0000000000000000e+00]
   Mask size: 100%
 Fields:
-  stream_function: 3.9135987051411512e+01
+  stream_function: 3.9135987051411504e+01
 Member 6: 
 Valid time:2010-01-01T12:00:00Z
 Quench geometry grid:
@@ -168,7 +162,7 @@
   - vert_coord: [1.0000000000000000e+00,2.0000000000000000e+00]
   Mask size: 100%
 Fields:
-  stream_function: 3.9132613514828478e+01
+  stream_function: 3.9132613514828485e+01
 Member 8: 
 Valid time:2010-01-01T12:00:00Z
 Quench geometry grid:
@@ -204,7 +198,7 @@
   - vert_coord: [1.0000000000000000e+00,2.0000000000000000e+00]
   Mask size: 100%
 Fields:
-  stream_function: 4.5877298040484085e+01
+  stream_function: 4.5877298040484078e+01
 Member 10: 
 Valid time:2010-01-01T12:00:00Z
 Quench geometry grid:
@@ -222,7 +216,7 @@
   - vert_coord: [1.0000000000000000e+00,2.0000000000000000e+00]
   Mask size: 100%
 Fields:
-  stream_function: 4.3299147696760429e+01
+  stream_function: 4.3299147696760421e+01
 Member 11: 
 Valid time:2010-01-01T12:00:00Z
 Quench geometry grid:
@@ -330,7 +324,7 @@
   - vert_coord: [1.0000000000000000e+00,2.0000000000000000e+00]
   Mask size: 100%
 Fields:
-  stream_function: 4.0667742561449998e+01
+  stream_function: 4.0667742561449991e+01
 Member 17: 
 Valid time:2010-01-01T12:00:00Z
 Quench geometry grid:
@@ -474,4 +468,4 @@
   - vert_coord: [1.0000000000000000e+00,2.0000000000000000e+00]
   Mask size: 100%
 Fields:
-  stream_function: 4.5278953578556937e+01
+  stream_function: 4.5278953578556930e+01
