--- /home/benjaminm/code/oops-bundle/ecsaber/saber/test/testref/randomization_bump_nicas_lam_2.ref.tmp	2024-06-12 18:33:08.586324871 +0200
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/test/testref/randomization_bump_nicas_lam_2.ref	2024-06-11 11:31:04.150178968 +0200
@@ -22,8 +22,8 @@
   - vert_coord: [1.0000000000000000e+00]
   Mask size: 100%
 Fields:
-  stream_function: 1.9344707669529276e+02
-  velocity_potential: 2.3681673304282140e+02
+  stream_function: 2.0791736278848481e+02
+  velocity_potential: 2.8725955524461284e+02
   surface_pressure: 0.0000000000000000e+00
 Member 1: 
 Valid time:2010-01-01T12:00:00Z
@@ -48,6 +48,6 @@
   - vert_coord: [1.0000000000000000e+00]
   Mask size: 100%
 Fields:
-  stream_function: 2.0629494512491718e+02
-  velocity_potential: 1.7272560666356097e+02
+  stream_function: 2.3132473129358473e+02
+  velocity_potential: 1.4463267383738523e+02
   surface_pressure: 0.0000000000000000e+00
