--- /home/benjaminm/code/oops-bundle/ecsaber/saber/test/testref/randomization_diffusion_2.ref.tmp	2024-06-12 18:33:08.138323349 +0200
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/test/testref/randomization_diffusion_2.ref	2024-06-03 11:15:00.201487181 +0200
@@ -69,7 +69,7 @@
   - vert_coord: [1.0000000000000000e+00,2.0000000000000000e+00,3.0000000000000000e+00,4.0000000000000000e+00,5.0000000000000000e+00,6.0000000000000000e+00,7.0000000000000000e+00,8.0000000000000000e+00,9.0000000000000000e+00,1.0000000000000000e+01]
   Mask size: 100%
 Fields:
-  stream_function: 2.7047739909863616e+01
+  stream_function: 2.7047739909863619e+01
 Member 4: 
 Valid time:2010-01-01T12:00:00Z
 Quench geometry grid:
