--- /home/benjaminm/code/oops-bundle/ecsaber/saber/test/testref/randomization_bump_nicas_L12L2.ref.tmp	2024-05-06 14:16:30.831699010 +0200
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/test/testref/randomization_bump_nicas_L12L2.ref	2024-05-06 13:02:04.699654685 +0200
@@ -21,12 +21,6 @@
                    Subgrid hash:   -791070867
                 Final nc1:      589
                 Effective horizontal resolution:  4.00
-                Horizontal support radius:    4000.00 km (   4000.00 km -    4000.00 km)
-                Estimated nc1 from horizontal support radius:      589
-                   Decimate full grid, at least      589 points required,    1098 valid points found
-                   Subgrid hash:   -791070867
-                Final nc1:      589
-                Effective horizontal resolution:  4.00
              nc1(  1) =                        589
              nc1(  2) =                        589
              ns =                             1178
@@ -65,7 +59,7 @@
   - vert_coord: [1.0000000000000000e+00]
   Mask size: 100%
 Fields:
-  stream_function: 5.2654209899567903e+01
+  stream_function: 5.2654209899567896e+01
   surface_pressure: 3.6493196800034859e+01
 Member 1: 
 Valid time:2010-01-01T12:00:00Z
@@ -89,8 +83,8 @@
   - vert_coord: [1.0000000000000000e+00]
   Mask size: 100%
 Fields:
-  stream_function: 6.1308623552923613e+01
-  surface_pressure: 3.0497521202377900e+01
+  stream_function: 6.1308623552923599e+01
+  surface_pressure: 3.0497521202377904e+01
 Member 2: 
 Valid time:2010-01-01T12:00:00Z
 Quench geometry grid:
@@ -113,7 +107,7 @@
   - vert_coord: [1.0000000000000000e+00]
   Mask size: 100%
 Fields:
-  stream_function: 5.4252304738225256e+01
+  stream_function: 5.4252304738225263e+01
   surface_pressure: 3.6546122174208186e+01
 Member 3: 
 Valid time:2010-01-01T12:00:00Z
@@ -137,7 +131,7 @@
   - vert_coord: [1.0000000000000000e+00]
   Mask size: 100%
 Fields:
-  stream_function: 4.6354739463248535e+01
+  stream_function: 4.6354739463248542e+01
   surface_pressure: 3.4440406150291579e+01
 Member 4: 
 Valid time:2010-01-01T12:00:00Z
@@ -162,7 +156,7 @@
   Mask size: 100%
 Fields:
   stream_function: 4.5381748346070736e+01
-  surface_pressure: 3.5574095937598663e+01
+  surface_pressure: 3.5574095937598656e+01
 Member 5: 
 Valid time:2010-01-01T12:00:00Z
 Quench geometry grid:
@@ -185,8 +179,8 @@
   - vert_coord: [1.0000000000000000e+00]
   Mask size: 100%
 Fields:
-  stream_function: 4.6614781327696001e+01
-  surface_pressure: 3.0985424312747210e+01
+  stream_function: 4.6614781327695987e+01
+  surface_pressure: 3.0985424312747202e+01
 Member 6: 
 Valid time:2010-01-01T12:00:00Z
 Quench geometry grid:
@@ -209,8 +203,8 @@
   - vert_coord: [1.0000000000000000e+00]
   Mask size: 100%
 Fields:
-  stream_function: 4.6400614870703784e+01
-  surface_pressure: 4.2166176348860205e+01
+  stream_function: 4.6400614870703777e+01
+  surface_pressure: 4.2166176348860212e+01
 Member 7: 
 Valid time:2010-01-01T12:00:00Z
 Quench geometry grid:
@@ -257,8 +251,8 @@
   - vert_coord: [1.0000000000000000e+00]
   Mask size: 100%
 Fields:
-  stream_function: 5.1837418589558872e+01
-  surface_pressure: 3.2426197268635661e+01
+  stream_function: 5.1837418589558865e+01
+  surface_pressure: 3.2426197268635669e+01
 Member 9: 
 Valid time:2010-01-01T12:00:00Z
 Quench geometry grid:
@@ -281,7 +275,7 @@
   - vert_coord: [1.0000000000000000e+00]
   Mask size: 100%
 Fields:
-  stream_function: 4.2221951885493191e+01
+  stream_function: 4.2221951885493198e+01
   surface_pressure: 3.4303396182742098e+01
 Member 10: 
 Valid time:2010-01-01T12:00:00Z
@@ -306,7 +300,7 @@
   Mask size: 100%
 Fields:
   stream_function: 4.5169534092091297e+01
-  surface_pressure: 2.8853381913731290e+01
+  surface_pressure: 2.8853381913731294e+01
 Member 11: 
 Valid time:2010-01-01T12:00:00Z
 Quench geometry grid:
@@ -354,7 +348,7 @@
   Mask size: 100%
 Fields:
   stream_function: 4.5619369981739645e+01
-  surface_pressure: 3.7830756443392929e+01
+  surface_pressure: 3.7830756443392943e+01
 Member 13: 
 Valid time:2010-01-01T12:00:00Z
 Quench geometry grid:
@@ -377,7 +371,7 @@
   - vert_coord: [1.0000000000000000e+00]
   Mask size: 100%
 Fields:
-  stream_function: 4.4721049734735828e+01
+  stream_function: 4.4721049734735843e+01
   surface_pressure: 2.8392676303788651e+01
 Member 14: 
 Valid time:2010-01-01T12:00:00Z
@@ -401,8 +395,8 @@
   - vert_coord: [1.0000000000000000e+00]
   Mask size: 100%
 Fields:
-  stream_function: 4.7343916399977161e+01
-  surface_pressure: 3.2405838516636301e+01
+  stream_function: 4.7343916399977182e+01
+  surface_pressure: 3.2405838516636308e+01
 Member 15: 
 Valid time:2010-01-01T12:00:00Z
 Quench geometry grid:
@@ -426,7 +420,7 @@
   Mask size: 100%
 Fields:
   stream_function: 5.0308184795334448e+01
-  surface_pressure: 3.4694217584989104e+01
+  surface_pressure: 3.4694217584989111e+01
 Member 16: 
 Valid time:2010-01-01T12:00:00Z
 Quench geometry grid:
@@ -449,7 +443,7 @@
   - vert_coord: [1.0000000000000000e+00]
   Mask size: 100%
 Fields:
-  stream_function: 4.2861341251630485e+01
+  stream_function: 4.2861341251630478e+01
   surface_pressure: 3.3373396614389542e+01
 Member 17: 
 Valid time:2010-01-01T12:00:00Z
@@ -474,7 +468,7 @@
   Mask size: 100%
 Fields:
   stream_function: 4.4929966649571092e+01
-  surface_pressure: 3.5376638446985140e+01
+  surface_pressure: 3.5376638446985133e+01
 Member 18: 
 Valid time:2010-01-01T12:00:00Z
 Quench geometry grid:
@@ -497,7 +491,7 @@
   - vert_coord: [1.0000000000000000e+00]
   Mask size: 100%
 Fields:
-  stream_function: 4.1237976043364824e+01
+  stream_function: 4.1237976043364839e+01
   surface_pressure: 3.2082315749252722e+01
 Member 19: 
 Valid time:2010-01-01T12:00:00Z
@@ -522,7 +516,7 @@
   Mask size: 100%
 Fields:
   stream_function: 4.2053902656124500e+01
-  surface_pressure: 3.2637287852385739e+01
+  surface_pressure: 3.2637287852385747e+01
 Member 20: 
 Valid time:2010-01-01T12:00:00Z
 Quench geometry grid:
@@ -545,7 +539,7 @@
   - vert_coord: [1.0000000000000000e+00]
   Mask size: 100%
 Fields:
-  stream_function: 4.4446643230752571e+01
+  stream_function: 4.4446643230752578e+01
   surface_pressure: 2.8887323602459212e+01
 Member 21: 
 Valid time:2010-01-01T12:00:00Z
@@ -569,7 +563,7 @@
   - vert_coord: [1.0000000000000000e+00]
   Mask size: 100%
 Fields:
-  stream_function: 4.0215588892258275e+01
+  stream_function: 4.0215588892258268e+01
   surface_pressure: 3.4121667223317800e+01
 Member 22: 
 Valid time:2010-01-01T12:00:00Z
@@ -593,8 +587,8 @@
   - vert_coord: [1.0000000000000000e+00]
   Mask size: 100%
 Fields:
-  stream_function: 4.6380739206188814e+01
-  surface_pressure: 3.5440975581038273e+01
+  stream_function: 4.6380739206188792e+01
+  surface_pressure: 3.5440975581038266e+01
 Member 23: 
 Valid time:2010-01-01T12:00:00Z
 Quench geometry grid:
@@ -617,8 +611,8 @@
   - vert_coord: [1.0000000000000000e+00]
   Mask size: 100%
 Fields:
-  stream_function: 4.6130791261390335e+01
-  surface_pressure: 3.5749882318396132e+01
+  stream_function: 4.6130791261390328e+01
+  surface_pressure: 3.5749882318396125e+01
 Member 24: 
 Valid time:2010-01-01T12:00:00Z
 Quench geometry grid:
@@ -641,5 +635,5 @@
   - vert_coord: [1.0000000000000000e+00]
   Mask size: 100%
 Fields:
-  stream_function: 4.4922445644650558e+01
-  surface_pressure: 3.2994136361575862e+01
+  stream_function: 4.4922445644650551e+01
+  surface_pressure: 3.2994136361575876e+01
