--- /home/benjaminm/code/oops-bundle/ecsaber/oops-patch/src/oops/generic/UnstructuredInterpolator.cc.tmp	2024-05-27 15:32:35.154341100 +0200
+++ /home/benjaminm/code/oops-bundle/ecsaber/oops-patch/src/oops/generic/UnstructuredInterpolator.cc	2024-05-27 15:26:48.185464307 +0200
@@ -26,10 +26,7 @@
 #include "eckit/config/Configuration.h"
 #include "eckit/exception/Exceptions.h"
 
-#include "oops/base/Geometry.h"
 #include "oops/base/GeometryData.h"
-#include "oops/base/Increment.h"
-#include "oops/base/State.h"
 #include "oops/base/Variables.h"
 #include "oops/util/Logger.h"
 #include "oops/util/missingValues.h"
