--- /home/benjaminm/code/oops-bundle/ecsaber/saber/quench/src/default/FieldsIODefault.h.tmp	2024-06-03 16:18:03.301814194 +0200
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/quench/src/default/FieldsIODefault.h	2024-06-03 10:52:41.461246723 +0200
@@ -15,6 +15,7 @@
 #include "oops/base/Variables.h"
 
 #include "src/FieldsIOBase.h"
+#include "src/Variables.h"
 
 namespace quench {
   class Geometry;
