--- /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/interpolation/Interpolation.h.tmp	2024-06-12 18:33:04.654311500 +0200
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/interpolation/Interpolation.h	2024-05-27 15:34:14.406610855 +0200
@@ -24,6 +24,7 @@
 #include "saber/blocks/SaberBlockParametersBase.h"
 #include "saber/blocks/SaberOuterBlockBase.h"
 #include "saber/interpolation/Geometry.h"
+#include "saber/interpolation/RegionalInterpolation.h"
 
 namespace saber {
 namespace interpolation {
@@ -77,10 +78,13 @@
   const oops::GeometryData & outerGeomData_;
   const oops::JediVariables innerVars_;
   const oops::JediVariables activeVars_;
+  bool regionalGrid_;
   // pointers for delayed initialization
   std::unique_ptr<oops::GeometryData> innerGeomData_;
   std::unique_ptr<oops::GlobalInterpolator> interp_;
+  std::unique_ptr<RegionalInterpolation> regInterp_;
   mutable std::unique_ptr<oops::GlobalInterpolator> inverseInterp_;
+  mutable std::unique_ptr<RegionalInterpolation> regInverseInterp_;
 };
 
 }  // namespace interpolation
