--- /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/util/CMakeLists.txt.tmp	2024-06-12 18:33:05.754315241 +0200
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/util/CMakeLists.txt	2024-06-11 11:30:38.878039888 +0200
@@ -8,6 +8,9 @@
 FieldSetHelpers.h
 mo_cvtcoord_mod.F90
 mo_netcdf_mod.F90
+
+# This should be removed once spack-stack uses the ATLAS version where this interface is merged
+atlas_buildnode2cellconnectivity.F90
 )
 
 set( util_src_files ${util_src_files_list} PARENT_SCOPE )
