--- /home/benjaminm/code/oops-bundle/ecsaber/vader/src/vader/recipes/AirPressure_A.cc.tmp.bak	2024-11-15 16:35:30.749782642 +0100
+++ /home/benjaminm/code/oops-bundle/ecsaber/vader/src/vader/recipes/AirPressure_A.cc	2024-11-15 16:36:51.384200989 +0100
@@ -21,7 +21,8 @@
 
 // Static attribute initialization
 const char AirPressure_A::Name[] = "AirPressure_A";
-const oops::JediVariables AirPressure_A::Ingredients{std::vector<std::string>{"air_pressure_levels"}};
+const oops::JediVariables
+  AirPressure_A::Ingredients{std::vector<std::string>{"air_pressure_levels"}};
 
 // -------------------------------------------------------------------------------------------------
 
