--- /home/benjaminm/code/oops-bundle/ecsaber/vader/src/vader/recipes/RelativeHumidity_A.cc.tmp	2024-11-05 11:37:15.873498151 +0100
+++ /home/benjaminm/code/oops-bundle/ecsaber/vader/src/vader/recipes/RelativeHumidity_A.cc	2024-11-05 11:36:26.033270175 +0100
@@ -22,7 +22,8 @@
 
 // Static attribute initialization
 const char RelativeHumidity_A::Name[] = "RelativeHumidity_A";
-const oops::JediVariables RelativeHumidity_A::Ingredients{std::vector<std::string>{"air_temperature",
+const oops::JediVariables
+  RelativeHumidity_A::Ingredients{std::vector<std::string>{"air_temperature",
                                     "dlsvpdT",
                                     "water_vapor_mixing_ratio_wrt_moist_air_and_condensed_water",
                                     "qsat"}};
