--- /home/benjaminm/code/oops-bundle/ecsaber/vader/src/vader/recipes/WaterVaporMixingRatioWrtMoistAir_A.cc.tmp	2024-11-05 11:37:16.372500433 +0100
+++ /home/benjaminm/code/oops-bundle/ecsaber/vader/src/vader/recipes/WaterVaporMixingRatioWrtMoistAir_A.cc	2024-11-05 11:30:45.017708283 +0100
@@ -22,7 +22,8 @@
 
 // Static attribute initialization
 const char WaterVaporMixingRatioWrtMoistAir_A::Name[] = "WaterVaporMixingRatioWrtMoistAir_A";
-const oops::JediVariables WaterVaporMixingRatioWrtMoistAir_A::Ingredients{{"humidity_mixing_ratio"}};
+const oops::JediVariables
+  WaterVaporMixingRatioWrtMoistAir_A::Ingredients{{"humidity_mixing_ratio"}};
 
 // Register the maker
 static RecipeMaker<WaterVaporMixingRatioWrtMoistAir_A> makerWaterVaporMixingRatioWrtMoistAir_(
