--- /home/benjaminm/code/oops-bundle/ecsaber/vader/src/vader/recipes/WaterVaporMixingRatioWrtMoistAir_A.cc.tmp.bak	2024-11-15 16:35:30.148779522 +0100
+++ /home/benjaminm/code/oops-bundle/ecsaber/vader/src/vader/recipes/WaterVaporMixingRatioWrtMoistAir_A.cc	2024-11-15 16:36:07.653974146 +0100
@@ -22,7 +22,8 @@
 
 // Static attribute initialization
 const char WaterVaporMixingRatioWrtMoistAir_A::Name[] = "WaterVaporMixingRatioWrtMoistAir_A";
-const oops::JediVariables WaterVaporMixingRatioWrtMoistAir_A::Ingredients{{"humidity_mixing_ratio"}};
+const oops::JediVariables
+  WaterVaporMixingRatioWrtMoistAir_A::Ingredients{{"humidity_mixing_ratio"}};
 
 // Register the maker
 static RecipeMaker<WaterVaporMixingRatioWrtMoistAir_A> makerWaterVaporMixingRatioWrtMoistAir_(
