--- /home/benjaminm/code/ecsaber/bundle/ecsaber/saber/src/saber/blocks/SaberEnsembleBlockChain.h.tmp.bak	2026-02-17 18:20:25.019253488 +0100
+++ /home/benjaminm/code/ecsaber/bundle/ecsaber/saber/src/saber/blocks/SaberEnsembleBlockChain.h	2026-02-17 18:21:14.215623577 +0100
@@ -18,9 +18,10 @@
 
 #include "oops/base/FieldSet4D.h"
 #include "oops/base/FieldSets.h"
-#include "oops/base/Geometry.h"
-#include "oops/base/Increment.h"
+#include "oops/interface/Geometry.h"
+#include "oops/interface/Increment.h"
 #include "oops/interface/ModelData.h"
+#include "oops/util/ECUtilities.h"
 #include "oops/util/FieldSetHelpers.h"
 #include "oops/util/Logger.h"
 #include "oops/util/Random.h"
@@ -218,7 +219,7 @@
                        oops::FieldSet4D & fset4dXb,
                        oops::FieldSet4D & fset4dFg,
                        const eckit::Configuration & conf)
-  : comm_(geom.getComm()), outerFunctionSpace_(geom.functionSpace()), outerVariables_(outerVars) {
+  : comm_(geom.geometry().getComm()), outerFunctionSpace_(geom.geometry().functionSpace()), outerVariables_(outerVars) {
   oops::Log::trace() << "SaberEnsembleBlockChain ctor starting" << std::endl;
 
   // Deserialize parameters and fill configuration with missing values
@@ -255,7 +256,7 @@
   oops::JediVariables currentOuterVars = outerBlockChain_ ?
                                      outerBlockChain_->innerVars() : outerVars;
   const oops::GeometryData & currentOuterGeom = outerBlockChain_ ?
-                                   outerBlockChain_->innerGeometryData() : geom.generic();
+                                     outerBlockChain_->innerGeometryData() : util::geomData(geom);
 
   // Get active variables
   const oops::JediVariables activeVars = currentOuterVars;
@@ -283,7 +284,7 @@
   // Read inflation field
   const double inflationValue = params.inflationValue;
   oops::Log::info() << "Info     : Read inflation field" << std::endl;
-  oops::FieldSet3D inflationField(fset4dXb[0].validTime(), geom.getComm());
+  oops::FieldSet3D inflationField(fset4dXb[0].validTime(), geom.geometry().getComm());
   if (params.inflationField.value()) {
     const auto & inflationParams = *params.inflationField.value();
     // Read ATLAS inflation file
@@ -306,11 +307,11 @@
       // Copy file
       // Read fieldsets as increments
       // Create increment
-      oops::Increment<MODEL> dx(geom, activeVars, fset4dXb[0].validTime());
+      oops::Increment<MODEL> dx(geom, util::templatedVars<MODEL>(activeVars), fset4dXb[0].validTime());
       dx.read(inflationConf);
       oops::Log::test() << "Norm of input parameter inflation"
                         << ": " << dx.norm() << std::endl;
-      inflationField.deepCopy(dx.fieldSet());
+      inflationField.deepCopy(dx.increment().fieldSet());
     }
   }
 
@@ -427,7 +428,7 @@
 
     // Get interpolator outer geometry data
     const oops::GeometryData & interpolatorOuterGeomData = outerBlockChain_ ?
-      outerBlockChain_->innerGeometryData() : geom.generic();
+      outerBlockChain_->innerGeometryData() : util::geomData(geom);
 
     // Interpolator outer block chain
     if (scaleParams.interpolatorParams.value()) {
@@ -489,7 +490,7 @@
 
       // Check consistency of `geom` and the current geometry
       const auto & currentFspace = filterOuterGeomData.functionSpace();
-      if (util::getGridUid(geom.functionSpace()) != util::getGridUid(currentFspace)) {
+      if (util::getGridUid(geom.geometry().functionSpace()) != util::getGridUid(currentFspace)) {
         oops::Log::info() << "Info     : Localization and ensemble are on different "
                              "functionSpaces, building localization with generic "
                              "constructor" << std::endl;
@@ -648,10 +649,10 @@
         for (size_t ie = 0; ie < ne; ++ie) {
           // Should be on the model geometry!
           auto pert = oops::Increment<MODEL>(geom,
-                                             scaleData.localization()->outerVariables(),
+                                             util::templatedVars<MODEL>(scaleData.localization()->outerVariables()),
                                              fset4dXb.times()[0]);
           pert.zero();
-          pert.fromFieldSet((*scaleData.ensemble())(0, ie).fieldSet());
+          pert.increment().fromFieldSet((*scaleData.ensemble())(0, ie).fieldSet());
 
           eckit::LocalConfiguration mConfMem(mConf);
           util::setMember(mConfMem, ie+1);
