--- /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/blocks/SaberEnsembleBlockChain.h.tmp.bak	2026-01-14 17:20:46.390403598 +0100
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/blocks/SaberEnsembleBlockChain.h	2026-01-14 15:33:14.983897645 +0100
@@ -17,9 +17,10 @@
 
 #include "oops/base/FieldSet4D.h"
 #include "oops/base/FieldSets.h"
-#include "oops/base/Geometry.h"
-#include "oops/base/Increment.h"
+#include "oops/interface/Geometry.h"
+#include "oops/interface/Increment.h"
 #include "oops/interface/ModelData.h"
+#include "oops/util/ECUtilities.h"
 #include "oops/util/FieldSetHelpers.h"
 #include "oops/util/Logger.h"
 #include "oops/util/Random.h"
@@ -92,7 +93,7 @@
                        oops::FieldSets & fsetEns,
                        const eckit::LocalConfiguration & covarConf,
                        const eckit::Configuration & conf)
-  : outerFunctionSpace_(geom.functionSpace()), outerVariables_(outerVars),
+  : outerFunctionSpace_(geom.geometry().functionSpace()), outerVariables_(outerVars),
     ensemble_(fsetEns), ctlVecSize_(0) {
   oops::Log::trace() << "SaberEnsembleBlockChain ctor starting" << std::endl;
 
@@ -118,7 +119,7 @@
   oops::JediVariables currentOuterVars = outerBlockChain_ ?
                                      outerBlockChain_->innerVars() : outerVars;
   const oops::GeometryData & currentOuterGeom = outerBlockChain_ ?
-                                     outerBlockChain_->innerGeometryData() : geom.generic();
+                                     outerBlockChain_->innerGeometryData() : util::geomData(geom);
 
   // Get parameters:
   SaberCentralBlockParametersWrapper saberCentralBlockParamsWrapper;
@@ -156,7 +157,7 @@
   eckit::LocalConfiguration centralBlockConf = conf.getSubConfiguration("saber central block");
   const double inflationValue = centralBlockConf.getDouble("inflation value", 1);
   oops::Log::info() << "Info     : Read inflation field" << std::endl;
-  oops::FieldSet3D inflationField(fset4dXb[0].validTime(), geom.getComm());
+  oops::FieldSet3D inflationField(fset4dXb[0].validTime(), geom.geometry().getComm());
   // Read ATLAS inflation file
   if (centralBlockConf.has("inflation field.atlas file")) {
     eckit::LocalConfiguration inflationConf =
@@ -179,11 +180,11 @@
     // Copy file
     // Read fieldsets as increments
     // Create increment
-    oops::Increment<MODEL> dx(geom, activeVars, fset4dXb[0].validTime());
+    oops::Increment<MODEL> dx(geom, util::templatedVars<MODEL>(activeVars), fset4dXb[0].validTime());
     dx.read(inflationConf);
     oops::Log::test() << "Norm of input parameter inflation"
                       << ": " << dx.norm() << std::endl;
-    inflationField.deepCopy(dx.fieldSet());
+    inflationField.deepCopy(dx.increment().fieldSet());
   }
 
   // Apply inflation on ensemble members
@@ -248,7 +249,7 @@
   }
 
   const oops::GeometryData & localizationOuterGeomData = outerBlockChain_ ?
-              outerBlockChain_->innerGeometryData() : geom.generic();
+              outerBlockChain_->innerGeometryData() : util::geomData(geom);
 
   // Localization
   const auto & locConf = saberCentralBlockParams.localization.value();
@@ -260,7 +261,7 @@
 
     // Check consistency of `geom` and the current geometry
     const auto & currentFspace = localizationOuterGeomData.functionSpace();
-    if (util::getGridUid(geom.functionSpace()) != util::getGridUid(currentFspace)) {
+    if (util::getGridUid(geom.geometry().functionSpace()) != util::getGridUid(currentFspace)) {
       oops::Log::info() << "Info     : Localization and ensemble are on different "
                            "functionSpaces, building localization with generic "
                            "constructor" << std::endl;
