--- /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/blocks/SaberEnsembleBlockChain.h.tmp.bak	2024-05-06 13:39:34.877664293 +0200
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/blocks/SaberEnsembleBlockChain.h	2024-05-06 13:37:59.713411217 +0200
@@ -17,8 +17,8 @@
 
 #include "oops/base/FieldSet4D.h"
 #include "oops/base/FieldSets.h"
-#include "oops/base/Geometry.h"
-#include "oops/base/Increment.h"
+#include "oops/interface/Geometry.h"
+#include "oops/interface/Increment.h"
 #include "oops/interface/ModelData.h"
 #include "oops/util/FieldSetHelpers.h"
 #include "oops/util/Logger.h"
@@ -28,6 +28,7 @@
 #include "saber/blocks/SaberBlockParametersBase.h"
 #include "saber/blocks/SaberOuterBlockChain.h"
 #include "saber/blocks/SaberParametricBlockChain.h"
+#include "saber/oops/ECUtilities.h"
 #include "saber/oops/Utilities.h"
 
 namespace saber {
@@ -80,6 +81,9 @@
   /// TODO(AS): check whether this is needed or can be inferred from ensemble.
   oops::patch::Variables vars_;
   int seed_ = 7;  // For reproducibility
+  /// @brief Geometry communicator.
+  const eckit::mpi::Comm & comm_;
+  const oops::GeometryData geomData_;
 };
 
 // -----------------------------------------------------------------------------
@@ -98,8 +102,10 @@
                        oops::FieldSets & fsetDualResEns,
                        const eckit::LocalConfiguration & covarConf,
                        const eckit::Configuration & conf)
-  : outerFunctionSpace_(geom.functionSpace()), outerVariables_(outerVars),
-    ensemble_(fsetEns), ctlVecSize_(0) {
+  : outerFunctionSpace_(geom.geometry().functionSpace()), outerVariables_(outerVars),
+    ensemble_(fsetEns), ctlVecSize_(0), comm_(geom.geometry().getComm()),
+    geomData_(geom.geometry().functionSpace(), geom.geometry().fields(),
+    geom.geometry().levelsAreTopDown(), geom.geometry().getComm()) {
   oops::Log::trace() << "SaberEnsembleBlockChain ctor starting" << std::endl;
 
   // Check that there is an ensemble of at least 2 members.
@@ -124,7 +130,7 @@
   oops::patch::Variables currentOuterVars = outerBlockChain_ ?
                                      outerBlockChain_->innerVars() : outerVars;
   const oops::GeometryData & currentOuterGeom = outerBlockChain_ ?
-                                     outerBlockChain_->innerGeometryData() : geom.generic();
+                                     outerBlockChain_->innerGeometryData() : geomData_;
 
   // Get parameters:
   SaberCentralBlockParametersWrapper saberCentralBlockParamsWrapper;
@@ -162,7 +168,7 @@
   eckit::LocalConfiguration centralBlockConf = conf.getSubConfiguration("saber central block");
   const double inflationValue = centralBlockConf.getDouble("inflation value", 1);
   oops::Log::info() << "Info     : Read inflation field" << std::endl;
-  oops::FieldSet3D inflationField(fset4dXb[0].validTime(), geom.getComm());
+  oops::FieldSet3D inflationField(fset4dXb[0].validTime(), geom.geometry().getComm());
   // Read ATLAS inflation file
   if (centralBlockConf.has("inflation field.atlas file")) {
     eckit::LocalConfiguration inflationConf =
@@ -184,12 +190,15 @@
                               centralBlockConf.getSubConfiguration("inflation field.model file");
     // Copy file
     // Read fieldsets as increments
+    // Create variables
+    oops::Variables<MODEL> activeVarsT(templatedVarsConf(activeVars));
+
     // Create increment
-    oops::Increment<MODEL> dx(geom, activeVars, fset4dXb[0].validTime());
+    oops::Increment<MODEL> dx(geom, activeVarsT, fset4dXb[0].validTime());
     dx.read(inflationConf);
     oops::Log::test() << "Norm of input parameter inflation"
                       << ": " << dx.norm() << std::endl;
-    inflationField.deepCopy(dx.fieldSet());
+    inflationField.deepCopy(dx.increment().fieldSet());
   }
 
   // Apply inflation on ensemble members
@@ -254,7 +263,7 @@
   }
 
   const oops::GeometryData & localizationOuterGeomData = outerBlockChain_ ?
-              outerBlockChain_->innerGeometryData() : geom.generic();
+              outerBlockChain_->innerGeometryData() : geomData_;
 
   // Localization
   const auto & locConf = saberCentralBlockParams.localization.value();
@@ -266,7 +275,7 @@
 
     // Check consistency of `geom` and the current geometry
     const auto & currentFspace = localizationOuterGeomData.functionSpace();
-    if (util::getGridUid(geom.functionSpace()) != util::getGridUid(currentFspace)) {
+    if (util::getGridUid(geom.geometry().functionSpace()) != util::getGridUid(currentFspace)) {
       oops::Log::info() << "Info     : Localization and ensemble are on different "
                            "functionSpaces, building localization with generic "
                            "constructor" << std::endl;
