--- /home/benjaminm/code/ecsaber/bundle/ecsaber/saber/src/saber/blocks/SaberParametricBlockChain.h.tmp.bak	2026-06-04 13:55:03.759203854 +0200
+++ /home/benjaminm/code/ecsaber/bundle/ecsaber/saber/src/saber/blocks/SaberParametricBlockChain.h	2026-02-12 15:57:35.990011381 +0100
@@ -21,6 +21,7 @@
 #include "oops/base/FieldSets.h"
 #include "oops/interface/ModelData.h"
 #include "oops/util/ConfigHelpers.h"
+#include "oops/util/ECUtilities.h"
 
 #include "saber/blocks/SaberBlockChainBase.h"
 #include "saber/blocks/SaberBlockParametersBase.h"
@@ -130,10 +131,8 @@
                        oops::FieldSet4D & fset4dXb,
                        oops::FieldSet4D & fset4dFg,
                        const eckit::Configuration & conf)
-  : outerFunctionSpace_(geom.functionSpace()),
-    outerVariables_(outerVars),
-    timeComm_(fset4dXb.commTime()),
-    size4D_(fset4dXb.size()) {
+  : outerFunctionSpace_(geom.geometry().functionSpace()), outerVariables_(outerVars),
+  timeComm_(fset4dXb.commTime()), size4D_(fset4dXb.size()) {
   oops::Log::trace() << "SaberParametricBlockChain ctor starting" << std::endl;
 
   // Deserialize parameters and fill configuration with missing values
@@ -165,7 +164,7 @@
 
   // Set outer geometry data for central block
   const oops::GeometryData & currentOuterGeom = outerBlockChain_ ?
-                             outerBlockChain_->innerGeometryData() : geom.generic();
+                             outerBlockChain_->innerGeometryData() : util::geomData(geom);
 
   // Create central block
   oops::Log::info() << "Info     : Creating central block: " << std::endl;
