--- /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/blocks/SaberOuterBlockChain.h.tmp	2024-07-04 13:56:18.900081979 +0200
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/blocks/SaberOuterBlockChain.h	2024-07-04 11:12:45.511034097 +0200
@@ -18,7 +18,7 @@
 
 #include "oops/base/FieldSet4D.h"
 #include "oops/base/FieldSets.h"
-#include "oops/base/Geometry.h"
+#include "oops/interface/Geometry.h"
 #include "oops/base/GeometryData.h"
 #include "oops/interface/ModelData.h"
 
@@ -175,6 +175,7 @@
   /// TODO(AS): Need to expand this to create different outer blocks for different
   /// times for the 4D with multiple times on one MPI task.
   std::vector<std::unique_ptr<SaberOuterBlockBase>> outerBlocks_;
+  std::unique_ptr<oops::GeometryData> geomData_;
 };
 
 // -----------------------------------------------------------------------------
@@ -186,7 +187,9 @@
                        oops::FieldSet4D & fset4dFg,
                        oops::FieldSets & fsetEns,
                        const eckit::LocalConfiguration & covarConf,
-                       const std::vector<saber::SaberOuterBlockParametersWrapper> & params) {
+                       const std::vector<saber::SaberOuterBlockParametersWrapper> & params) :
+  geomData_(new oops::GeometryData(geom.geometry().functionSpace(), geom.geometry().fields(),
+  geom.geometry().levelsAreTopDown(), geom.geometry().getComm())) {
   oops::Log::trace() << "SaberOuterBlockChain ctor starting" << std::endl;
   oops::Log::info() << "Info     : Creating outer blocks" << std::endl;
 
@@ -203,7 +206,7 @@
     boost::adaptors::reverse(params)) {
     // Initialize current outer geometry data
     const oops::GeometryData & currentOuterGeometryData = outerBlocks_.size() == 0 ?
-                                       geom.generic() : outerBlocks_.back()->innerGeometryData();
+                                       *geomData_ : outerBlocks_.back()->innerGeometryData();
 
     // Initialize outer block
     const auto[saberOuterBlockParams,
@@ -282,7 +285,7 @@
 
     for (size_t ie = 0; ie < nens; ++ie) {
       // Read ensemble member
-      oops::FieldSet3D fset(fset4dXb[0].validTime(), geom.getComm());
+      oops::FieldSet3D fset(fset4dXb[0].validTime(), geom.geometry().getComm());
       readEnsembleMember(geom,
                          outerVars,
                          ensembleConf,
