--- /home/benjaminm/code/ecsaber/bundle/ecsaber/saber/src/saber/blocks/SaberParametricBlockChain.h.tmp.bak	2026-02-09 13:47:02.854673020 +0100
+++ /home/benjaminm/code/ecsaber/bundle/ecsaber/saber/src/saber/blocks/SaberParametricBlockChain.h	2026-02-09 13:50:43.187802773 +0100
@@ -21,6 +21,7 @@
 #include "oops/base/FieldSets.h"
 #include "oops/interface/ModelData.h"
 #include "oops/util/ConfigHelpers.h"
+#include "oops/util/ECUtilities.h"
 
 #include "saber/blocks/SaberBlockChainBase.h"
 #include "saber/blocks/SaberBlockParametersBase.h"
@@ -132,7 +133,7 @@
                        oops::FieldSet4D & fset4dXb,
                        oops::FieldSet4D & fset4dFg,
                        const eckit::Configuration & conf)
-  : outerFunctionSpace_(geom.functionSpace()), outerVariables_(outerVars),
+  : outerFunctionSpace_(geom.geometry().functionSpace()), outerVariables_(outerVars),
   timeComm_(fset4dXb.commTime()), size4D_(fset4dXb.size()) {
   oops::Log::trace() << "SaberParametricBlockChain ctor starting" << std::endl;
 
@@ -165,13 +166,13 @@
 
   // Set outer geometry data for central block
   const oops::GeometryData & currentOuterGeom = outerBlockChain_ ?
-                             outerBlockChain_->innerGeometryData() : geom.generic();
+                             outerBlockChain_->innerGeometryData() : util::geomData(geom);
 
   // Create central block
   oops::Log::info() << "Info     : Creating central block: " << std::endl;
 
   const auto currentOuterVars = initCentralBlock(currentOuterGeom,
-                                                 geom.levelsAreTopDown(),
+                                                 geom.geometry().levelsAreTopDown(),
                                                  fullConf,
                                                  saberCentralBlockParams,
                                                  fset4dXb,
