--- /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/blocks/SaberBlockParametersBase.h.tmp	2024-03-21 10:06:06.003034629 +0100
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/blocks/SaberBlockParametersBase.h	2024-03-21 09:18:34.837611448 +0100
@@ -46,6 +46,10 @@
   // Flag to run the left inverse instead of the adjoint.
   oops::Parameter<bool> filterMode{"filter mode", false, this};
 
+  // Fields metadata (e.g. geographical mask name, vertical coordinate field name)
+  oops::Parameter<eckit::LocalConfiguration> fieldsMetaData{"fields metadata",
+    eckit::LocalConfiguration(), this};
+
   // OPTIONAL
   // Active variables
   oops::OptionalParameter<oops::patch::Variables> activeVars{"active variables", this};
