--- /home/benjaminm/code/ecsaber/bundle/ecsaber/saber/src/saber/blocks/SaberCentralBlockBase.h.tmp.bak	2026-01-29 15:41:02.421744487 +0100
+++ /home/benjaminm/code/ecsaber/bundle/ecsaber/saber/src/saber/blocks/SaberCentralBlockBase.h	2026-01-29 15:43:25.908128125 +0100
@@ -28,6 +28,7 @@
 #include "oops/util/Printable.h"
 
 #include "saber/blocks/SaberBlockParametersBase.h"
+#include "oops/util/ECUtilities.h"
 
 // Forward declaration
 namespace oops {
@@ -215,11 +216,11 @@
   std::vector<oops::FieldSet3D> fsetVec;
   for (const auto & input : this->getReadConfs()) {
     // Create increment
-    oops::Increment<MODEL> dx(geom, vars, validTime_);
+    oops::Increment<MODEL> dx(geom, util::templatedVars<MODEL>(vars), validTime_);
     dx.read(input.second);
     oops::Log::test() << "Norm of input parameter " << input.first
                       << ": " << dx.norm() << std::endl;
-    fsetVec.push_back(dx.fieldSet());
+    fsetVec.push_back(util::fieldSet(dx));
     fsetVec.back().name() = input.first;
   }
   this->setReadFields(fsetVec);
@@ -239,8 +240,8 @@
 
   // Write fieldsets as increments
   for (const auto & output : outputs) {
-    oops::Increment<MODEL> dx(geom, output.second.variables(), validTime_);
-    dx.fromFieldSet(output.second.fieldSet());
+    oops::Increment<MODEL> dx(geom, util::templatedVars<MODEL>(output.second.variables()), validTime_);
+    dx.increment().fromFieldSet(output.second.fieldSet());
     oops::Log::test() << "Norm of output parameter " << output.second.name()
                       << ": " << dx.norm() << std::endl;
     dx.write(output.first);
