--- /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/blocks/SaberOuterBlockChain.h.tmp.bak	2024-02-15 12:57:27.038736479 +0100
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/blocks/SaberOuterBlockChain.h	2024-02-15 12:58:32.496617697 +0100
@@ -16,7 +16,7 @@
 
 #include "oops/base/FieldSet4D.h"
 #include "oops/base/FieldSets.h"
-#include "oops/base/Geometry.h"
+#include "oops/interface/Geometry.h"
 #include "oops/interface/ModelData.h"
 
 #include "saber/blocks/SaberBlockParametersBase.h"
@@ -152,7 +152,7 @@
     oops::patch::Variables currentOuterVars = outerBlocks_.size() == 0 ?
                                        outerVars : outerBlocks_.back()->innerVars();
     const oops::GeometryData & currentOuterGeometryData = outerBlocks_.size() == 0 ?
-                                       geom.generic() : outerBlocks_.back()->innerGeometryData();
+      geom.geometry().generic() : outerBlocks_.back()->innerGeometryData();
 
     // Get outer block parameters
     const SaberBlockParametersBase & saberOuterBlockParams =
@@ -194,7 +194,7 @@
 
         for (size_t ie = 0; ie < nens; ++ie) {
           // Read ensemble member
-          oops::FieldSet3D fset(fset4dXb[0].validTime(), geom.getComm());
+          oops::FieldSet3D fset(fset4dXb[0].validTime(), geom.geometry().getComm());
           readEnsembleMember(geom,
                              currentOuterVars,
                              ensembleConf,
@@ -266,8 +266,8 @@
       && saberOuterBlockParams.inverseVars.value().size() > 0) {
       oops::Log::info() << "Info     : Left inverse multiplication on xb and fg" << std::endl;
       // Share fields pointers
-      oops::FieldSet3D fsetXbInv(fset4dXb[0].validTime(), geom.getComm());
-      oops::FieldSet3D fsetFgInv(fset4dXb[0].validTime(), geom.getComm());
+      oops::FieldSet3D fsetXbInv(fset4dXb[0].validTime(), geom.geometry().getComm());
+      oops::FieldSet3D fsetFgInv(fset4dXb[0].validTime(), geom.geometry().getComm());
       for (const auto & var : saberOuterBlockParams.inverseVars.value().variables()) {
         fsetXbInv.fieldSet().add(fset4dXb[0].fieldSet().field(var));
         fsetFgInv.fieldSet().add(fset4dFg[0].fieldSet().field(var));
