--- /home/benjaminm/code/ecsaber/bundle/ecsaber/saber/src/saber/blocks/SaberOuterBlockChain.h.tmp.bak	2026-06-11 09:34:50.647365835 +0200
+++ /home/benjaminm/code/ecsaber/bundle/ecsaber/saber/src/saber/blocks/SaberOuterBlockChain.h	2026-06-11 09:35:14.473396453 +0200
@@ -21,7 +21,7 @@
 
 #include "oops/base/FieldSet4D.h"
 #include "oops/base/FieldSets.h"
-#include "oops/base/Geometry.h"
+#include "oops/interface/Geometry.h"
 #include "oops/base/GeometryData.h"
 #include "oops/interface/ModelData.h"
 
@@ -266,7 +266,7 @@
   for (int jb = params.size()-1; jb >= 0; --jb) {
     // Initialize current outer geometry data
     const oops::GeometryData & currentOuterGeometryData = outerBlocks_.size() == 0 ?
-      geom.generic() : innerGeometryData();
+      util::geomData(geom) : innerGeometryData();
 
     // Initialize outer block
     const auto[saberOuterBlockParams,
@@ -380,7 +380,7 @@
 
     for (size_t ie = 0; ie < nens; ++ie) {
       // Read ensemble member
-      oops::FieldSet3D fset(fset4dXb[0].validTime(), geom.getComm());
+      oops::FieldSet3D fset(fset4dXb[0].validTime(), geom.geometry().getComm());
       readEnsembleMember(geom,
                          outerVars,
                          conf,
