--- /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/bump/BUMP.cc.tmp.bak	2024-03-07 12:40:35.924712458 +0100
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/bump/BUMP.cc	2024-03-07 12:33:34.644376854 +0100
@@ -165,9 +165,49 @@
     grid.set("model.2d variables", var2d);
 
     // Add level index for 2D fields
-    if (!grid.has("model.lev2d")) {
+    if (!grid.has("model.level for 2d variables")) {
       ModelDef def;
-      grid.set("model.lev2d", def.lev2d.second);
+      grid.set("model.level for 2d variables", def.lev2d.second);
+    }
+
+    // Add vertical coordinate name
+    if (!grid.has("model.vertical coordinate name")) {
+      std::string vert_coordName;
+      for (const auto & var : vars_str) {
+        if (vars.hasMetaData(var, "vert_coord")) {
+          if (var2d.size() == vars_str.size() || vars_.getLevels(var) > 1) {
+            if (vert_coordName.empty()) {
+              vert_coordName = vars.getMetaData<std::string>(var, "vert_coord");
+            } else {
+              ASSERT(vars.getMetaData<std::string>(var, "vert_coord") == vert_coordName);
+            }
+          }
+        }
+      }
+      if (vert_coordName.empty() && geometryData.fieldSet().has("vert_coord")) {
+        vert_coordName = "vert_coord";
+      }
+      grid.set("model.vertical coordinate name", vert_coordName);
+    }
+
+    // Add geographical mask name
+    if (!grid.has("model.geographical mask name")) {
+      std::string gmaskName;
+      for (const auto & var : vars_str) {
+        if (vars.hasMetaData(var, "gmask")) {
+          if (var2d.size() == vars_str.size() || vars_.getLevels(var) > 1) {
+            if (gmaskName.empty()) {
+              gmaskName = vars.getMetaData<std::string>(var, "gmask");
+            } else {
+              ASSERT(vars.getMetaData<std::string>(var, "gmask") == gmaskName);
+            }
+          }
+        }
+      }
+      if (gmaskName.empty() && geometryData.fieldSet().has("gmask")) {
+        gmaskName = "gmask";
+      }
+      grid.set("model.geographical mask name", gmaskName);
     }
 
     // Create BUMP instance
