--- /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/bump/BUMPParameters.h.tmp	2024-05-22 11:25:07.046617536 +0200
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/bump/BUMPParameters.h	2024-05-22 10:49:59.959295489 +0200
@@ -628,6 +628,10 @@
   oops::Parameter<int> file_component = param(def.file_component, this);
   // Same horizontal convolution for all levels, no vertical convolution
   oops::Parameter<bool> same_horizontal = param(def.same_horizontal, this);
+  // Similar levels threshold (relative tolerance on horizontal length-scale)
+  oops::Parameter<double> sim_levs_th = param(def.sim_levs_th, this);
+  // Read/write interpolation in global file
+  oops::Parameter<bool> interp_in_global_file = param(def.interp_in_global_file, this);
 };
 
 // -----------------------------------------------------------------------------
