--- /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/bump/BUMPParameters.h.tmp	2024-04-15 14:44:00.317838648 +0200
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/bump/BUMPParameters.h	2024-04-15 12:18:32.439184520 +0200
@@ -462,6 +462,8 @@
   oops::Parameter<int> ne_lr = param(def.ne_lr, this);
   // Gaussian approximation for asymptotic quantities
   oops::Parameter<bool> gau_approx = param(def.gau_approx, this);
+  // Compute localization from correlation
+  oops::Parameter<bool> loc_from_cor = param(def.loc_from_cor, this);
   // Threshold on generalized kurtosis (3.0 = Gaussian distribution)
   oops::Parameter<double> gen_kurt_th = param(def.gen_kurt_th, this);
   // Number of bins for averaged statistics histograms
