--- /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/bump/BUMPParameters.h.tmp	2024-02-14 13:20:32.435915727 +0100
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/bump/BUMPParameters.h	2024-02-13 13:47:07.386177411 +0100
@@ -565,6 +565,8 @@
   bump_lib::FitDef def;
 
  public:
+  // Threshold to filter out lower raw values
+  oops::Parameter<double> diag_raw_th = param(def.diag_raw_th, this);
   // Horizontal filtering suport radius [in meters]
   oops::Parameter<double> diag_rhflt = param(def.diag_rhflt, this);
   // Vertical filtering support radius
