--- /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/bump/interpolatorbump_mod.fypp.tmp	2024-03-07 12:22:36.887915458 +0100
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/bump/interpolatorbump_mod.fypp	2024-03-07 11:36:34.101881390 +0100
@@ -191,7 +191,7 @@
 
 ! Unpack output grid from output functionspace
 bint%geom_out%nl0 = bint%bump%nam%nl0
-call bint%geom_out%from_atlas(bint%bump%mpl,bint%afunctionspace_out)
+call bint%geom_out%from_atlas(bint%bump%mpl,bint%bump%nam,bint%afunctionspace_out)
 bint%nouta = size(bint%geom_out%lon_mga)
 
 ! Run basic BUMP drivers
