--- /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/bump/NICAS.h.tmp	2024-04-04 15:19:53.373076333 +0200
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/bump/NICAS.h	2024-04-04 14:55:23.035374455 +0200
@@ -81,8 +81,8 @@
 
  private:
   void print(std::ostream &) const override;
-  BUMPParameters bumpParams_;
   oops::patch::Variables activeVars_;
+  BUMPParameters bumpParams_;
   std::unique_ptr<BUMP> bump_;
   size_t memberIndex_;
 };
