--- /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/bump/NICAS.cc.tmp	2024-03-21 10:06:07.379035315 +0100
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/bump/NICAS.cc	2024-03-21 09:18:58.977623509 +0100
@@ -53,6 +53,7 @@
                        activeVars_,
                        covarConf,
                        bumpParams_,
+                       params.fieldsMetaData.value(),
                        xb));
 
   // Read input ATLAS files
@@ -102,7 +103,7 @@
   for (const auto & fset : fsetVec) {
     bump_->addField(fset);
   }
-  oops::Log::trace() << classname() << "::setReadFields starting" << std::endl;
+  oops::Log::trace() << classname() << "::setReadFields done" << std::endl;
 }
 
 // -----------------------------------------------------------------------------
