--- /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/bump/PsiChiToUV.cc.tmp.bak	2024-03-07 12:40:36.032712546 +0100
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/bump/PsiChiToUV.cc	2024-03-07 12:36:12.056500277 +0100
@@ -34,6 +34,17 @@
   const int modelLevels(outerVars.getLevels("eastward_wind"));
   innerVars.addMetaData("stream_function", "levels", modelLevels);
   innerVars.addMetaData("velocity_potential", "levels", modelLevels);
+  if (outerVars.hasMetaData("eastward_wind", "vert_coord")) {
+    const std::string vert_coordName = outerVars.getMetaData<std::string>("eastward_wind",
+      "vert_coord");
+    innerVars.addMetaData("stream_function", "vert_coord", vert_coordName);
+    innerVars.addMetaData("velocity_potential", "vert_coord", vert_coordName);
+  }
+  if (outerVars.hasMetaData("eastward_wind", "gmask")) {
+    const std::string gmaskName = outerVars.getMetaData<std::string>("eastward_wind", "gmask");
+    innerVars.addMetaData("stream_function", "gmask", gmaskName);
+    innerVars.addMetaData("velocity_potential", "gmask", gmaskName);
+  }
   return innerVars;
 }
 
@@ -94,6 +105,16 @@
 void PsiChiToUV::multiply(oops::FieldSet3D & fset) const {
   oops::Log::trace() << classname() << "::multiply starting" << std::endl;
   bump_->multiplyPsiChiToUV(fset);
+  if (fset[innerVars_[0]].metadata().has("vert_coord")) {
+    const std::string vert_coordName = fset[innerVars_[0]].metadata().getString("vert_coord");
+    fset[outerVars_[0]].metadata().set("vert_coord", vert_coordName);
+    fset[outerVars_[1]].metadata().set("vert_coord", vert_coordName);
+  }
+  if (fset[innerVars_[0]].metadata().has("gmask")) {
+    const std::string gmaskName = fset[innerVars_[0]].metadata().getString("gmask");
+    fset[outerVars_[0]].metadata().set("gmask", gmaskName);
+    fset[outerVars_[1]].metadata().set("gmask", gmaskName);
+  }
   fset.removeFields(innerVars_);
   oops::Log::trace() << classname() << "::multiply done" << std::endl;
 }
@@ -103,6 +124,16 @@
 void PsiChiToUV::multiplyAD(oops::FieldSet3D & fset) const {
   oops::Log::trace() << classname() << "::multiplyAD starting" << std::endl;
   bump_->multiplyPsiChiToUVAd(fset);
+  if (fset[outerVars_[0]].metadata().has("vert_coord")) {
+    const std::string vert_coordName = fset[outerVars_[0]].metadata().getString("vert_coord");
+    fset[outerVars_[0]].metadata().set("vert_coord", vert_coordName);
+    fset[outerVars_[1]].metadata().set("vert_coord", vert_coordName);
+  }
+  if (fset[outerVars_[0]].metadata().has("gmask")) {
+    const std::string gmaskName = fset[outerVars_[0]].metadata().getString("gmask");
+    fset[innerVars_[0]].metadata().set("gmask", gmaskName);
+    fset[innerVars_[1]].metadata().set("gmask", gmaskName);
+  }
   fset.removeFields(outerVars_);
   oops::Log::trace() << classname() << "::multiplyAD done" << std::endl;
 }
