--- /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/bump/PsiChiToUV.h.tmp	2024-04-04 15:19:53.453076447 +0200
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/bump/PsiChiToUV.h	2024-04-04 14:55:43.931390450 +0200
@@ -71,6 +71,7 @@
   const oops::GeometryData & innerGeometryData_;
   oops::patch::Variables innerVars_;
   oops::patch::Variables outerVars_;
+  oops::patch::Variables activeVars_;
   BUMPParameters bumpParams_;
   std::unique_ptr<BUMP> bump_;
 };
