--- /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/bump/StdDev.h.tmp	2024-04-04 15:19:53.569076613 +0200
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/bump/StdDev.h	2024-04-04 14:55:50.151395259 +0200
@@ -81,8 +81,8 @@
   void print(std::ostream &) const override;
   const oops::GeometryData & innerGeometryData_;
   oops::patch::Variables innerVars_;
-  BUMPParameters bumpParams_;
   oops::patch::Variables activeVars_;
+  BUMPParameters bumpParams_;
   std::unique_ptr<BUMP> bump_;
   size_t memberIndex_;
 };
