--- /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/bump/StdDev.cc.tmp	2024-03-21 10:06:07.139035196 +0100
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/bump/StdDev.cc	2024-03-21 09:18:55.297621670 +0100
@@ -56,6 +56,7 @@
                        activeVars_,
                        covarConf,
                        bumpParams_,
+                       params.fieldsMetaData.value(),
                        xb));
 
   // Read input ATLAS files
