--- /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/bump/StdDev.cc.tmp.bak	2024-04-15 14:43:59.949839833 +0200
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/bump/StdDev.cc	2024-04-04 15:35:09.282252474 +0200
@@ -34,31 +34,14 @@
   : SaberOuterBlockBase(params, xb.validTime()),
     innerGeometryData_(outerGeometryData),
     innerVars_(outerVars),
-    bumpParams_(),
-    bump_(),
+    activeVars_(getActiveVars(params, outerVars)),
+    bumpParams_(params.calibrationParams.value() != boost::none ? *params.calibrationParams.value()
+      : *params.readParams.value()),
+    bump_(new BUMP(outerGeometryData, activeVars_, covarConf, bumpParams_,
+      params.fieldsMetaData.value(), xb)),
     memberIndex_(0) {
   oops::Log::trace() << classname() << "::StdDev starting" << std::endl;
 
-  // Get active variables
-  activeVars_ = getActiveVars(params, outerVars);
-
-  // Get BUMP parameters
-  if (params.doCalibration()) {
-    bumpParams_ = *params.calibrationParams.value();
-  } else if (params.doRead()) {
-    bumpParams_ = *params.readParams.value();
-  } else {
-    throw eckit::UserError("calibration or read required in BUMP", Here());
-  }
-
-  // Initialize BUMP
-  bump_.reset(new BUMP(outerGeometryData,
-                       activeVars_,
-                       covarConf,
-                       bumpParams_,
-                       params.fieldsMetaData.value(),
-                       xb));
-
   // Read input ATLAS files
   bump_->readAtlasFiles();
 
