--- /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/bump/subr_list.fypp.tmp	2024-06-12 18:33:06.450317608 +0200
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/bump/subr_list.fypp	2024-06-11 11:30:43.890067328 +0200
@@ -35,7 +35,7 @@
 #:set subr_list = subr_list + ["func_xyz2lonlat"]
 #:set subr_list = subr_list + ["func_vector_product"]
 #:set subr_list = subr_list + ["func_det"]
-#:set subr_list = subr_list + ["func_order_cc"]
+#:set subr_list = subr_list + ["func_area"]
 #:set subr_list = subr_list + ["func_add"]
 #:set subr_list = subr_list + ["func_divide"]
 #:set subr_list = subr_list + ["func_vert_interp_size"]
@@ -293,7 +293,8 @@
 #:set subr_list = subr_list + ["hdiag_get_value"]
 #:set subr_list = subr_list + ["hdiag_run_hdiag"]
 #:set subr_list = subr_list + ["hdiag_from_gsi"]
-#:set subr_list = subr_list + ["hull_init"]
+#:set subr_list = subr_list + ["hull_init_centroid_radius"]
+#:set subr_list = subr_list + ["hull_init_boundary_nodes"]
 #:set subr_list = subr_list + ["hull_dealloc"]
 #:set subr_list = subr_list + ["hull_inside"]
 #:set subr_list = subr_list + ["io_dealloc"]
@@ -627,7 +628,6 @@
 #:set subr_list = subr_list + ["ssrfpack_setup"]
 #:set subr_list = subr_list + ["ssrfpack_snhcsh"]
 #:set subr_list = subr_list + ["stripack_addnod"]
-#:set subr_list = subr_list + ["stripack_area"]
 #:set subr_list = subr_list + ["stripack_bdyadd"]
 #:set subr_list = subr_list + ["stripack_bnodes"]
 #:set subr_list = subr_list + ["stripack_covsph"]
