--- /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/bump/type_hdiag.fypp.tmp.bak	2024-04-15 14:44:00.137839226 +0200
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/bump/type_hdiag.fypp	2024-04-15 12:18:25.251446325 +0200
@@ -319,14 +319,14 @@
    ! Ensemble 1
    write(mpl%info,'(a7,a)') '','Ensemble 1:'
    call mpl%flush
-   call hdiag%loc(1)%localization(mpl,nam,geom(1),samp(1),hdiag%avg(1),'loc1')
+   call hdiag%loc(1)%localization(mpl,nam,geom(1),samp(1),hdiag%avg(1),nam%ne,'loc1')
 end if
 
 if (nam%compute_loc2) then
    ! Ensemble 2
    write(mpl%info,'(a7,a)') '','Ensemble 2:'
    call mpl%flush
-   call hdiag%loc(2)%localization(mpl,nam,geom(2),samp(2),hdiag%avg(2),'loc2')
+   call hdiag%loc(2)%localization(mpl,nam,geom(2),samp(2),hdiag%avg(2),nam%ne_lr,'loc2')
 end if
 
 if (nam%compute_hyb) then
