--- /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/bump/type_bump_parameters.cc.tmp	2024-03-07 12:22:37.023915538 +0100
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/bump/type_bump_parameters.cc	2024-03-07 11:36:34.101881390 +0100
@@ -180,6 +180,10 @@
   param(modelDef.lev2d, modelConf);
   // Check that sampling couples and interpolations do not cross mask boundaries
   param(modelDef.mask_check, modelConf);
+  // Vertical coordinate field name in geometry fields
+  param(modelDef.vert_coord_name, modelConf);
+  // Geographical mask name in geometry fields
+  param(modelDef.gmask_name, modelConf);
 
   // Ensemble sizes section
   EnsembleSizesDef ensembleSizesDef;
