--- /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/bump/type_geom.fypp.tmp.bak	2024-05-27 15:33:44.898529997 +0200
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/bump/type_geom.fypp	2024-05-27 15:34:19.342624433 +0200
@@ -430,6 +430,37 @@
    deallocate(order_c0a)
 end if
 
+if (((nam%new_nicas.or.nam%load_nicas_local).and.nam%write_nicas_global).or.nam%load_nicas_global) then
+   ! Allocation
+   if (mpl%main) then
+      allocate(lon_c0(geom%nc0))
+      allocate(lat_c0(geom%nc0))
+      allocate(order_c0(geom%nc0))
+   else
+      allocate(lon_c0(0))
+      allocate(lat_c0(0))
+      allocate(order_c0(0))
+   end if
+   allocate(order_c0a(geom%nc0a))
+
+   ! Communication
+   call mpl%loc_to_glb(geom%nc0a,geom%nc0,geom%c0a_to_c0,geom%lon_c0a,lon_c0)
+   call mpl%loc_to_glb(geom%nc0a,geom%nc0,geom%c0a_to_c0,geom%lat_c0a,lat_c0)
+
+   ! Compute global order
+   if (mpl%main) call qsort(geom%nc0,lon_c0,lat_c0,order_c0,.false.)
+   call mpl%glb_to_loc(geom%nc0a,geom%nc0,geom%c0a_to_c0,order_c0,order_c0a)
+
+   ! Setup fields output
+   call geom%io%init(mpl,7,'0',geom%nc0,geom%nc0a,geom%c0a_to_c0,geom%proc_to_c0_offset,order_c0a)
+
+   ! Release memory
+   deallocate(lon_c0)
+   deallocate(lat_c0)
+   deallocate(order_c0)
+   deallocate(order_c0a)
+end if
+
 ! Print summary
 write(mpl%info,'(a7,a)') '','Geometry summary:'
 call mpl%flush
