--- /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/bump/type_bump_parameters.h.tmp.bak	2024-04-03 17:47:41.281008550 +0200
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/bump/type_bump_parameters.h	2024-03-22 15:23:12.970496911 +0100
@@ -587,6 +587,14 @@
   // Iterative algorithm (ensemble members loaded sequentially)
   std::pair<const char *, bool> iterative_algo =
     std::make_pair("iterative algorithm", false);
+
+  // Vertical coordinate field name in geometry fields
+  std::pair<const char *, std::string> vert_coord_name =
+    std::make_pair("vertical coordinate name", "");
+
+  // Geographical mask name in geometry fields
+  std::pair<const char *, std::string> gmask_name =
+    std::make_pair("geographical mask name", "");
 };
 
 // -----------------------------------------------------------------------------
