--- /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/bump/type_var.fypp.tmp.bak	2024-05-10 15:10:15.630090363 +0200
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/bump/type_var.fypp	2024-05-10 15:11:29.870255257 +0200
@@ -322,7 +322,7 @@
 end if
 
 ! Reset random seed if necessary
-call rng%reseed(mpl)
+call rng%reseed
 
 ! Probe out
 @:probe_out()
@@ -439,7 +439,7 @@
 end if
 
 ! Reset random seed if necessary
-call rng%reseed(mpl)
+call rng%reseed
 
 ! Probe out
 @:probe_out()
