--- /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/bump/type_bump_parameters.h.tmp	2024-03-07 12:22:36.927915481 +0100
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/bump/type_bump_parameters.h	2024-03-07 11:36:34.101881390 +0100
@@ -302,6 +302,14 @@
   // Check that sampling couples and interpolations do not cross mask boundaries
   std::pair<const char *, bool> mask_check =
     std::make_pair("do not cross mask boundaries", false);
+
+  // Vertical coordinate field name in geometry fields
+  std::pair<const char *, std::string> vert_coord_name =
+    std::make_pair("vertical coordinate name", "");
+
+  // Geographical mask name in geometry fields
+  std::pair<const char *, std::string> gmask_name =
+    std::make_pair("geographical mask name", "");
 };
 
 // Ensemble sizes section
