--- /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/bump/type_bump_parameters.h.tmp	2024-04-15 14:43:59.833840212 +0200
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/bump/type_bump_parameters.h	2024-04-15 12:15:50.655592273 +0200
@@ -408,6 +408,10 @@
   std::pair<const char *, bool> gau_approx =
     std::make_pair("gaussian approximation", false);
 
+  // Compute localization from correlation
+  std::pair<const char *, bool> loc_from_cor =
+    std::make_pair("localization from correlation", false);
+
   // Threshold on generalized kurtosis (3.0 = Gaussian distribution)
   std::pair<const char *, double> gen_kurt_th =
     std::make_pair("generalized kurtosis threshold", std::numeric_limits<double>().max());
