--- /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/bump/type_samp.fypp.tmp	2024-04-19 11:55:53.438004882 +0200
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/bump/type_samp.fypp	2024-04-19 11:48:40.307579514 +0200
@@ -2310,7 +2310,6 @@
          do ildwv=1,nam%nldwv
             iproc = samp%ldwv_to_proc(ildwv)
             if (iproc==mpl%myproc) then
-
                ic0a = samp%ldwv_to_c0a(ildwv)
                smask_hor_c0a(ic0a) = .false.
             end if
