--- /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/bump/type_bump_parameters.cc.tmp.bak	2024-04-03 17:47:41.493008962 +0200
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/bump/type_bump_parameters.cc	2024-03-22 15:23:12.970496911 +0100
@@ -333,6 +333,10 @@
   eckit::LocalConfiguration externalConf;
   // Iterative algorithm (ensemble members loaded sequentially)
   param(externalDef.iterative_algo, externalConf);
+  // Vertical coordinate field name in geometry fields
+  param(externalDef.vert_coord_name, externalConf);
+  // Geographical mask name in geometry fields
+  param(externalDef.gmask_name, externalConf);
 
   // General parameters
   config->set("general", generalConf);
