--- /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/bump/type_bump_parameters.h.tmp	2024-05-22 11:25:06.614616785 +0200
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/bump/type_bump_parameters.h	2024-05-22 10:49:43.491269304 +0200
@@ -568,7 +568,15 @@
 
   // Same horizontal convolution for all levels, no vertical convolution
   std::pair<const char *, bool> same_horizontal =
-    std::make_pair("same horizontal convolution", 0);
+    std::make_pair("same horizontal convolution", false);
+
+  // Similar levels threshold (relative tolerance on horizontal length-scale)
+  std::pair<const char *, double> sim_levs_th =
+    std::make_pair("similar levels threshold", 0.0);
+
+  // Read/write interpolation in global file
+  std::pair<const char *, bool> interp_in_global_file =
+    std::make_pair("interpolation in global file", false);
 };
 
 // Psichitouv section
