--- /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/bump/type_bump.fypp.tmp	2024-05-22 11:25:06.790617091 +0200
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/bump/type_bump.fypp	2024-05-22 10:49:49.163278227 +0200
@@ -120,7 +120,7 @@
 call initialize_constants
 
 ! Initialize MPL
-call bump%mpl%init(comm,conf)
+call bump%mpl%init(comm)
 
 ! Initialize parameters
 call bump%nam%init(bump%mpl)
@@ -1728,12 +1728,6 @@
          fld_c0a(:,il0) = bump%geom(igeom)%lat_c0a*rad2deg
       end do
       found = .true.
-   case ('area')
-      if (.not.allocated(bump%geom(igeom)%area_c0a)) call bump%mpl%abort('${subr}$',trim(param)//' is not allocated')
-      do il0=1,bump%geom(igeom)%nl0
-         fld_c0a(:,il0) = bump%geom(igeom)%area_c0a*req**2
-      end do
-      found = .true.
    case ('vert_coord')
       if (.not.allocated(bump%geom(igeom)%vert_coord_c0a)) call bump%mpl%abort('${subr}$',trim(param)//' is not allocated')
       fld_c0a = bump%geom(igeom)%vert_coord_c0a
