--- /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/bump/type_bump_parameters.cc.tmp	2024-04-15 14:44:00.033839560 +0200
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/bump/type_bump_parameters.cc	2024-04-15 12:15:51.539664802 +0200
@@ -233,6 +233,8 @@
   param(diagnosticsDef.ne_lr, diagnosticsConf);
   // Gaussian approximation for asymptotic quantities
   param(diagnosticsDef.gau_approx, diagnosticsConf);
+  // Compute localization from correlation
+  param(diagnosticsDef.loc_from_cor, diagnosticsConf);
   // Threshold on generalized kurtosis (3.0 = Gaussian distribution)
   param(diagnosticsDef.gen_kurt_th, diagnosticsConf);
   // Number of bins for averaged statistics histograms
