--- /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/bump/type_hull.fypp.tmp	2024-07-04 13:56:20.416086228 +0200
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/bump/type_hull.fypp	2024-07-04 11:24:48.445569257 +0200
@@ -123,9 +123,6 @@
 integer,intent(in) :: n                 !< Size
 real(kind_real),intent(in) :: vbnd(3,n) !< Boundary nodes
 
-! Local variables
-integer :: ib
-
 ! Set name
 @:set_name(hull_init_boundary_nodes)
 
