--- /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/bump/type_geom.fypp.tmp.bak	2024-08-27 12:41:01.209468555 +0200
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/bump/type_geom.fypp	2024-08-27 12:46:48.632810244 +0200
@@ -192,8 +192,6 @@
    procedure :: c0_to_c0u => geom_c0_to_c0u
 end type geom_type
 
-integer,parameter :: nzoom = 10         !< Number of zoom iteration to compute the work grid
-integer,parameter :: ntest = 100         !< Number of test values to compute the work grid
 integer,parameter :: nfacmin = 4        !< Minimum size factor for the first subsampling
 integer,parameter :: nsmax = 1000000    !< Maximum sampling size
 logical,parameter :: parallel = .false. !< Parallel sampling (can be slow because of communications)
