--- /home/benjaminm/code/ecsaber/bundle/ecsaber/saber/test/testlist/saber_spectralb-vader.cmake.tmp.bak	2026-06-04 13:55:08.274228031 +0200
+++ /home/benjaminm/code/ecsaber/bundle/ecsaber/saber/test/testlist/saber_spectralb-vader.cmake	2026-06-04 14:00:18.245966414 +0200
@@ -336,22 +336,22 @@
 
   # dirac_spectral_parallel_gauss_vader_with_smv_6
   # (should give similar results to dirac_spectral_gauss_vader_6_2_1)
-  saber_add_test( TARGET saber_dirac_spectralb_parallel_gauss_vader_6_4-2
-                  MPI 4
-                  OMP 2
-                  COMMAND ${CMAKE_BINARY_DIR}/bin/saber_quench_error_covariance_toolbox.x
-                  ARGS testinput/dirac_spectralb_parallel_gauss_vader_6.yaml
-                  DEPENDS saber_quench_error_covariance_toolbox.x )
+#  saber_add_test( TARGET saber_dirac_spectralb_parallel_gauss_vader_6_4-2
+#                  MPI 4
+#                  OMP 2
+#                  COMMAND ${CMAKE_BINARY_DIR}/bin/saber_quench_error_covariance_toolbox.x
+#                  ARGS testinput/dirac_spectralb_parallel_gauss_vader_6.yaml
+#                  DEPENDS saber_quench_error_covariance_toolbox.x )
 
   # dirac_spectral_parallel_gauss_vader_7
   # (should give similar results to dirac_spectral_gauss_vader_7_1_1)
-  saber_add_test( TARGET saber_dirac_spectralb_parallel_gauss_vader_7_7-1
-                  MPI 7
-                  OMP 1
-                  COMMAND ${CMAKE_BINARY_DIR}/bin/saber_quench_subcomm_error_covariance_toolbox.x
-                  ARGS testinput/dirac_spectralb_parallel_gauss_vader_7.yaml
-                  DEPENDS saber_quench_subcomm_error_covariance_toolbox.x
-                  TEST_DEPENDS saber_process_perts_spectralb_from_gauss_perts_1_1-1 )
+#  saber_add_test( TARGET saber_dirac_spectralb_parallel_gauss_vader_7_7-1
+#                  MPI 7
+#                  OMP 1
+#                  COMMAND ${CMAKE_BINARY_DIR}/bin/saber_quench_subcomm_error_covariance_toolbox.x
+#                  ARGS testinput/dirac_spectralb_parallel_gauss_vader_7.yaml
+#                  DEPENDS saber_quench_subcomm_error_covariance_toolbox.x
+#                  TEST_DEPENDS saber_process_perts_spectralb_from_gauss_perts_1_1-1 )
 
   # dirac_write_fields_with_smv
   saber_add_test( TARGET saber_dirac_write_fields_with_smv_1-1
