--- /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/oops/CMakeLists.txt.tmp.bak	2024-07-15 08:34:47.242335585 +0200
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/oops/CMakeLists.txt	2024-07-15 08:06:39.376227997 +0200
@@ -11,17 +11,31 @@
 ErrorCovariance.h
 Localization.h
 
-# Application
+# Applications
 ErrorCovarianceToolbox.h
 ProcessPerts.h
+AssimEnsemble.h
+EnsembleVariance.h
+EvilUpdate.h
+MakeObs.h
+BGOS.h
 
 # Utilities
+ECUtilities.cc
+ECUtilities.h
 Utilities.h
 Utilities.cc
 
 # Factories
 instantiateCovarFactory.h
 instantiateLocalizationFactory.h
+instantiateMinFactory.h
+
+# EVIL
+RitzPairs.h
+PLanczosEVILMinimizer.h
+RPLanczosEVILMinimizer.h
+SQRTPLanczosEVILMinimizer.h
 )
 
 set( oops_src_files ${oops_src_files_list} PARENT_SCOPE )
