--- /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/oops/CMakeLists.txt.tmp.bak	2024-11-21 12:25:50.747765089 +0100
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/oops/CMakeLists.txt	2024-11-20 14:41:00.801851288 +0100
@@ -22,6 +22,28 @@
 # Factories
 instantiateCovarFactory.h
 instantiateLocalizationFactory.h
+
+# Other applications
+AssimEnsemble.h
+EnsembleVariance.h
+MakeObs.h
+BGOS.h
+SyntheticData.h
+
+# EVIL
+DRPLanczosEVILMinimizer.h
+EvilUpdate.h
+instantiateMinFactory.h
+RitzPairs.h
+PLanczosEVILMinimizer.h
+RPLanczosEVILMinimizer.h
+SQRTPLanczosEVILMinimizer.h
+
+# Temporary versions for cy46
+DRPLanczosEVILMinimizer_cy46.h
+EvilUpdate_cy46.h
+instantiateMinFactory_cy46.h
+RitzPairs_cy46.h
 )
 
 set( oops_src_files ${oops_src_files_list} PARENT_SCOPE )
