--- /home/benjaminm/code/ecsaber/bundle/ecsaber/saber/src/saber/oops/ErrorCovarianceToolbox.h.tmp.bak	2026-03-02 14:06:07.611077624 +0100
+++ /home/benjaminm/code/ecsaber/bundle/ecsaber/saber/src/saber/oops/ErrorCovarianceToolbox.h	2026-03-02 14:07:29.241590160 +0100
@@ -27,17 +27,16 @@
 #include "eckit/exception/Exceptions.h"
 #include "eckit/mpi/Comm.h"
 
-#include "oops/base/Geometry.h"
-#include "oops/base/Increment.h"
+#include "oops/interface/Geometry.h"
+#include "oops/interface/Increment.h"
 #include "oops/base/Increment4D.h"
-#include "oops/base/instantiateCovarFactory.h"
-#include "oops/base/ModelSpaceCovarianceBase.h"
+#include "saber/oops/instantiateCovarFactory.h"
+#include "oops/base/ModelSpaceCovariance4DBase.h"
 #include "oops/base/PostProcessor.h"
-#include "oops/base/State.h"
+#include "oops/interface/State.h"
 #include "oops/base/State4D.h"
-#include "oops/base/StateWriter.h"
-#include "oops/base/Variables.h"
-#include "oops/mpi/mpi.h"
+#include "oops/interface/Model.h"
+#include "oops/interface/Variables.h"
 #include "oops/runs/Application.h"
 #include "oops/util/ConfigFunctions.h"
 #include "oops/util/ConfigHelpers.h"
@@ -51,6 +50,7 @@
 
 #include "saber/blocks/SaberParametricBlockChain.h"
 #include "saber/oops/ErrorCovarianceParameters.h"
+#include "oops/util/ECUtilities.h"
 #include "saber/oops/Utilities.h"
 #include "saber/util/HorizontalProfiles.h"
 
@@ -207,8 +207,8 @@
     const auto & diracParams = params.dirac.value();
     if (diracParams != boost::none) {
       // Setup Dirac field
-      Increment4D_ dxi(geom, vars, xx.times(), *commTime);
-      dxi.dirac(*diracParams);
+      Increment4D_ dxi(geom, util::templatedVars<MODEL>(vars), xx.times());
+      dirac4D(*diracParams, dxi);
       oops::Log::test() << "Input Dirac increment:" << dxi << std::endl;
 
       // Test configuration
@@ -270,15 +270,15 @@
 
     // Create diagnostic field
     Increment4D_ diagPoints(data);
-    diagPoints.dirac(diagConf);
+    dirac4D(diagConf, diagPoints);
 
     // Get diagnostic values
-    for (size_t jj = 0; jj < data.size(); ++jj) {
-      util::printDiagValues(diagPoints.commTime(),
-                            geom.getComm(),
-                            geom.functionSpace(),
-                            data[jj].fieldSet().fieldSet(),
-                            diagPoints[jj].fieldSet().fieldSet());
+    for (size_t jj = 0; jj < data.times().size(); ++jj) {
+      util::printDiagValues(oops::mpi::myself(),
+                            geom.geometry().getComm(),
+                            geom.geometry().functionSpace(),
+                            data[jj].increment().fieldSet(),
+                            diagPoints[jj].increment().fieldSet());
     }
 
     oops::Log::trace() << appname() << "::print_value_at_position done" << std::endl;
@@ -384,7 +384,7 @@
     oops::Log::trace() << appname() << "::dirac starting" << std::endl;
 
     // Get covariance model
-    const std::string covarianceModel(covarConf.getString("covariance model"));
+    const std::string covarianceModel(covarConf.getString("covariance"));
 
     // Update ID
     std::string updId = id;
@@ -404,8 +404,11 @@
     Increment4D_ dxo(dxi, false);
 
     // Covariance
-    std::unique_ptr<CovarianceBase_> Bmat(CovarianceFactory_::create(
-                                          geom, vars, covarConf, xx, xx));
+    std::unique_ptr<Covariance4DBase_> Bmat(Covariance4DFactory_::create(
+                                            covarConf, geom, util::templatedVars<MODEL>(vars), xx));
+
+    // Linearize
+    Bmat->linearize(xx, geom, covarConf);
 
     // Multiply
     Bmat->multiply(dxi, dxo);
@@ -494,10 +497,18 @@
       Increment4D_ dxo(dxi);
 
       // Setup localization
-      Localization_ Lmat(geom, vars, locConfig);
+      Localization_ Lmat(geom, util::templatedVars<MODEL>(vars), locConfig);
 
       // Apply localization
-      Lmat.multiply(dxo);
+      Increment_ dxoSum(dxo[0]);
+      for (size_t jsub = 1; jsub < dxo.times().size(); ++jsub) {
+        dxoSum.axpy(1.0, dxo[jsub], false);
+      }
+      Lmat.multiply(dxoSum);
+      for (size_t jsub = 0; jsub < dxo.times().size(); ++jsub) {
+        dxo[jsub].zero();
+        dxo[jsub].axpy(1.0, dxoSum, false);
+      }
 
       // Copy configuration
       eckit::LocalConfiguration outputLConf(testConf.getSubConfiguration("output dirac"));
@@ -542,7 +553,7 @@
         for (const auto & conf : confs) {
           // Prepare sub-covariance configuration
           eckit::LocalConfiguration componentConfig(conf, "covariance");
-          componentConfig.set("covariance model", "SABER");
+          componentConfig.set("covariance", "SABER");
 
           // Merge configuration with full configuration (order of arguments matters!)
           componentConfig = util::mergeConfigs(componentConfig, paramsBase.toConfiguration());
@@ -587,11 +598,16 @@
                                              locConfig);
 
         // Define output increment
-        Increment4D_ dxo(dxi);
-        oops::FieldSet4D fset4dDxo(dxo);
+        oops::FieldSet4D fset4dDx(dxi);
 
         // Apply localization
-        Lmat.multiply(fset4dDxo);
+        Lmat.multiply(fset4dDx);
+
+        // Reset output
+        Increment4D_ dxo(dxi);
+        for (int jsub = 0; jsub < dxo.times().size(); ++jsub) {
+          dxo[jsub].increment().fromFieldSet(fset4dDx[jsub].fieldSet());
+        }
 
         // Copy configuration
         eckit::LocalConfiguration outputLConf(testConf.getSubConfiguration("output dirac"));
