--- /home/benjaminm/code/ecsaber/bundle/ecsaber/saber/src/saber/oops/ErrorCovarianceParameters.h.tmp	2026-01-30 12:45:54.977861980 +0100
+++ /home/benjaminm/code/ecsaber/bundle/ecsaber/saber/src/saber/oops/ErrorCovarianceParameters.h	2026-01-30 12:12:58.773109218 +0100
@@ -52,7 +52,7 @@
   /// ModelSpaceCovarianceBase class parameters
 
   // Covariance model
-  oops::OptionalParameter<std::string> covarianceModel{"covariance model", this};
+  oops::OptionalParameter<std::string> covarianceModel{"covariance", this};
 
   // Randomization size
   oops::OptionalParameter<size_t> randomizationSize{"randomization size", this};
