--- /home/benjaminm/code/ecsaber/bundle/ecsaber/saber/src/saber/oops/ErrorCovarianceToolbox.h.tmp.bak	2026-02-02 14:49:16.344056056 +0100
+++ /home/benjaminm/code/ecsaber/bundle/ecsaber/saber/src/saber/oops/ErrorCovarianceToolbox.h	2026-02-02 14:07:52.296750756 +0100
@@ -26,17 +26,16 @@
 #include "eckit/exception/Exceptions.h"
 #include "eckit/mpi/Comm.h"
 
-#include "oops/base/Geometry.h"
-#include "oops/base/Increment.h"
+#include "oops/interface/Geometry.h"
+#include "oops/interface/Increment.h"
 #include "oops/base/Increment4D.h"
-#include "oops/base/instantiateCovarFactory.h"
-#include "oops/base/ModelSpaceCovarianceBase.h"
+#include "saber/oops/instantiateCovarFactory.h"
+#include "oops/base/ModelSpaceCovariance4DBase.h"
 #include "oops/base/PostProcessor.h"
-#include "oops/base/State.h"
+#include "oops/interface/State.h"
 #include "oops/base/State4D.h"
-#include "oops/base/StateWriter.h"
-#include "oops/base/Variables.h"
-#include "oops/mpi/mpi.h"
+#include "oops/interface/Model.h"
+#include "oops/interface/Variables.h"
 #include "oops/runs/Application.h"
 #include "oops/util/ConfigFunctions.h"
 #include "oops/util/ConfigHelpers.h"
@@ -49,6 +48,7 @@
 #include "oops/util/parameters/RequiredParameter.h"
 
 #include "saber/oops/ErrorCovarianceParameters.h"
+#include "oops/util/ECUtilities.h"
 #include "saber/oops/Utilities.h"
 #include "saber/util/HorizontalProfiles.h"
 
@@ -104,68 +104,32 @@
 // -----------------------------------------------------------------------------
 
 template <typename MODEL> class ErrorCovarianceToolbox : public oops::Application {
-  typedef oops::ModelSpaceCovarianceBase<MODEL>           CovarianceBase_;
-  typedef oops::CovarianceFactory<MODEL>                  CovarianceFactory_;
+  typedef oops::ModelSpaceCovariance4DBase<MODEL>         Covariance4DBase_;
+  typedef oops::Covariance4DFactory<MODEL>                Covariance4DFactory_;
   typedef oops::Geometry<MODEL>                           Geometry_;
   typedef oops::Increment<MODEL>                          Increment_;
   typedef oops::Increment4D<MODEL>                        Increment4D_;
+  typedef oops::Model<MODEL>                              Model_;
   typedef oops::State<MODEL>                              State_;
   typedef oops::State4D<MODEL>                            State4D_;
   typedef oops::Localization<MODEL>                       Localization_;
+  typedef oops::Variables<MODEL>                          Variables_;
 
  public:
 // -----------------------------------------------------------------------------
-  explicit ErrorCovarianceToolbox(const eckit::mpi::Comm & comm = eckit::mpi::comm()) :
-    Application(comm) {
-    oops::instantiateCovarFactory<MODEL>();
+  ErrorCovarianceToolbox() {
+    instantiateCovarFactory<MODEL>();
   }
 // -----------------------------------------------------------------------------
   virtual ~ErrorCovarianceToolbox() {}
 // -----------------------------------------------------------------------------
-  int execute(const eckit::Configuration & fullConfig) const override {
+  int execute(const eckit::Configuration & fullConfig) const {
     // Deserialize parameters
     ErrorCovarianceToolboxParameters params;
     params.deserialize(fullConfig);
 
-    // Define number of subwindows
-    const eckit::LocalConfiguration backgroundConfig(fullConfig, "background");
-    size_t nsubwin = 1;
-    if (backgroundConfig.has("states")) {
-      std::vector<eckit::LocalConfiguration> confs;
-      backgroundConfig.get("states", confs);
-      nsubwin = confs.size();
-    }
-
     // Define space and time communicators
-    const eckit::mpi::Comm * commSpace = &this->getComm();
-    const eckit::mpi::Comm * commTime = &oops::mpi::myself();
-    if (nsubwin > 1) {
-      // Define sub-windows
-      const size_t ntasks = this->getComm().size();
-      size_t mysubwin = 0;
-      size_t nsublocal = nsubwin;
-      if (params.parallel && (ntasks % nsubwin == 0)) {
-        nsublocal = 1;
-        mysubwin = this->getComm().rank() / (ntasks / nsubwin);
-        ASSERT(mysubwin < nsubwin);
-      } else if (params.parallel) {
-        oops::Log::warning() << "Parallel time subwindows specified in yaml "
-                             << "but number of tasks is not divisible by "
-                             << "the number of subwindows, ignoring." << std::endl;
-      }
-
-      // Create a communicator for same sub-window, to be used for communications in space
-      const std::string sgeom = "comm_geom_" + std::to_string(mysubwin);
-      char const *geomName = sgeom.c_str();
-      commSpace = &this->getComm().split(mysubwin, geomName);
-
-      // Create a communicator for same local area, to be used for communications in time
-      const size_t myarea = commSpace->rank();
-      const std::string stime = "comm_time_" + std::to_string(myarea);
-      char const *timeName = stime.c_str();
-      commTime = &this->getComm().split(myarea, timeName);
-      ASSERT(commTime->size() == (nsubwin / nsublocal));
-    }
+    const eckit::mpi::Comm * commSpace = &eckit::mpi::comm();
 
     // Get number of MPI tasks and OpenMP threads
     size_t ntasks = commSpace->size();
@@ -184,22 +148,30 @@
     util::seekAndReplace(fullConfigUpdated, "_OMP_", std::to_string(nthreads));
     params.deserialize(fullConfigUpdated);
 
+    // Set precision for test channel
+    oops::Log::test() << std::scientific
+                      << std::setprecision(std::numeric_limits<double>::digits10+1);
+
     // Setup geometry
-    const Geometry_ geom(params.geometry, *commSpace, *commTime);
+    const Geometry_ geom(params.geometry);
+
+    // Setup model
+    eckit::LocalConfiguration modelConf;
+    if (fullConfigUpdated.has("model")) {
+      modelConf = fullConfigUpdated.getSubConfiguration("model");
+    }
+    const Model_ model(geom, modelConf);
 
     // Setup background
-    const State4D_ xx(geom, params.background.value(), *commTime);
+    const State4D_ xx(params.background, geom, model);
 
     // Setup variables
-    oops::JediVariables tmpVars = xx.variables();
+    oops::JediVariables tmpVars(xx[0].state().fieldSet().field_names());
     if (params.incrementVars.value() != boost::none) {
       tmpVars = params.incrementVars.value().value();
     }
     const oops::JediVariables vars = tmpVars;
 
-    // Setup time
-    util::DateTime time = xx[0].validTime();
-
     // Covariance parameters
     const eckit::LocalConfiguration covarConf = params.backgroundError.value().toConfiguration();
 
@@ -207,8 +179,8 @@
     const auto & diracParams = params.dirac.value();
     if (diracParams != boost::none) {
       // Setup Dirac field
-      Increment4D_ dxi(geom, vars, xx.times(), *commTime);
-      dxi.dirac(*diracParams);
+      Increment4D_ dxi(geom, util::templatedVars<MODEL>(vars), xx.times());
+      dirac4D(*diracParams, dxi);
       oops::Log::test() << "Input Dirac increment:" << dxi << std::endl;
 
       // Test configuration
@@ -236,18 +208,21 @@
 
       // Apply B matrix components recursively
       std::string id;
-      dirac(covarConf, testConf, id, geom, vars, xx, dxi);
+      dirac(covarConf, testConf, id, geom, util::templatedVars<MODEL>(vars), xx, dxi);
     }
 
     // Background error covariance base parameters
     if ((!diracParams) || (params.backgroundError.value().randomizationSize.value())) {
       // Background error covariance training
-      std::unique_ptr<CovarianceBase_> Bmat(CovarianceFactory_::create(
-                                            geom, vars, covarConf, xx, xx));
+      std::unique_ptr<Covariance4DBase_> Bmat(Covariance4DFactory_::create(
+                                              covarConf, geom, util::templatedVars<MODEL>(vars), xx));
+
+      // Linearize
+      Bmat->linearize(xx, geom, covarConf);
 
       // Randomization
       if (params.backgroundError.value().randomizationSize.value()) {
-        randomization(params, geom, vars, xx, Bmat, ntasks);
+        randomization(params, geom, util::templatedVars<MODEL>(vars), xx, Bmat, ntasks);
       }
     }
 
@@ -267,15 +242,15 @@
 
     // Create diagnostic field
     Increment4D_ diagPoints(data);
-    diagPoints.dirac(diagConf);
+    dirac4D(diagConf, diagPoints);
 
     // Get diagnostic values
-    for (size_t jj = 0; jj < data.size(); ++jj) {
-      util::printDiagValues(diagPoints.commTime(),
-                            geom.getComm(),
-                            geom.functionSpace(),
-                            data[jj].fieldSet().fieldSet(),
-                            diagPoints[jj].fieldSet().fieldSet());
+    for (size_t jj = 0; jj < data.times().size(); ++jj) {
+      util::printDiagValues(oops::mpi::myself(),
+                            geom.geometry().getComm(),
+                            geom.geometry().functionSpace(),
+                            data[jj].increment().fieldSet(),
+                            diagPoints[jj].increment().fieldSet());
     }
 
     oops::Log::trace() << appname() << "::print_value_at_position done" << std::endl;
@@ -288,13 +263,13 @@
                               const Increment4D_ & data) const {
   oops::Log::trace() << appname() << "::extract_1d_covariances starting" << std::endl;
 
-  if (data.size() > 1) {
+  if (data.times().size() > 1) {
     throw eckit::NotImplemented("Not implemented for 4D covariances", Here());
   }
 
   // Create dirac field
   Increment4D_ diracPoints(data);
-  diracPoints.dirac(diracConf);
+  dirac4D(diracConf, diracPoints);
 
   // Define maximum length of horizontal profile
   double maxLength = std::numeric_limits<double>::infinity();
@@ -303,7 +278,7 @@
     maxLength = profileConf.getDouble("maximum distance");
   } else {
     // If no maximum length input is given, try to compute a default value from the grid
-    const auto & fspace = geom.functionSpace();
+    const auto & fspace = geom.geometry().functionSpace();
     if (fspace.type() == "StructuredColumns") {
       const atlas::functionspace::StructuredColumns fs(fspace);
       if (fs.grid().name().compare(0, 1, "F") == 0) {
@@ -330,19 +305,19 @@
 
   // Get values as a function of separation distance
   auto[distances, covariances, lons, lats, levs, fieldIndexes] =
-      util::sortBySeparationDistance(geom.getComm(),
-                                     geom.functionSpace(),
-                                     data[0].fieldSet().fieldSet(),
-                                     diracPoints[0].fieldSet().fieldSet(),
+      util::sortBySeparationDistance(geom.geometry().getComm(),
+                                     geom.geometry().functionSpace(),
+                                     data[0].increment().fieldSet(),
+                                     diracPoints[0].increment().fieldSet(),
                                      maxLength,
                                      removeDuplicates);
 
   // Write to file or to test Log
-  const auto & names = data[0].fieldSet().fieldSet().field_names();
+  const auto & names = data[0].increment().fieldSet().field_names();
   if (profileConf.has("output filepath")) {
     // Write to file
     const auto outputPath = profileConf.getString("output filepath");
-    util::write_1d_covariances(geom.getComm(),
+    util::write_1d_covariances(geom.geometry().getComm(),
                                distances,
                                covariances,
                                lons,
@@ -372,15 +347,18 @@
              const eckit::LocalConfiguration & testConf,
              std::string & id,
              const Geometry_ & geom,
-             const oops::JediVariables & vars,
+             const Variables_ & vars,
              const State4D_ & xx,
              const Increment4D_ & dxi) const {
     // Define output increment
     Increment4D_ dxo(dxi, false);
 
     // Covariance
-    std::unique_ptr<CovarianceBase_> Bmat(CovarianceFactory_::create(
-                                          geom, vars, covarConf, xx, xx));
+    std::unique_ptr<Covariance4DBase_> Bmat(Covariance4DFactory_::create(
+                                            covarConf, geom, vars, xx));
+
+    // Linearize
+    Bmat->linearize(xx, geom, covarConf);
 
     // Multiply
     Bmat->multiply(dxi, dxo);
@@ -426,7 +404,7 @@
     oops::Log::test() << "Covariance(" << id << ") * Increment:" << dxo << std::endl;
 
     // Look for hybrid or ensemble covariance models
-    const std::string covarianceModel(covarConf.getString("covariance model"));
+    const std::string covarianceModel(covarConf.getString("covariance"));
     bool runComponentsRecursively =
       covarConf.has("run components recursively") ?
       covarConf.getBool("run components recursively") : false;
@@ -464,7 +442,7 @@
           for (const auto & conf : confs) {
             std::string idC(id + std::to_string(componentIndex));
             eckit::LocalConfiguration componentConfig(conf, "covariance");
-            componentConfig.set("covariance model", "SABER");
+            componentConfig.set("covariance", "SABER");
             if (covarConf.has("adjoint test")) {
               componentConfig.set("adjoint test", covarConf.getBool("adjoint test"));
             }
@@ -492,7 +470,15 @@
       Localization_ Lmat(geom, vars, locConfig);
 
       // Apply localization
-      Lmat.multiply(dxo);
+      Increment_ dxoSum(dxo[0]);
+      for (size_t jsub = 1; jsub < dxo.times().size(); ++jsub) {
+        dxoSum.axpy(1.0, dxo[jsub], false);
+      }
+      Lmat.multiply(dxoSum);
+      for (size_t jsub = 0; jsub < dxo.times().size(); ++jsub) {
+        dxo[jsub].zero();
+        dxo[jsub].axpy(1.0, dxoSum, false);
+      }
 
       // Update ID
       std::string idL(id);
@@ -523,9 +509,9 @@
 // -----------------------------------------------------------------------------
   void randomization(const ErrorCovarianceToolboxParameters & params,
                      const Geometry_ & geom,
-                     const oops::JediVariables & vars,
+                     const Variables_ & vars,
                      const State4D_ & xx,
-                     const std::unique_ptr<CovarianceBase_> & Bmat,
+                     const std::unique_ptr<Covariance4DBase_> & Bmat,
                      const size_t & ntasks) const {
     // Get randomization size
     ASSERT(params.backgroundError.value().randomizationSize.value());
@@ -537,9 +523,9 @@
       oops::Log::info() << "Info     : -----------------------" << std::endl;
 
       // Create increments
-      Increment4D_ dx(geom, vars, xx.times(), xx.commTime());
-      Increment4D_ dxsq(geom, vars, xx.times(), xx.commTime());
-      Increment4D_ variance(geom, vars, xx.times(), xx.commTime());
+      Increment4D_ dx(geom, vars, xx.times());
+      Increment4D_ dxsq(geom, vars, xx.times());
+      Increment4D_ variance(geom, vars, xx.times());
 
       // Initialize variance
       variance.zero();
@@ -574,7 +560,9 @@
 
             // Add background state to perturbation
             State4D_ xp(xx);
-            xp += dx;
+            for (int jsub = 0; jsub < dx.times().size(); ++jsub) {
+              xp[jsub-dx.first()] += dx[jsub];
+            }
 
             // Write state
             xp.write(outputStatesUpdated);
@@ -585,7 +573,9 @@
 
         // Square perturbation
         dxsq = dx;
-        dxsq.schur_product_with(dx);
+        for (int jsub = 0; jsub < dx.times().size(); ++jsub) {
+          dxsq[jsub].schur_product_with(dx[jsub]);
+        }
 
         // Update variance
         variance += dxsq;
