--- /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/oops/Localization.h.tmp.bak	2024-06-12 18:30:30.621788997 +0200
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/oops/Localization.h	2024-06-12 18:22:28.564179769 +0200
@@ -1,8 +1,6 @@
 /*
- * (C) Copyright 2021 UCAR
+ * (C) Copyright 2023 Meteorologisk Institutt
  *
- * This software is licensed under the terms of the Apache Licence Version 2.0
- * which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
  */
 
 #pragma once
@@ -17,13 +15,19 @@
 #include "eckit/config/Configuration.h"
 
 #include "oops/base/FieldSets.h"
-#include "oops/base/State4D.h"
 #include "oops/base/Variables.h"
+#include "oops/generic/GenericCtlVec.h"
 #include "oops/generic/LocalizationBase.h"
+#include "oops/interface/Geometry.h"
+#include "oops/interface/Increment.h"
+#include "oops/interface/State.h"
+#include "oops/interface/Variables.h"
 #include "oops/util/Duration.h"
+#include "oops/util/FieldSetHelpers.h"
 #include "oops/util/Logger.h"
 
 #include "saber/blocks/SaberParametricBlockChain.h"
+#include "saber/oops/ECUtilities.h"
 #include "saber/oops/Utilities.h"
 
 namespace saber {
@@ -32,19 +36,26 @@
 
 template<typename MODEL>
 class Localization : public oops::LocalizationBase<MODEL> {
-  typedef oops::Geometry<MODEL>               Geometry_;
-  typedef oops::Increment<MODEL>              Increment_;
-  typedef oops::State<MODEL>                  State_;
+  using GenericCtlVec_ = oops::GenericCtlVec;
+  using Geometry_ = oops::Geometry<MODEL>;
+  using Increment_ = oops::Increment<MODEL>;
+  using Model_ = oops::Model<MODEL>;
+  using State_ = oops::State<MODEL>;
+  using Variables_ = oops::Variables<MODEL>;
 
  public:
   Localization(const Geometry_ &,
-               const oops::JediVariables &,
+               const Variables_ &,
                const eckit::Configuration &);
   ~Localization();
 
-  void randomize(Increment_ &) const override;
   void multiply(Increment_ &) const override;
 
+  // Square-root formulation
+  size_t ctlVecSize() const {return loc_->ctlVecSize();}
+  void multiplySqrt(const GenericCtlVec_ & dv, Increment_ & dx) const;
+  void multiplySqrtTrans(const Increment_ & dx, GenericCtlVec_ & dv) const;
+
  private:
   void print(std::ostream &) const override;
   std::unique_ptr<SaberParametricBlockChain> loc_;
@@ -54,7 +65,7 @@
 
 template<typename MODEL>
 Localization<MODEL>::Localization(const Geometry_ & geom,
-                                  const oops::JediVariables & incVarsNoMeta,
+                                  const Variables_ & incVarsNoMeta,
                                   const eckit::Configuration & conf)
   : loc_()
 {
@@ -64,21 +75,17 @@
   util::DateTime dummyTime(1977, 5, 25, 0, 0, 0);
 
   // Initialize
-  const std::vector<std::size_t> vlevs = geom.variableSizes(incVarsNoMeta);
-  oops::JediVariables incVars(incVarsNoMeta);
+  const std::vector<std::size_t> vlevs = geom.geometry().variableSizes(incVarsNoMeta.variables());
+  oops::JediVariables incVars(incVarsNoMeta.variables().variablesList());
   for (std::size_t i = 0; i < vlevs.size() ; ++i) {
     incVars[i].setLevels(vlevs[i]);
   }
 
   // Create dummy xb and fg
-  const State_ xb_state(geom, incVars, dummyTime);
-  oops::FieldSet3D xb(dummyTime, geom.getComm());
-  xb.shallowCopy(xb_state.fieldSet());
-  oops::FieldSet4D xb4d(xb);
-  const State_ fg_state(geom, incVars, dummyTime);
-  oops::FieldSet3D fg(dummyTime, geom.getComm());
-  fg.shallowCopy(fg_state.fieldSet());
-  oops::FieldSet4D fg4d(fg);
+  oops::FieldSet3D fset3d(dummyTime, eckit::mpi::comm());
+  fset3d.deepCopy(util::createFieldSet(geom.geometry().functionSpace(), incVars, 0.0));
+  oops::FieldSet4D fset4dXb(fset3d);
+  oops::FieldSet4D fset4dFg(fset3d);
 
   oops::FieldSets emptyFsetEns({}, oops::mpi::myself(), {}, oops::mpi::myself());
   // TODO(AS): revisit what configuration needs to be passed to SaberParametricBlockChain.
@@ -99,7 +106,7 @@
   covarConf.set("time covariance", "univariate");
   // Initialize localization blockchain
   loc_ = std::make_unique<SaberParametricBlockChain>(geom, geom,
-              incVars, xb4d, fg4d,
+              incVars, fset4dXb, fset4dFg,
               emptyFsetEns, emptyFsetEns, covarConf, conf);
 
   oops::Log::trace() << "Localization:Localization done" << std::endl;
@@ -115,18 +122,35 @@
 // -----------------------------------------------------------------------------
 
 template<typename MODEL>
-void Localization<MODEL>::randomize(Increment_ & dx) const {
-  oops::Log::trace() << "Localization:randomize starting" << std::endl;
+void Localization<MODEL>::multiplySqrt(const GenericCtlVec_ & dv,
+                                       Increment_ & dx) const {
+  oops::Log::trace() << "Localization:multiplySqrt starting" << std::endl;
 
-  // SABER block chain randomization
-  oops::FieldSet3D fset3d(dx.validTime(), dx.geometry().getComm());
+  // SABER block chain square-root
+  oops::FieldSet3D fset3d(dx.validTime(), eckit::mpi::comm());
   oops::FieldSet4D fset4d(fset3d);
-  loc_->randomize(fset4d);
+  loc_->multiplySqrt(dv.data(), fset4d, 0);
 
   // ATLAS fieldset to Increment_
-  dx.fromFieldSet(fset4d[0].fieldSet());
+  dx.increment().fromFieldSet(fset4d[0].fieldSet());
+
+  oops::Log::trace() << "Localization:multiplySqrt done" << std::endl;
+}
+
+// -----------------------------------------------------------------------------
+
+template<typename MODEL>
+void Localization<MODEL>::multiplySqrtTrans(const Increment_ & dx,
+                                            GenericCtlVec_ & dv) const {
+  oops::Log::trace() << "Localization:multiplySqrtTrans starting" << std::endl;
+
+  // SABER block chain square-root adjoint
+  oops::FieldSet3D fset3d(dx.validTime(), eckit::mpi::comm());
+  fset3d.shallowCopy(dx.increment().fieldSet());
+  oops::FieldSet4D fset4d(fset3d);
+  loc_->multiplySqrtAD(fset4d, dv.data(), 0);
 
-  oops::Log::trace() << "Localization:randomize done" << std::endl;
+  oops::Log::trace() << "Localization:multiplySqrtTrans done" << std::endl;
 }
 
 // -----------------------------------------------------------------------------
@@ -136,12 +160,12 @@
   oops::Log::trace() << "Localization:multiply starting" << std::endl;
 
   // SABER block chain multiplication
-  oops::FieldSet4D fset4d({dx.validTime(), dx.geometry().getComm()});
-  fset4d[0].shallowCopy(dx.fieldSet());
+  oops::FieldSet4D fset4d({dx.validTime(), eckit::mpi::comm()});
+  fset4d[0].shallowCopy(dx.increment().fieldSet());
   loc_->multiply(fset4d);
 
   // ATLAS fieldset to Increment_
-  dx.synchronizeFields();
+  dx.increment().synchronizeFields();
 
   oops::Log::trace() << "Localization:multiply done" << std::endl;
 }
