--- /home/benjaminm/code/ecsaber/bundle/ecsaber/saber/src/saber/oops/Localization.h.tmp.bak	2026-03-12 12:21:09.856412742 +0100
+++ /home/benjaminm/code/ecsaber/bundle/ecsaber/saber/src/saber/oops/Localization.h	2026-03-12 12:14:44.385306949 +0100
@@ -17,14 +17,20 @@
 #include "eckit/config/Configuration.h"
 
 #include "oops/base/FieldSets.h"
-#include "oops/base/State.h"
+#include "oops/base/State4D.h"
 #include "oops/base/Variables.h"
+#include "oops/generic/GenericCtlVec.h"
 #include "oops/generic/LocalizationBase.h"
+#include "oops/interface/Geometry.h"
+#include "oops/interface/Increment.h"
+#include "oops/interface/Variables.h"
 #include "oops/util/Duration.h"
+#include "oops/util/FieldSetHelpers.h"
 #include "oops/util/Logger.h"
 #include "oops/util/Timer.h"
 
 #include "saber/blocks/SaberParametricBlockChain.h"
+#include "oops/util/ECUtilities.h"
 #include "saber/oops/Utilities.h"
 
 namespace saber {
@@ -33,20 +39,26 @@
 
 template<typename MODEL>
 class Localization : public oops::LocalizationBase<MODEL> {
+  typedef oops::GenericCtlVec                 GenericCtlVec_;
   typedef oops::Geometry<MODEL>               Geometry_;
   typedef oops::Increment<MODEL>              Increment_;
-  typedef oops::State<MODEL>                  State_;
+  typedef oops::Model<MODEL>                  Model_;
+  typedef oops::Variables<MODEL>              Variables_;
 
  public:
   Localization(const Geometry_ &,
-               const oops::JediVariables &,
+               const Variables_ &,
                const eckit::Configuration &);
   ~Localization();
   static const std::string classname() {return "saber::Localization";}
 
-  void randomize(Increment_ &) const override;
   void multiply(Increment_ &) const override;
 
+  // Square-root formulation
+  size_t ctlVecSize() const {return loc_->ctlVecSize();}
+  void multiplySqrt(const GenericCtlVec_ & dv, Increment_ & dx) const;
+  void multiplySqrtTrans(const Increment_ & dx, GenericCtlVec_ & dv) const;
+
  private:
   void print(std::ostream &) const override;
   std::unique_ptr<SaberParametricBlockChain> loc_;
@@ -56,7 +68,7 @@
 
 template<typename MODEL>
 Localization<MODEL>::Localization(const Geometry_ & geom,
-                                  const oops::JediVariables & incVarsNoMeta,
+                                  const Variables_ & incVarsNoMeta,
                                   const eckit::Configuration & conf)
   : loc_()
 {
@@ -67,21 +79,27 @@
   util::DateTime dummyTime(1977, 5, 25, 0, 0, 0);
 
   // Initialize increment variables with levels metadata
-  const std::vector<std::size_t> vlevs = geom.variableSizes(incVarsNoMeta);
-  oops::JediVariables incVars(incVarsNoMeta);
+  const std::vector<std::size_t> vlevs = geom.geometry().variableSizes(incVarsNoMeta.variables());
+  oops::JediVariables incVars(incVarsNoMeta.variables().variables());
   for (std::size_t i = 0; i < vlevs.size() ; ++i) {
     incVars[i].setLevels(vlevs[i]);
   }
 
+  // Create dummy FieldSet3D
+  oops::FieldSet3D fset3d(dummyTime, eckit::mpi::comm());
+  fset3d.deepCopy(util::createFieldSet(geom.geometry().functionSpace(), incVars, 0.0));
+
+  // Add nearest 3D level metadata if needed
+  if (conf.has("nearest 3d level")) {
+    const int nearest3dLevel = conf.getInt("nearest 3d level");
+    for (auto & field : fset3d) {
+      field.metadata().set("nearest 3d level", nearest3dLevel);
+    }
+  }
+
   // Create dummy xb and fg
-  const State_ xb_state(geom, incVars, dummyTime);
-  oops::FieldSet3D xb(dummyTime, geom.getComm());
-  xb.shallowCopy(xb_state.fieldSet());
-  oops::FieldSet4D xb4d(xb);
-  const State_ fg_state(geom, incVars, dummyTime);
-  oops::FieldSet3D fg(dummyTime, geom.getComm());
-  fg.shallowCopy(fg_state.fieldSet());
-  oops::FieldSet4D fg4d(fg);
+  oops::FieldSet4D xb4d(fset3d);
+  oops::FieldSet4D fg4d(fset3d);
 
   // 3D localization always used here (4D aspects handled in oops::Localization),
   // so this parameter can be anything.
@@ -106,21 +124,37 @@
 // -----------------------------------------------------------------------------
 
 template<typename MODEL>
-void Localization<MODEL>::randomize(Increment_ & dx) const {
-  util::Timer timer(classname(), "randomize");
-  oops::Log::trace() << "Localization:randomize starting" << std::endl;
+void Localization<MODEL>::multiplySqrt(const GenericCtlVec_ & dv,
+                                       Increment_ & dx) const {
+  util::Timer timer(classname(), "multiplySqrt");
+  oops::Log::trace() << "Localization:multiplySqrt starting" << std::endl;
 
-  // Create 4D fieldset
-  oops::FieldSet3D fset3d(dx.validTime(), dx.geometry().getComm());
+  // SABER block chain square-root
+  oops::FieldSet3D fset3d(dx.validTime(), eckit::mpi::comm());
   oops::FieldSet4D fset4d(fset3d);
-
-  // SABER block chain randomization
-  loc_->randomize(fset4d);
+  loc_->multiplySqrt(dv.data(), fset4d, 0);
 
   // ATLAS fieldset to Increment_
-  dx.fromFieldSet(fset4d[0].fieldSet());
+  dx.increment().fromFieldSet(fset4d[0].fieldSet());
+
+  oops::Log::trace() << "Localization:multiplySqrt done" << std::endl;
+}
+
+// -----------------------------------------------------------------------------
+
+template<typename MODEL>
+void Localization<MODEL>::multiplySqrtTrans(const Increment_ & dx,
+                                            GenericCtlVec_ & dv) const {
+  util::Timer timer(classname(), "multiplySqrtTrans");
+  oops::Log::trace() << "Localization:multiplySqrtTrans starting" << std::endl;
+
+  // SABER block chain square-root adjoint
+  oops::FieldSet3D fset3d(dx.validTime(), eckit::mpi::comm());
+  fset3d.shallowCopy(dx.increment().fieldSet());
+  oops::FieldSet4D fset4d(fset3d);
+  loc_->multiplySqrtAD(fset4d, dv.data(), 0);
 
-  oops::Log::trace() << "Localization:randomize done" << std::endl;
+  oops::Log::trace() << "Localization:multiplySqrtTrans done" << std::endl;
 }
 
 // -----------------------------------------------------------------------------
@@ -131,14 +165,14 @@
   oops::Log::trace() << "Localization:multiply starting" << std::endl;
 
   // Create 4D fieldset
-  oops::FieldSet4D fset4d({dx.validTime(), dx.geometry().getComm()});
-  fset4d[0].shallowCopy(dx.fieldSet());
+  oops::FieldSet4D fset4d({dx.validTime(), eckit::mpi::comm()});
+  fset4d[0].shallowCopy(dx.increment().fieldSet());
 
   // SABER block chain multiplication
   loc_->multiply(fset4d);
 
   // ATLAS fieldset to Increment_
-  dx.fromFieldSet(fset4d[0].fieldSet());
+  dx.increment().synchronizeFields();
 
   oops::Log::trace() << "Localization:multiply done" << std::endl;
 }
