--- /home/benjaminm/code/ecsaber/bundle/ecsaber/saber/src/saber/oops/Utilities.h.tmp.bak	2026-02-09 13:47:04.409680732 +0100
+++ /home/benjaminm/code/ecsaber/bundle/ecsaber/saber/src/saber/oops/Utilities.h	2026-02-09 13:53:10.174589209 +0100
@@ -21,13 +21,12 @@
 #include "eckit/exception/Exceptions.h"
 
 #include "oops/base/FieldSets.h"
-#include "oops/base/Geometry.h"
-#include "oops/base/Increment.h"
-#include "oops/base/IncrementSet.h"
+#include "oops/interface/Geometry.h"
+#include "oops/interface/Increment.h"
 #include "oops/base/State4D.h"
-#include "oops/base/StateSet.h"
 #include "oops/base/Variables.h"
 #include "oops/interface/ModelData.h"
+#include "oops/interface/Variables.h"
 #include "oops/util/ConfigFunctions.h"
 #include "oops/util/DateTime.h"
 #include "oops/util/FieldSetHelpers.h"
@@ -37,6 +36,7 @@
 #include "oops/util/ParallelFieldSetIO.h"
 
 #include "saber/blocks/SaberBlockParametersBase.h"
+#include "oops/util/ECUtilities.h"
 
 namespace oops {
   class FieldSet3D;
@@ -153,19 +153,26 @@
     if (!ensembleConf.empty()) {
       oops::Log::info() << "Info     : Ensemble of states, perturbation using the mean"
                         << std::endl;
-      oops::StateSet<MODEL> tmp(geom, ensembleConf, commTime);
-      oops::IncrementSet<MODEL> ensemble(geom, vars, tmp, true);
-      ensemble -= ensemble.ens_mean();
-      oops::FieldSets fsetEns(ensemble);
+      std::vector<eckit::LocalConfiguration> memConfs;
+      std::vector<int> ensmems;
+      for (size_t ie = 0; ie < getNensFromConfig(ensembleConf); ++ie) {
+        memConfs.push_back(getEnsSubconfig(ensembleConf, ie));
+        ensmems.push_back(ie);
+      }
+      oops::FieldSets fsetEns(geom, vars, times, ensmems, memConfs, true);
       return fsetEns;
     }
 
     // Increment ensemble from increments on disk
     if (!ensemblePertConf.empty()) {
       oops::Log::info() << "Info     : Increment ensemble from increments on disk" << std::endl;
-      oops::IncrementSet<MODEL> ensemble(geom, vars, times,
-                                         ensemblePertConf, commTime);
-      oops::FieldSets fsetEns(ensemble);
+      std::vector<eckit::LocalConfiguration> memConfs;
+      std::vector<int> ensmems;
+      for (size_t ie = 0; ie < getNensFromConfig(ensemblePertConf); ++ie) {
+        memConfs.push_back(getEnsSubconfig(ensemblePertConf, ie));
+        ensmems.push_back(ie);
+      }
+      oops::FieldSets fsetEns(geom, vars, times, ensmems, memConfs, false);
       return fsetEns;
     }
 
@@ -173,13 +180,7 @@
     if (!ensembleBaseConf.empty() && !ensemblePairsConf.empty()) {
       oops::Log::info() << "Info     : Increment ensemble from difference of two states"
                         << std::endl;
-      oops::StateSet<MODEL> states1(geom, ensembleBaseConf, commTime);
-      oops::StateSet<MODEL> states2(geom, ensemblePairsConf, commTime);
-      oops::IncrementSet<MODEL> ensemble(geom, vars, states1.times(), states1.commTime(),
-                                         states1.members(), states1.commEns());
-      ensemble.diff(states1, states2);
-      oops::FieldSets fsetEns(ensemble);
-      return fsetEns;
+      throw eckit::Exception("not implemented yet", Here());
     }
 
     // Increment ensemble from increments on disk on other geometry
@@ -233,7 +234,7 @@
         // Read perturbations into oops::FieldSets
         oops::FieldSets fsetEns(fspace, vars, io, times,
                                 ensemblePertOtherGeomConf,
-                                commGeom, commTime);
+                                commGeom, eckit::mpi::self());
 
         return fsetEns;
       } else {
@@ -268,13 +269,13 @@
   eckit::LocalConfiguration localConf(conf);
 
   // Create Increment
-  oops::Increment<MODEL> dx(geom, vars, date);
+  oops::Increment<MODEL> dx(geom, util::templatedVars<MODEL>(vars), date);
 
   // Read file
   dx.read(localConf);
 
   // Get FieldSet
-  fset.shallowCopy(dx.fieldSet());
+  fset.shallowCopy(dx.increment().fieldSet());
 
   oops::Log::trace() << "readHybridWeight done" << std::endl;
 }
@@ -295,13 +296,14 @@
   size_t ensembleFound = 0;
 
   if (conf.has("ensemble")) {
-    // Read state
+    // Read state as increment
     eckit::LocalConfiguration memConf = getEnsSubconfig(
       conf.getSubConfiguration("ensemble"), ie);
-    oops::State<MODEL> xx(geom, memConf);
+    oops::Increment<MODEL> dx(geom, util::templatedVars<MODEL>(vars), fset.validTime());
+    dx.read(memConf);
 
     // Copy FieldSet
-    fset.deepCopy(xx.fieldSet());
+    fset.deepCopy(dx.increment().fieldSet());
 
     ++ensembleFound;
   }
@@ -312,33 +314,18 @@
       conf.getSubConfiguration("ensemble pert"), ie);
 
     // Read Increment
-    oops::Increment<MODEL> dx(geom, vars, fset.validTime());
+    oops::Increment<MODEL> dx(geom, util::templatedVars<MODEL>(vars), fset.validTime());
     dx.read(memConf);
 
     // Get FieldSet
-    fset.deepCopy(dx.fieldSet());
+    fset.deepCopy(dx.increment().fieldSet());
 
     ++ensembleFound;
   }
 
   if (conf.has("ensemble base") && conf.has("ensemble pairs")) {
-    // Increment ensemble from difference of two states
-    eckit::LocalConfiguration memConfBase = getEnsSubconfig(
-      conf.getSubConfiguration("ensemble base"), ie);
-    eckit::LocalConfiguration memConfPairs = getEnsSubconfig(
-      conf.getSubConfiguration("ensemble pairs"), ie);
-
-    // Read states
-    oops::State<MODEL> xxBase(geom, memConfBase);
-    oops::State<MODEL> xxPairs(geom, memConfPairs);
-    // Compute difference
-    oops::Increment<MODEL> dx(geom, vars, fset.validTime());
-    dx.diff(xxPairs, xxBase);
-
-    // Get FieldSet
-    fset.deepCopy(dx.fieldSet());
-
-    ++ensembleFound;
+    throw eckit::NotImplemented("readEnsembleMember not yet implemented for a"
+                                "difference of two states", Here());
   }
 
   if (conf.has("ensemble pert on other geometry")
