--- /home/benjaminm/code/ecsaber/bundle/ecsaber/saber/src/saber/oops/Utilities.cc.tmp	2026-01-30 16:40:46.151465584 +0100
+++ /home/benjaminm/code/ecsaber/bundle/ecsaber/saber/src/saber/oops/Utilities.cc	2026-01-30 16:27:46.135220604 +0100
@@ -88,7 +88,6 @@
   return activeVars;
 }
 
-
 // -----------------------------------------------------------------------------
 
 // Return inner variables as outer variables + inner active variables
