--- /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/bump/lib/type_bump_parameters.h.tmp	2024-02-14 13:20:32.207915600 +0100
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/bump/lib/type_bump_parameters.h	2024-02-13 13:47:02.558174007 +0100
@@ -490,6 +490,10 @@
 
 // Fit section
 struct FitDef {
+  // Threshold to filter out lower raw values
+  std::pair<const char *, double> diag_raw_th =
+    std::make_pair("raw diagnostic lower threshold", -std::numeric_limits<double>().max());
+
   // Horizontal filtering suport radius [in meters]
   std::pair<const char *, double> diag_rhflt =
     std::make_pair("horizontal filtering length-scale", 0.0);
