--- /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/bump/lib/type_nicas_blk.fypp.tmp	2024-02-20 08:01:58.410651567 +0100
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/bump/lib/type_nicas_blk.fypp	2024-02-20 08:00:06.721399683 +0100
@@ -1408,7 +1408,7 @@
 deallocate(cv_blk2%cmp)
 
 ! Reset random seed if necessary
-if (nam%default_seed) call rng%reseed(mpl)
+call rng%reseed(mpl)
 
 ! Probe out
 @:probe_out()
@@ -1492,7 +1492,7 @@
 if (nicas_blk%verbosity) call mpl%flush
 
 ! Reset random seed if necessary
-if (nam%default_seed) call rng%reseed(mpl)
+call rng%reseed(mpl)
 
 ! Probe out
 @:probe_out()
