--- /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/bump/lib/type_nicas.fypp.tmp	2024-02-20 08:01:58.054718844 +0100
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/bump/lib/type_nicas.fypp	2024-02-20 08:00:06.721399683 +0100
@@ -880,7 +880,7 @@
 call nicas%partial_dealloc
 
 ! Reset random seed if necessary
-if (nam%default_seed) call rng%reseed(mpl)
+call rng%reseed(mpl)
 
 ! Probe out
 @:probe_out()
@@ -1437,7 +1437,7 @@
 call ens%normalize(mpl,nam,geom)
 
 ! Reset random seed if necessary
-if (nam%default_seed) call rng%reseed(mpl)
+call rng%reseed(mpl)
 
 ! Probe out
 @:probe_out()
@@ -1545,7 +1545,7 @@
 call mpl%flush
 
 ! Reset random seed if necessary
-if (nam%default_seed) call rng%reseed(mpl)
+call rng%reseed(mpl)
 
 ! Probe out
 @:probe_out()
@@ -1732,7 +1732,7 @@
 deallocate(fld_save)
 
 ! Reset random seed if necessary
-if (nam%default_seed) call rng%reseed(mpl)
+call rng%reseed(mpl)
 
 ! Probe out
 @:probe_out()
