--- /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/bump/lib/type_bump_parameters.cc.tmp	2024-02-20 08:01:58.178695410 +0100
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/bump/lib/type_bump_parameters.cc	2024-02-20 08:00:06.721399683 +0100
@@ -37,7 +37,7 @@
   param(generalDef.color_log, generalConf);
   // Stream test messages into a dedicated channel
   param(generalDef.testing, generalConf);
-  // Default seed for random numbers
+  // Default seed for random numbers (0 for time-dependent seed)
   param(generalDef.default_seed, generalConf);
   // Inter-compilers reproducibility
   param(generalDef.repro_ops, generalConf);
