--- /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/bump/lib/type_bump.fypp.tmp	2024-02-14 13:20:32.275915638 +0100
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/bump/lib/type_bump.fypp	2024-02-13 13:47:05.138175826 +0100
@@ -1782,9 +1782,17 @@
 
    ! Select parameter from mom
    select case (trim(param))
+   case ('dirac_mom_cov')
+      if (.not.allocated(bump%mom(1)%dirac)) call bump%mpl%abort('${subr}$',trim(param)//' is not allocated')
+      fld_c0a = bump%mom(1)%dirac(:,:,iv,1)
+      found = .true.
+   case ('dirac_mom_single_obs')
+      if (.not.allocated(bump%mom(1)%dirac)) call bump%mpl%abort('${subr}$',trim(param)//' is not allocated')
+      fld_c0a = bump%mom(1)%dirac(:,:,iv,2)
+      found = .true.
    case ('dirac_mom')
       if (.not.allocated(bump%mom(1)%dirac)) call bump%mpl%abort('${subr}$',trim(param)//' is not allocated')
-      fld_c0a = bump%mom(1)%dirac(:,:,iv)
+      fld_c0a = bump%mom(1)%dirac(:,:,iv,3)
       found = .true.
    end select
 
