--- /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/bump/lib/type_nam.fypp.tmp	2024-02-20 08:01:58.274677268 +0100
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/bump/lib/type_nam.fypp	2024-02-20 08:00:06.721399683 +0100
@@ -21,7 +21,7 @@
 
 type nam_type
    ! General section
-   logical :: default_seed                                    !< Default seed for random numbers
+   integer :: default_seed                                    !< Default seed for random numbers (0 for time-dependent seed)
    logical :: repro_ops                                       !< Reproducibility operators
    real(kind_real) :: repro_th                                !< Reproducibility threshold
    real(kind_real) :: universe_radius                         !< Universe radius [in meters]
