--- /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/bump/lib/type_geom.fypp.tmp.bak	2024-02-26 14:06:25.545899763 +0100
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/bump/lib/type_geom.fypp	2024-02-26 12:52:51.487928214 +0100
@@ -340,9 +340,9 @@
 
 ! Copy function space pointer
 if (present(fieldset)) then
-   call geom%from_atlas(mpl,afunctionspace,fieldset)
+   call geom%from_atlas(mpl,nam,afunctionspace,fieldset)
 else
-   call geom%from_atlas(mpl,afunctionspace)
+   call geom%from_atlas(mpl,nam,afunctionspace)
 end if
 
 ! Setup subset Sc0
@@ -427,7 +427,7 @@
 end do
 
 ! Reset random seed if necessary
-if (nam%default_seed) call rng%reseed(mpl)
+call rng%reseed(mpl)
 
 ! Probe out
 @:probe_out()
@@ -438,18 +438,19 @@
 ! Subroutine: geom_from_atlas
 !> Set geometry from fieldset
 !----------------------------------------------------------------------
-subroutine geom_from_atlas(geom,mpl,afunctionspace,fieldset)
+subroutine geom_from_atlas(geom,mpl,nam,afunctionspace,fieldset)
 
 implicit none
 
 ! Passed variables
 class(geom_type),intent(inout) :: geom                 !< Geometry
 type(mpl_type),intent(inout) :: mpl                    !< MPI data
+type(nam_type),intent(in) :: nam                       !< Namelist
 type(atlas_functionspace),intent(in) :: afunctionspace !< ATLAS function space
 type(fieldset_type),intent(in),optional :: fieldset    !< Fieldset
 
 ! Local variables
-integer :: il0,imga
+integer :: il0,imga,ifield
 real(kind_real),allocatable :: area_mga(:,:)
 real(kind_real),pointer :: real_ptr(:,:)
 type(atlas_field) :: afield,afield_lonlat
@@ -510,14 +511,14 @@
 
    ! Get vertical coordinate
    if (fieldset%has_field('vert_coord')) then
-      afield = fieldset%field('vert_coord')
+      call fieldset%select_associated_field(mpl,nam%nv,nam%variables(1:nam%nv),nam%var2d(1:nam%nv),'vert_coord',afield)
       call field_to_array(afield,mpl,geom%vert_coord_mga)
       call afield%final()
    end if
 
    ! Get geometry mask
    if (fieldset%has_field('gmask')) then
-      afield = fieldset%field('gmask')
+      call fieldset%select_associated_field(mpl,nam%nv,nam%variables(1:nam%nv),nam%var2d(1:nam%nv),'gmask',afield)
       call field_to_array(afield,mpl,geom%gmask_mga)
       call afield%final()
    end if
@@ -878,7 +879,7 @@
 end do
 
 ! Reset random seed if necessary
-if (nam%default_seed) call rng%reseed(mpl)
+call rng%reseed(mpl)
 
 ! Communicate data from halo A to universe on subset Sc0
 write(mpl%info,'(a10,a)') '','Communicate data from halo A to universe on subset Sc0'
@@ -2560,7 +2561,7 @@
 call rng%desync(mpl)
 
 ! Reset random seed if necessary
-if (nam%default_seed) call rng%reseed(mpl)
+call rng%reseed(mpl)
 
 ! Probe out
 @:probe_out()
