--- /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/bump/lib/type_diag.fypp.tmp	2024-02-14 13:20:32.175915582 +0100
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/bump/lib/type_diag.fypp	2024-02-13 13:46:59.758172034 +0100
@@ -654,6 +654,16 @@
       end do
    end if
 
+   if (.not.lhyb_coef_only) then
+      ! Normalize amplitude
+      do il0=1,geom%nl0
+         do ic0a=1,geom%nc0a
+            if (geom%gmask_c0a(ic0a,il0)) diag%blk(0,ig)%a_c0a(ic0a,il0,:) = diag%blk(0,ig)%a_c0a(ic0a,il0,:) &
+ & /sum(diag%blk(0,ig)%a_c0a(ic0a,il0,:))
+         end do
+      end do
+   end if
+
    ! Release memory
    deallocate(profile)
    if (nam%local_diag) then
