--- /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/bump/lib/type_nicas_cmp.fypp.tmp	2024-02-20 08:01:58.334665929 +0100
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/bump/lib/type_nicas_cmp.fypp	2024-02-20 08:00:06.721399683 +0100
@@ -5176,7 +5176,7 @@
 call cv_cmp2%dealloc
 
 ! Reset random seed if necessary
-if (nam%default_seed) call rng%reseed(mpl)
+call rng%reseed(mpl)
 
 ! End associate
 end associate
@@ -5258,7 +5258,7 @@
 call mpl%flush(flush=nicas_cmp%verbosity)
 
 ! Reset random seed if necessary
-if (nam%default_seed) call rng%reseed(mpl)
+call rng%reseed(mpl)
 
 ! End associate
 end associate
