--- /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/bump/lib/type_samp.fypp.tmp	2024-02-20 08:01:57.918744714 +0100
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/bump/lib/type_samp.fypp	2024-02-20 08:00:06.721399683 +0100
@@ -1880,7 +1880,7 @@
 end if
 
 ! Reset random seed if necessary
-if (nam%default_seed) call rng%reseed(mpl)
+call rng%reseed(mpl)
 
 ! Probe out
 @:probe_out()
@@ -1956,7 +1956,7 @@
 end if
 
 ! Reset random seed if necessary
-if (nam%default_seed) call rng%reseed(mpl)
+call rng%reseed(mpl)
 
 ! Probe out
 @:probe_out()
