--- /home/benjaminm/code/ecsaber/bundle/ecsaber/saber/src/saber/CMakeLists.txt.tmp.bak	2026-05-20 09:51:07.887290819 +0200
+++ /home/benjaminm/code/ecsaber/bundle/ecsaber/saber/src/saber/CMakeLists.txt	2026-05-20 09:51:43.989921481 +0200
@@ -23,12 +23,13 @@
 endif()
 
 target_link_libraries( ${PROJECT_NAME} PUBLIC NetCDF::NetCDF_Fortran NetCDF::NetCDF_C )
-target_link_libraries( ${PROJECT_NAME} PUBLIC MPI::MPI_Fortran )
+target_link_libraries( ${PROJECT_NAME} PUBLIC MPI::MPI_Fortran MPI::MPI_CXX )
 target_link_libraries( ${PROJECT_NAME} PUBLIC ${LAPACK_LIBRARIES} )
 target_link_libraries( ${PROJECT_NAME} PUBLIC eckit )
 target_link_libraries( ${PROJECT_NAME} PUBLIC fckit )
 target_link_libraries( ${PROJECT_NAME} PUBLIC atlas_f )
 target_link_libraries( ${PROJECT_NAME} PUBLIC ${oops_LIBRARIES} )
+target_link_libraries( ${PROJECT_NAME} PUBLIC oops_patch )
 target_link_libraries( ${PROJECT_NAME} PUBLIC vader )
 
 if( ECTRANS_FOUND AND ectrans_HAVE_ETRANS AND ectrans_HAVE_TRANSI )
