--- /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/CMakeLists.txt.tmp	2024-07-04 13:56:20.132085431 +0200
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/CMakeLists.txt	2024-07-04 11:12:45.507034084 +0200
@@ -29,6 +29,7 @@
 target_link_libraries( ${PROJECT_NAME} PUBLIC fckit )
 target_link_libraries( ${PROJECT_NAME} PUBLIC atlas_f )
 target_link_libraries( ${PROJECT_NAME} PUBLIC ${oops_LIBRARIES} )
+target_link_libraries( ${PROJECT_NAME} PUBLIC oops-patch )
 
 if( gsibec_FOUND )
     target_link_libraries( ${PROJECT_NAME} PUBLIC gsibec )
