--- /home/benjaminm/code/oops-bundle/ecsaber/saber/quench/src/CMakeLists.txt.tmp.bak	2024-06-12 18:33:07.086319771 +0200
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/quench/src/CMakeLists.txt	2024-06-03 16:20:19.306188951 +0200
@@ -8,22 +8,69 @@
 Covariance.h
 Fields.cc
 Fields.h
+FieldsIOBase.cc
+FieldsIOBase.h
 Geometry.cc
 Geometry.h
+GeoVaLs.cc
+GeoVaLs.h
+HorizScaleDecomposition.h
+HybridWeight.cc
+HybridWeight.h
 Increment.cc
 Increment.h
-LinearVariableChange.h
-LinearVariableChange.cc
-ModelData.h
+Interpolation.cc
+Interpolation.h
+IncrEnsCtlVec.h
+IncrModCtlVec.h
+instantiateQuenchMatrices.h
+Interpolator.h
+LinearModel.cc
+LinearModel.h
+LinearObsOperator.cc
+LinearObsOperator.h
+LocalizationMatrix.h
+Locations.cc
+Locations.h
+Logbook.cc
+Logbook.h
+Model.cc
+Model.h
+ModelAuxControl.h
+ModelAuxCovariance.h
+ModelAuxCtlVec.h
+ModelAuxControlEstimator.h
+ModelAuxIncrement.h
+ObsAuxControl.cc
+ObsAuxControl.h
+ObsAuxCovariance.cc
+ObsAuxCovariance.h
+ObsAuxCtlVec.cc
+ObsAuxCtlVec.h
+ObsAuxIncrement.cc
+ObsAuxIncrement.h
+ObsError.cc
+ObsError.h
+ObsOperator.cc
+ObsOperator.h
+ObsSpace.cc
+ObsSpace.h
+ObsVector.cc
+ObsVector.h
 State.cc
 State.h
-VariableChange.h
+Traits.h
+Variables.h
+Variables.cc
+
+default/FieldsIODefault.cc
+default/FieldsIODefault.h
 
 )
 
 ecbuild_add_library( TARGET   quench
                      SOURCES  ${quench_src_files}
-                     PUBLIC_LIBS  oops saber
+                     PUBLIC_LIBS  oops oops-patch saber
                      HEADER_DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/${PROJECT_NAME}/quench
                      INSTALL_HEADERS LISTED
                      LINKER_LANGUAGE CXX )
