--- /home/benjaminm/code/oops-bundle/ecsaber/saber/quench/src/Covariance.h.tmp	2024-06-03 16:18:03.357814346 +0200
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/quench/src/Covariance.h	2024-06-03 10:49:46.064740347 +0200
@@ -17,6 +17,7 @@
 #include "oops/util/ObjectCounter.h"
 #include "oops/util/Printable.h"
 
+#include "src/CovarianceECInc.h"
 #include "src/Increment.h"
 
 namespace quench {
@@ -59,6 +60,9 @@
   // Print
   void print(std::ostream & os) const
     {os << "Covariance";}
+
+/// ECSABER-specific definitions
+#include "src/CovarianceECDef.h"
 };
 
 // -----------------------------------------------------------------------------
