--- /home/benjaminm/code/oops-bundle/ecsaber/saber/quench/src/Fields.cc.tmp.bak	2024-06-12 17:37:23.479549688 +0200
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/quench/src/Fields.cc	2024-06-12 17:41:55.605923867 +0200
@@ -1046,4 +1046,8 @@
 
 // -----------------------------------------------------------------------------
 
+#include "src/FieldsECImp.h"
+
+// -----------------------------------------------------------------------------
+
 }  // namespace quench
