--- /home/benjaminm/code/oops-bundle/ecsaber/saber/quench/src/Fields.cc.tmp.bak	2024-07-19 09:24:15.003436742 +0200
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/quench/src/Fields.cc	2024-07-19 09:13:51.952149249 +0200
@@ -1115,4 +1115,8 @@
 
 // -----------------------------------------------------------------------------
 
+#include "src/FieldsECImp.h"
+
+// -----------------------------------------------------------------------------
+
 }  // namespace quench
