--- /home/benjaminm/code/oops-bundle/ecsaber/saber/quench/src/Fields.cc.tmp.bak	2024-09-25 13:10:31.165084498 +0200
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/quench/src/Fields.cc	2024-09-25 13:11:28.968320614 +0200
@@ -1393,4 +1393,8 @@
 
 // -----------------------------------------------------------------------------
 
+#include "src/FieldsECImp.h"
+
+// -----------------------------------------------------------------------------
+
 }  // namespace quench
