--- /home/benjaminm/code/oops-bundle/ecsaber/saber/quench/src/Fields.h.tmp.bak	2024-07-04 11:14:00.303282393 +0200
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/quench/src/Fields.h	2024-07-04 11:12:45.491034031 +0200
@@ -21,6 +21,7 @@
 #include "oops/util/Printable.h"
 #include "oops/util/Serializable.h"
 
+#include "src/FieldsECInc.h"
 #include "src/Interpolation.h"
 
 namespace quench {
@@ -115,6 +116,9 @@
 
   // Fieldset
   mutable atlas::FieldSet fset_;
+
+/// ECSABER-specific definitions
+#include "src/FieldsECDef.h"
 };
 
 // -----------------------------------------------------------------------------
