--- /home/benjaminm/code/oops-bundle/ecsaber/saber/quench/src/Fields.cc.tmp.bak	2024-11-05 09:32:18.449363967 +0100
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/quench/src/Fields.cc	2024-11-05 09:34:39.280458135 +0100
@@ -1416,4 +1416,8 @@
 
 // -----------------------------------------------------------------------------
 
+#include "src/FieldsECImp.h"
+
+// -----------------------------------------------------------------------------
+
 }  // namespace quench
