--- /home/benjaminm/code/oops-bundle/ecsaber/saber/quench/src/Geometry.h.tmp.bak	2024-11-05 09:32:18.320362895 +0100
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/quench/src/Geometry.h	2024-11-05 09:34:01.065175233 +0100
@@ -28,6 +28,8 @@
 #include "oops/util/parameters/RequiredParameter.h"
 #include "oops/util/Printable.h"
 
+#include "src/Variables.h"
+
 namespace eckit {
   class Configuration;
 }
@@ -279,6 +281,9 @@
 
   // Duplicate points
   bool duplicatePoints_;
+
+/// ECSABER-specific interface
+#include "src/GeometryECDef.h"
 };
 
 // -----------------------------------------------------------------------------
