--- /home/benjaminm/code/oops-bundle/ecsaber/saber/quench/src/Geometry.h.tmp.bak	2024-07-04 11:14:00.315282433 +0200
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/quench/src/Geometry.h	2024-07-04 11:26:41.797983708 +0200
@@ -27,6 +27,8 @@
 #include "oops/util/parameters/RequiredParameter.h"
 #include "oops/util/Printable.h"
 
+#include "src/Variables.h"
+
 namespace eckit {
   class Configuration;
 }
@@ -265,6 +267,9 @@
 
   // Interpolation configuration
   eckit::LocalConfiguration interpolation_;
+
+/// ECSABER-specific interface
+#include "src/GeometryECDef.h"
 };
 
 // -----------------------------------------------------------------------------
