--- /home/benjaminm/code/oops-bundle/ecsaber/saber/quench/src/Geometry.h.tmp	2024-07-19 08:41:32.035710038 +0200
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/quench/src/Geometry.h	2024-07-19 09:08:40.054046842 +0200
@@ -28,6 +28,8 @@
 #include "oops/util/parameters/RequiredParameter.h"
 #include "oops/util/Printable.h"
 
+#include "src/Variables.h"
+
 namespace eckit {
   class Configuration;
 }
@@ -271,6 +273,9 @@
 
   // Duplicate points
   bool duplicatePoints_;
+
+/// ECSABER-specific interface
+#include "src/GeometryECDef.h"
 };
 
 // -----------------------------------------------------------------------------
