--- /home/benjaminm/code/oops-bundle/ecsaber/saber/quench/src/Increment.h.tmp	2024-07-19 08:41:32.211710154 +0200
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/quench/src/Increment.h	2024-07-19 09:11:54.927414843 +0200
@@ -23,6 +23,7 @@
 #include "oops/util/Serializable.h"
 
 #include "src/Fields.h"
+#include "src/IncrementECInc.h"
 #include "src/State.h"
 
 namespace quench {
@@ -117,6 +118,9 @@
 
   // Fields
   std::unique_ptr<Fields> fields_;
+
+/// ECSABER-specific definitions
+#include "src/IncrementECDef.h"
 };
 
 // -----------------------------------------------------------------------------
