--- /home/benjaminm/code/oops-bundle/ecsaber/saber/quench/src/Increment.cc.tmp.bak	2024-07-04 11:14:00.487283009 +0200
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/quench/src/Increment.cc	2024-07-04 11:12:45.495034045 +0200
@@ -144,4 +144,8 @@
 
 // -----------------------------------------------------------------------------
 
+#include "src/IncrementECImp.h"
+
+// -----------------------------------------------------------------------------
+
 }  // namespace quench
