--- /home/benjaminm/code/oops-bundle/ecsaber/saber/quench/src/Increment.h.tmp.bak	2024-07-04 11:14:00.503283063 +0200
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/quench/src/Increment.h	2024-07-04 11:12:45.495034045 +0200
@@ -22,6 +22,7 @@
 #include "oops/util/Serializable.h"
 
 #include "src/Fields.h"
+#include "src/IncrementECInc.h"
 #include "src/State.h"
 
 namespace quench {
@@ -116,6 +117,9 @@
 
   // Fields
   std::unique_ptr<Fields> fields_;
+
+/// ECSABER-specific definitions
+#include "src/IncrementECDef.h"
 };
 
 // -----------------------------------------------------------------------------
