--- /home/benjaminm/code/oops-bundle/ecsaber/saber/quench/src/Interpolation.cc.tmp	2024-07-19 08:41:32.079710067 +0200
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/quench/src/Interpolation.cc	2024-07-19 09:10:00.918639944 +0200
@@ -37,6 +37,8 @@
   if (type == "atlas interpolation wrapper") {
     atlasInterpWrapper_ = std::make_shared<saber::interpolation::AtlasInterpWrapper>(srcPartitioner,
       srcFspace, dstGrid, dstFspace);
+  } else if (type == "regional interpolation") {
+    regionalInterpolation_ = std::make_shared<RegionalInterpolation>(srcFspace, dstFspace);
   } else {
     throw eckit::Exception("wrong interpolation type", Here());
   }
@@ -52,6 +54,8 @@
 
   if (atlasInterpWrapper_) {
     atlasInterpWrapper_->execute(srcFieldSet, targetFieldSet);
+  } else if (regionalInterpolation_) {
+    regionalInterpolation_->execute(srcFieldSet, targetFieldSet);
   }
 
   oops::Log::trace() << classname() << "::execute done" << std::endl;
@@ -65,6 +69,8 @@
 
   if (atlasInterpWrapper_) {
     atlasInterpWrapper_->executeAdjoint(srcFieldSet, targetFieldSet);
+  } else if (regionalInterpolation_) {
+    regionalInterpolation_->executeAdjoint(srcFieldSet, targetFieldSet);
   }
 
   oops::Log::trace() << classname() << "::executeAdjoint done" << std::endl;
