--- /home/benjaminm/code/oops-bundle/ecsaber/saber/quench/src/Interpolation.h.tmp.bak	2024-08-27 12:41:02.092471934 +0200
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/quench/src/Interpolation.h	2024-08-27 12:51:27.147877792 +0200
@@ -94,6 +94,9 @@
   // Regional ATLAS interpolation
   std::shared_ptr<atlas::Interpolation> regionalInterp_;
 
+  // Regional interpolation
+  std::shared_ptr<RegionalInterpolation> regionalInterpolation_;
+
   // Vertical interpolations
   std::unordered_map<std::string, std::vector<InterpElement>> verInterps_;
 };
