--- /home/benjaminm/code/oops-bundle/ecsaber/saber/quench/src/State.h.tmp.bak	2024-06-12 17:37:23.235547196 +0200
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/quench/src/State.h	2024-06-12 17:41:53.853910752 +0200
@@ -17,6 +17,7 @@
 #include "oops/util/Printable.h"
 
 #include "src/Fields.h"
+#include "src/StateECInc.h"
 
 namespace eckit {
   class Configuration;
@@ -108,6 +109,9 @@
 
   // Fields
   std::unique_ptr<Fields> fields_;
+
+/// ECSABER-specific definitions
+#include "src/StateECDef.h"
 };
 
 // -----------------------------------------------------------------------------
