--- /home/benjaminm/code/oops-bundle/ecsaber/saber/quench/src/VariableChangeParameters.h.tmp	2024-07-04 13:56:20.812087337 +0200
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/quench/src/VariableChangeParameters.h	2024-07-04 11:12:45.507034084 +0200
@@ -8,16 +8,14 @@
 #pragma once
 
 #include "oops/base/VariableChangeParametersBase.h"
-#include "oops/util/Printable.h"
 
 namespace quench {
 
 // -------------------------------------------------------------------------------------------------
-/// \brief Parameters passed to the VariableChange class.
+/// VariableChange parameters class
 
 class VariableChangeParameters : public oops::VariableChangeParametersBase {
   OOPS_CONCRETE_PARAMETERS(VariableChangeParameters, VariableChangeParametersBase)
- public:
 };
 
 // -------------------------------------------------------------------------------------------------
