--- /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/fastlam/InterpElement.h.tmp	2024-03-14 14:49:29.520329147 +0100
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/fastlam/InterpElement.h	2024-03-11 11:23:03.830424874 +0100
@@ -46,7 +46,7 @@
   // Interpolation type ("c": colocated, "x": linear along x, "y": linear along y, "b": bilinear)
   std::string interpType_;
 
-  // Global indices of the most southwestern point
+  // Global indices of the most southwestern point or of the lower level
   size_t index1_;
   size_t index2_;
 
