--- /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/fastlam/LayerSpec.h.tmp.bak	2024-04-04 09:54:15.836297747 +0200
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/fastlam/LayerSpec.h	2024-03-21 09:28:46.629916978 +0100
@@ -33,13 +33,14 @@
 
   // Constructor
   LayerSpec(const FastLAMParametersBase & params,
+            const eckit::LocalConfiguration & fieldsMetaData,
             const oops::GeometryData & gdata,
             const std::string & myGroup,
             const std::vector<std::string> & myVars,
             const size_t & nx0,
             const size_t & ny0,
             const size_t & nz0) :
-    LayerBase(params, gdata, myGroup, myVars, nx0, ny0, nz0) {}
+    LayerBase(params, fieldsMetaData, gdata, myGroup, myVars, nx0, ny0, nz0) {}
     ~LayerSpec() = default;
 
   // Setups
