--- /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/fastlam/LayerHalo.cc.tmp	2024-03-14 14:49:29.432328975 +0100
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/fastlam/LayerHalo.cc	2024-03-11 16:29:49.264895022 +0100
@@ -29,6 +29,9 @@
 void LayerHalo::setupParallelization() {
   oops::Log::trace() << classname() << "::setupParallelization starting" << std::endl;
 
+  // Check that we are not using fake levels
+  ASSERT(fakeLevels_.size() == 0);
+
   // Get index fields
   atlas::Field fieldIndexI = fset_["index_i"];
   atlas::Field fieldIndexJ = fset_["index_j"];
