--- /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/fastlam/LayerBase.h.tmp.bak	2024-04-15 14:43:59.329841856 +0200
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/fastlam/LayerBase.h	2024-04-04 15:35:09.290252483 +0200
@@ -167,6 +167,7 @@
   atlas::FieldSet fset_;
 
   // Reduced grid <=> model grid (interpolation)
+  bool noInterp_;
   size_t rSendSize_;
   size_t mRecvSize_;
   std::vector<int> rSendCounts_;
