--- /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/fastlam/LayerRC.h.tmp	2024-03-07 12:22:36.543915254 +0100
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/fastlam/LayerRC.h	2024-03-07 11:36:54.473904631 +0100
@@ -32,11 +32,12 @@
   // Constructor
   LayerRC(const FastLAMParametersBase & params,
           const oops::GeometryData & gdata,
-          const std::string & myVar,
+          const std::string & myGroup,
+          const std::vector<std::string> & myVars,
           const size_t & nx0,
           const size_t & ny0,
           const size_t & nz0) :
-    LayerBase(params, gdata, myVar, nx0, ny0, nz0) {}
+    LayerBase(params, gdata, myGroup, myVars, nx0, ny0, nz0) {}
     ~LayerRC() = default;
 
   // Setups
