--- /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/fastlam/LayerRC.cc.tmp	2024-02-26 11:26:08.855352741 +0100
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/fastlam/LayerRC.cc	2024-02-26 11:18:49.766734346 +0100
@@ -340,7 +340,7 @@
 
   // Full normalization
   atlas::Field normField = gdata_.functionSpace().createField<double>(
-    atlas::option::name(myVar_) | atlas::option::levels(nz0_));
+    atlas::option::name(myGroup_) | atlas::option::levels(nz0_));
   auto normView = atlas::array::make_view<double, 2>(normField);
   norm_.add(normField);
 
@@ -578,7 +578,7 @@
     // Compute normalization accuracy
     oops::Log::info() << "Info     :     Compute exact normalization" << std::endl;
     atlas::Field normAccField = gdata_.functionSpace().createField<double>(
-      atlas::option::name(myVar_) | atlas::option::levels(nz0_));
+      atlas::option::name(myGroup_) | atlas::option::levels(nz0_));
     auto normAccView = atlas::array::make_view<double, 2>(normAccField);
     normAccView.assign(util::missingValue<double>());
     double normAccMax = 0.0;;
