--- /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/fastlam/LayerRC.h.tmp.bak	2024-04-04 09:54:15.752297615 +0200
+++ /home/benjaminm/code/oops-bundle/ecsaber/saber/src/saber/fastlam/LayerRC.h	2024-03-21 09:18:42.537615296 +0100
@@ -31,13 +31,14 @@
 
   // Constructor
   LayerRC(const FastLAMParametersBase & params,
+          const eckit::LocalConfiguration & fieldsMetaData,
           const oops::GeometryData & gdata,
           const std::string & myGroup,
           const std::vector<std::string> & myVars,
           const size_t & nx0,
           const size_t & ny0,
           const size_t & nz0) :
-    LayerBase(params, gdata, myGroup, myVars, nx0, ny0, nz0) {}
+    LayerBase(params, fieldsMetaData, gdata, myGroup, myVars, nx0, ny0, nz0) {}
     ~LayerRC() = default;
 
   // Setups
