--- /home/benjaminm/code/oops-bundle/ecsaber/oops-patch/src/oops/base/GeometryData.h.tmp	2024-03-07 11:39:32.118082412 +0100
+++ /home/benjaminm/code/oops-bundle/ecsaber/oops-patch/src/oops/base/GeometryData.h	2024-03-07 11:31:54.721563075 +0100
@@ -50,7 +50,7 @@
 // Accessors
   const atlas::FunctionSpace & functionSpace() const {return fspace_;}
   const atlas::FieldSet & fieldSet() const {return fset_;}
-  const eckit::mpi::Comm & comm() const {return *comm_;}
+  const eckit::mpi::Comm & comm() const {return comm_;}
 
   bool has(const std::string & name) const {return fset_.has(name);}
   const atlas::Field & getField(const std::string & name) const {return fset_.field(name);}
@@ -62,7 +62,7 @@
 
   atlas::FunctionSpace fspace_;
   atlas::FieldSet fset_;
-  const eckit::mpi::Comm * comm_;
+  const eckit::mpi::Comm & comm_;
   bool topdown_;
 
   atlas::Mesh mesh_;
