--- /home/benjaminm/code/oops-bundle/ecsaber/oops-patch/src/oops/base/Variables.cc.tmp.bak	2024-03-07 12:39:56.552680472 +0100
+++ /home/benjaminm/code/oops-bundle/ecsaber/oops-patch/src/oops/base/Variables.cc	2024-03-07 12:29:57.072212152 +0100
@@ -17,7 +17,6 @@
 
 #include "eckit/config/Configuration.h"
 #include "eckit/config/LocalConfiguration.h"
-#include "eckit/exception/Exceptions.h"
 
 #include "eckit/types/Types.h"
 #include "eckit/utils/Hash.h"
@@ -27,16 +26,17 @@
 
 // -----------------------------------------------------------------------------
 namespace oops {
+namespace patch {
 // -----------------------------------------------------------------------------
 
-patch::Variables::Variables()
+Variables::Variables()
   : convention_(""), vars_(0), varMetaData_() {
   Log::trace() << "Variables::Variables" << std::endl;
 }
 
 // -----------------------------------------------------------------------------
 
-patch::Variables::Variables(const eckit::Configuration & conf, const std::string & name)
+Variables::Variables(const eckit::Configuration & conf, const std::string & name)
   : convention_(""), vars_(0), channels_(0), varMetaData_() {
   Log::trace() << "Variables::Variables start " << conf << std::endl;
   std::vector<std::string> vars;
@@ -62,7 +62,7 @@
 }
 
 // -----------------------------------------------------------------------------
-patch::Variables::Variables(const std::vector<std::string> & vars, const std::string & conv)
+Variables::Variables(const std::vector<std::string> & vars, const std::string & conv)
   : convention_(conv), vars_(vars), varMetaData_() {
   Log::trace() << "Variables::Variables start " << vars << std::endl;
   Log::trace() << "Variables::Variables done" << std::endl;
@@ -70,7 +70,7 @@
 
 // -----------------------------------------------------------------------------
 
-patch::Variables::Variables(const std::vector<std::string> & vars, const std::vector<int> & channels)
+Variables::Variables(const std::vector<std::string> & vars, const std::vector<int> & channels)
   : convention_(""), vars_(0), channels_(channels), varMetaData_() {
   Log::trace() << "Variables::Variables start " << vars << " @ " << channels << std::endl;
   if (channels.empty()) {
@@ -87,7 +87,7 @@
 
 // -----------------------------------------------------------------------------
 
-patch::Variables::Variables(const eckit::Configuration & conf, const std::vector<std::string> & vars)
+Variables::Variables(const eckit::Configuration & conf, const std::vector<std::string> & vars)
   : convention_(""), vars_(vars), varMetaData_(conf)
 {
   Log::trace() << "Variables::Variables start " << vars << " @ " << conf << std::endl;
@@ -96,14 +96,14 @@
 
 // -----------------------------------------------------------------------------
 
-patch::Variables::Variables(const patch::Variables & other)
+Variables::Variables(const Variables & other)
   : convention_(other.convention_), vars_(other.vars_), channels_(other.channels_),
     varMetaData_(other.varMetaData_)
 {}
 
 // -----------------------------------------------------------------------------
 
-patch::Variables & patch::Variables::operator+=(const patch::Variables & rhs) {
+Variables & Variables::operator+=(const Variables & rhs) {
   ASSERT(convention_ == rhs.convention_);
   vars_.insert(vars_.end(), rhs.vars_.begin(), rhs.vars_.end());
   // revisit late, should we add channels this way ?
@@ -121,18 +121,15 @@
   // this operation adds and updates the metadata in the object from the
   // rhs object.
   for (const std::string & var : rhs.varMetaData_.keys()) {
-    for (const std::string & key : rhs.varMetaData_.getSubConfiguration(var).keys()) {
-      int value(-1);
-      getVariableSubKeyValue(var, key, rhs.varMetaData_, value);
-      setVariableSubKeyValue(var, key, value, varMetaData_);
-    }
+    eckit::LocalConfiguration varConf = rhs.varMetaData_.getSubConfiguration(var);
+    varMetaData_.set(var, varConf);
   }
   return *this;
 }
 
 // -----------------------------------------------------------------------------
 
-patch::Variables & patch::Variables::operator-=(const patch::Variables & rhs) {
+Variables & Variables::operator-=(const Variables & rhs) {
   ASSERT(convention_ == rhs.convention_);
   if (!rhs.channels().empty()) {
     throw eckit::NotImplemented(
@@ -146,14 +143,14 @@
 
 // -----------------------------------------------------------------------------
 
-patch::Variables & patch::Variables::operator-=(const std::string & var) {
+Variables & Variables::operator-=(const std::string & var) {
   vars_.erase(std::remove(vars_.begin(), vars_.end(), var), vars_.end());
   return *this;
 }
 
 // -----------------------------------------------------------------------------
 
-bool patch::Variables::operator==(const patch::Variables & rhs) const {
+bool Variables::operator==(const Variables & rhs) const {
   if ((convention_  != rhs.convention_) ||
       (channels_    != rhs.channels_)   ||
       (vars_.size() != rhs.vars_.size())) {
@@ -186,13 +183,13 @@
 
 // -----------------------------------------------------------------------------
 
-bool patch::Variables::operator!=(const patch::Variables & rhs) const {
+bool Variables::operator!=(const Variables & rhs) const {
   return (!(*this == rhs));
 }
 
 // -----------------------------------------------------------------------------
 
-bool patch::Variables::operator<=(const patch::Variables & rhs) const {
+bool Variables::operator<=(const Variables & rhs) const {
   ASSERT(convention_ == rhs.convention_);
   ASSERT(channels_.empty());
   bool is_in_rhs = true;
@@ -203,15 +200,8 @@
 }
 
 // -----------------------------------------------------------------------------
-void patch::Variables::addMetaData(const std::string & varname,
-                            const std::string & keyname,
-                            const int & keyvalue) {
-  setVariableSubKeyValue(varname, keyname, keyvalue, varMetaData_);
-}
 
-// -----------------------------------------------------------------------------
-
-void patch::Variables::intersection(const patch::Variables & rhs) {
+void Variables::intersection(const Variables & rhs) {
   ASSERT(convention_ == rhs.convention_);
   if (this->size() * rhs.size() > 0) ASSERT(channels_.empty() == rhs.channels_.empty());
 
@@ -239,7 +229,7 @@
 
 // -----------------------------------------------------------------------------
 
-bool patch::Variables::has(const std::string & var) const {
+bool Variables::has(const std::string & var) const {
   bool found = false;
   for (size_t jj = 0; jj < vars_.size(); ++jj) {
     found = found || vars_[jj] == var;
@@ -249,7 +239,7 @@
 
 // -----------------------------------------------------------------------------
 
-size_t patch::Variables::find(const std::string & var) const {
+size_t Variables::find(const std::string & var) const {
   size_t ii = vars_.size();
   for (size_t jj = 0; jj < vars_.size(); ++jj) {
     if (vars_[jj] == var) ii = jj;
@@ -263,20 +253,20 @@
 
 // -----------------------------------------------------------------------------
 
-void patch::Variables::push_back(const std::string & vname) {
+void Variables::push_back(const std::string & vname) {
   vars_.push_back(vname);
 }
 
 // -----------------------------------------------------------------------------
 
-void patch::Variables::sort() {
+void Variables::sort() {
   std::sort(vars_.begin(), vars_.end());
   std::sort(channels_.begin(), channels_.end());
 }
 
 // -----------------------------------------------------------------------------
 
-std::vector<std::string> patch::Variables::asCanonical() const {
+std::vector<std::string> Variables::asCanonical() const {
   std::vector<std::string> vars(vars_);
   std::sort(vars.begin(), vars.end());
   return vars;
@@ -284,7 +274,7 @@
 
 // -----------------------------------------------------------------------------
 
-void patch::Variables::print(std::ostream & os) const {
+void Variables::print(std::ostream & os) const {
   os << vars_.size() << " variables: ";
   for (size_t jj = 0; jj < vars_.size(); ++jj) {
     if (jj > 0) os << ", ";
@@ -297,35 +287,26 @@
 
 // -----------------------------------------------------------------------------
 
-int patch::Variables::getLevels(const std::string & fieldname) const {
-  int levels(-1);
-  getVariableSubKeyValue(fieldname, "levels",
-                         varMetaData_, levels);
-  return levels;
+bool Variables::hasMetaData(const std::string & varname,
+                            const std::string & keyname) const {
+  bool has = false;
+  if (!varMetaData_.empty()) {
+    if (varMetaData_.has(varname)) {
+      has = varMetaData_.getSubConfiguration(varname).has(keyname);
+    }
+  }
+  return has;
 }
 
-// -----------------------------------------------------------------------------
-
-void patch::Variables::getVariableSubKeyValue(const std::string & varname,
-                                       const std::string & keyname,
-                                       const eckit::Configuration & variablesconf,
-                                       int & intvalue) const {
-  ASSERT(!variablesconf.empty());
-  ASSERT(variablesconf.has(varname));
-  ASSERT(variablesconf.getSubConfiguration(varname).has(keyname));
-  variablesconf.getSubConfiguration(varname).get(keyname, intvalue);
-}
 
 // -----------------------------------------------------------------------------
 
-void patch::Variables::setVariableSubKeyValue(const std::string & varname,
-                                       const std::string & keyname,
-                                       const int & keyvalue,
-                                       eckit::LocalConfiguration & variableslconf) {
-  eckit::LocalConfiguration variablelconf =
-    variableslconf.has(varname) ? variableslconf.getSubConfiguration(varname) :
-                                  eckit::LocalConfiguration();
-  variablelconf.set(keyname, keyvalue);
-  variableslconf.set(varname, variablelconf);
+int Variables::getLevels(const std::string & fieldname) const {
+  int levels(-1);
+  getVariableSubKeyValue(fieldname, "levels",
+                         varMetaData_, levels);
+  return levels;
 }
+
+}  // namespace patch
 }  // namespace oops
